// react-native-agora/lib/commonjs/internal/AgoraH265TranscoderInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.H265TranscoderInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _IAgoraH265TranscoderImpl = require("../impl/IAgoraH265TranscoderImpl");
var _IAgoraH265TranscoderTi = _interopRequireDefault(require("../ti/IAgoraH265Transcoder-ti"));
var _event = require("./event");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_IAgoraH265TranscoderTi.default);
class H265TranscoderInternal extends _IAgoraH265TranscoderImpl.IH265TranscoderImpl {
  static _h265_transcoder_observers = [];
  release() {
    H265TranscoderInternal._h265_transcoder_observers = [];
    this.removeAllListeners();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IH265Transc;
    if ((_checkers$IH265Transc = checkers.IH265TranscoderObserver) !== null && _checkers$IH265Transc !== void 0 && _checkers$IH265Transc.strictTest({
      [eventType]: undefined
    })) {
      if (H265TranscoderInternal._h265_transcoder_observers.length === 0) {
        this.registerTranscoderObserver({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== _event.EVENT_TYPE.IAgoraH265Transcoder) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    (0, _event.addScopedEventListener)(this, eventType, callback);
  }
  removeListener(eventType, listener) {
    (0, _event.removeScopedEventListener)(this, eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    (0, _event.removeAllScopedEventListeners)(this, eventType);
  }
  registerTranscoderObserver(observer) {
    if (!H265TranscoderInternal._h265_transcoder_observers.find(value => value === observer)) {
      H265TranscoderInternal._h265_transcoder_observers.push(observer);
    }
    return super.registerTranscoderObserver(observer);
  }
  unregisterTranscoderObserver(observer) {
    H265TranscoderInternal._h265_transcoder_observers = H265TranscoderInternal._h265_transcoder_observers.filter(value => value !== observer);
    return super.unregisterTranscoderObserver(observer);
  }
}
exports.H265TranscoderInternal = H265TranscoderInternal;
