// react-native-agora/lib/commonjs/internal/call.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.callIrisApi = callIrisApi;
var _buffer = require("buffer");
var _base64Js = _interopRequireDefault(require("base64-js"));
var _jsonBigint = _interopRequireDefault(require("json-bigint"));
var _Utils = require("../Utils");
var _specs = _interopRequireDefault(require("../specs"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const JSON = (0, _jsonBigint.default)({
  storeAsString: true
});
/**
 * @internal
 */
function callIrisApi(funcName, params) {
  try {
    const buffers = [];
    if (funcName.startsWith('MediaEngine_')) {
      switch (funcName) {
        case 'MediaEngine_pushAudioFrame_c71f4ab':
          // frame.buffer
          buffers.push(_base64Js.default.fromByteArray(params.frame.buffer ?? _buffer.Buffer.from('')));
          break;
        case 'MediaEngine_pushVideoFrame_4e544e2':
          // frame.buffer
          buffers.push(_base64Js.default.fromByteArray(params.frame.buffer ?? _buffer.Buffer.from('')));
          // frame.eglContext
          buffers.push(_base64Js.default.fromByteArray(_buffer.Buffer.from('')));
          // frame.metadata_buffer
          buffers.push(_base64Js.default.fromByteArray(_buffer.Buffer.from('')));
          // frame.alphaBuffer
          buffers.push(_base64Js.default.fromByteArray(params.frame.alphaBuffer ?? _buffer.Buffer.from('')));
          // frame.d3d11_texture_2d
          buffers.push(_base64Js.default.fromByteArray(_buffer.Buffer.from('')));
          break;
        case 'MediaEngine_pushEncodedVideoImage_e71452b':
          // imageBuffer
          buffers.push(_base64Js.default.fromByteArray(params.imageBuffer ?? _buffer.Buffer.from('')));
          break;
      }
    } else if (funcName.startsWith('MediaPlayer_') || funcName.startsWith('MusicPlayer_')) {
      var _params$toJSON;
      // @ts-ignore
      params.mediaPlayerId = this.getMediaPlayerId();
      const json = (_params$toJSON = params.toJSON) === null || _params$toJSON === void 0 ? void 0 : _params$toJSON.call();
      params.toJSON = function () {
        return {
          ...json,
          playerId: params.mediaPlayerId
        };
      };
    } else if (funcName.startsWith('VideoEffectObject_')) {
      var _params$toJSON2;
      params.videoEffectObjectId =
      // @ts-ignore
      this.getVideoEffectObjectId();
      const json = (_params$toJSON2 = params.toJSON) === null || _params$toJSON2 === void 0 ? void 0 : _params$toJSON2.call();
      params.toJSON = function () {
        return {
          ...json,
          objectId: params.videoEffectObjectId
        };
      };
    } else if (funcName.startsWith('MediaRecorder_')) {
      var _params$toJSON3;
      // @ts-ignore
      params.nativeHandle = this.nativeHandle;
      const json = (_params$toJSON3 = params.toJSON) === null || _params$toJSON3 === void 0 ? void 0 : _params$toJSON3.call();
      params.toJSON = function () {
        return {
          ...json,
          nativeHandle: params.nativeHandle
        };
      };
    } else if (funcName.startsWith('RtcEngine_')) {
      switch (funcName) {
        case 'RtcEngine_initialize_0320339':
          _specs.default.newIrisApiEngine();
          break;
        case 'RtcEngine_release':
          _specs.default.callApi({
            funcName,
            params: JSON.stringify(params),
            buffers
          });
          _specs.default.destroyIrisApiEngine();
          return;
        case 'RtcEngine_sendMetaData':
          // metadata.buffer
          buffers.push(_base64Js.default.fromByteArray(params.metadata.buffer ?? _buffer.Buffer.from('')));
          break;
        case 'RtcEngine_sendStreamMessage_8715a45':
        case 'RtcEngineEx_sendStreamMessageEx_0c34857':
          // data
          buffers.push(_base64Js.default.fromByteArray(params.data ?? _buffer.Buffer.from('')));
          break;
        case 'RtcEngine_destroyMediaPlayer_328a49b':
          params.mediaPlayerId = params.media_player.getMediaPlayerId();
          params.toJSON = function () {
            return {
              playerId: params.mediaPlayerId
            };
          };
          break;
        case 'RtcEngine_destroyVideoEffectObject_66d092b':
          params.videoEffectObjectId = params.videoEffectObject.getVideoEffectObjectId();
          params.toJSON = function () {
            return {
              objectId: params.videoEffectObjectId
            };
          };
          break;
        case 'RtcEngine_destroyMediaRecorder_95cdef5':
          // @ts-ignore
          params.nativeHandle = params.mediaRecorder.nativeHandle;
          params.toJSON = function () {
            return {
              nativeHandle: params.nativeHandle
            };
          };
          break;
      }
    }
    let ret = _specs.default.callApi({
      funcName,
      params: JSON.stringify(params),
      buffers
    });
    if (ret !== undefined && ret !== null && ret !== '' && ret !== 'null') {
      const retObj = JSON.parse(ret);
      if ((0, _Utils.isDebuggable)()) {
        if (typeof retObj.result === 'number' && retObj.result < 0) {
          console.error('callApi', funcName, JSON.stringify(params), ret);
        } else {
          console.log('callApi', funcName, JSON.stringify(params), ret);
        }
      }
      return retObj;
    }
  } catch (e) {
    if ((0, _Utils.isDebuggable)()) {
      console.error('callApi', funcName, JSON.stringify(params), e);
    } else {
      console.warn('callApi', funcName, JSON.stringify(params), e);
    }
  }
  return {};
}
