// react-native-agora/lib/commonjs/internal/IAgoraRtcRenderView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
exports.getFuncName = getFuncName;
exports.getParams = getParams;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _AgoraMediaBase = require("../AgoraMediaBase");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function getFuncName(canvas, connection) {
  let funcName;
  if (canvas.sourceType === undefined) {
    if (canvas.uid) {
      funcName = 'RtcEngine_setupRemoteVideo_acc9c38';
    } else {
      funcName = 'RtcEngine_setupLocalVideo_acc9c38';
    }
  } else if (canvas.sourceType === _AgoraMediaBase.VideoSourceType.VideoSourceRemote) {
    funcName = 'RtcEngine_setupRemoteVideo_acc9c38';
  } else {
    funcName = 'RtcEngine_setupLocalVideo_acc9c38';
  }
  if (funcName === 'RtcEngine_setupRemoteVideo_acc9c38' && connection) {
    funcName = 'RtcEngineEx_setupRemoteVideoEx_522a409';
  }
  return funcName;
}
function getParams(props) {
  const {
    canvas,
    connection
  } = props;
  let {
    mediaPlayerId
  } = canvas;
  if (canvas.sourceType === _AgoraMediaBase.VideoSourceType.VideoSourceMediaPlayer && canvas.mediaPlayerId === undefined) {
    mediaPlayerId = canvas.uid;
  }
  return {
    funcName: getFuncName(canvas, connection),
    params: JSON.stringify({
      canvas: {
        ...canvas,
        mediaPlayerId
      },
      connection
    })
  };
}
class IAgoraRtcRenderView extends _react.Component {
  render() {
    const {
      ...others
    } = this.props;
    const AgoraRtcRenderer = this.view;
    return /*#__PURE__*/_react.default.createElement(AgoraRtcRenderer, _extends({
      style: styles.renderer,
      callApi: getParams(this.props)
    }, others));
  }
}
exports.default = IAgoraRtcRenderView;
const styles = _reactNative.StyleSheet.create({
  renderer: {
    flex: 1
  }
});
