// react-native-agora/lib/commonjs/internal/MediaRecorderInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MediaRecorderInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _AgoraBase = require("../AgoraBase");
var _IAgoraMediaRecorderImpl = require("../impl/IAgoraMediaRecorderImpl");
var _AgoraMediaBaseTi = _interopRequireDefault(require("../ti/AgoraMediaBase-ti"));
var _event = require("./event");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_AgoraMediaBaseTi.default);
class MediaRecorderInternal extends _IAgoraMediaRecorderImpl.IMediaRecorderImpl {
  static _observers = new Map();
  constructor(nativeHandle) {
    super();
    this._nativeHandle = nativeHandle;
  }
  release() {
    MediaRecorderInternal._observers.delete(this._nativeHandle);
    this.removeAllListeners();
  }
  get nativeHandle() {
    return this._nativeHandle;
  }
  setMediaRecorderObserver(callback) {
    const key = this._nativeHandle;
    if (MediaRecorderInternal._observers.get(key) === callback) {
      return _AgoraBase.ErrorCodeType.ErrOk;
    }
    MediaRecorderInternal._observers.set(key, callback);
    return super.setMediaRecorderObserver(callback);
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IMediaRecor;
    if ((_checkers$IMediaRecor = checkers.IMediaRecorderObserver) !== null && _checkers$IMediaRecor !== void 0 && _checkers$IMediaRecor.strictTest({
      [eventType]: undefined
    })) {
      if (MediaRecorderInternal._observers.get(this._nativeHandle) === undefined) {
        this.setMediaRecorderObserver({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== _event.EVENT_TYPE.IMediaRecorder) {
        return;
      }
      if (data.nativeHandle !== this._nativeHandle) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    (0, _event.addScopedEventListener)(this, eventType, callback);
  }
  removeListener(eventType, listener) {
    (0, _event.removeScopedEventListener)(this, eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    (0, _event.removeAllScopedEventListeners)(this, eventType);
  }
}
exports.MediaRecorderInternal = MediaRecorderInternal;
