// react-native-agora/lib/commonjs/internal/MusicContentCenterInternal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MusicPlayerInternal = exports.MusicContentCenterInternal = exports.MusicCollectionInternal = void 0;
var _tsInterfaceChecker = require("ts-interface-checker");
var _IAgoraMusicContentCenter = require("../IAgoraMusicContentCenter");
var _IAgoraMusicContentCenterImpl = require("../impl/IAgoraMusicContentCenterImpl");
var _IAgoraMusicContentCenterTi = _interopRequireDefault(require("../ti/IAgoraMusicContentCenter-ti"));
var _MediaPlayerInternal = require("./MediaPlayerInternal");
var _event = require("./event");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const checkers = (0, _tsInterfaceChecker.createCheckers)(_IAgoraMusicContentCenterTi.default);
class MusicContentCenterInternal extends _IAgoraMusicContentCenterImpl.IMusicContentCenterImpl {
  static _event_handlers = [];
  _addListenerPreCheck(eventType) {
    var _checkers$IMusicConte;
    if ((_checkers$IMusicConte = checkers.IMusicContentCenterEventHandler) !== null && _checkers$IMusicConte !== void 0 && _checkers$IMusicConte.strictTest({
      [eventType]: undefined
    })) {
      if (MusicContentCenterInternal._event_handlers.length === 0) {
        this.registerEventHandler({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== _event.EVENT_TYPE.IMusicContentCenter) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    (0, _event.addScopedEventListener)(this, eventType, callback);
  }
  removeListener(eventType, listener) {
    (0, _event.removeScopedEventListener)(this, eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    (0, _event.removeAllScopedEventListeners)(this, eventType);
  }
  registerEventHandler(eventHandler) {
    if (!MusicContentCenterInternal._event_handlers.find(value => value === eventHandler)) {
      MusicContentCenterInternal._event_handlers.push(eventHandler);
    }
    return super.registerEventHandler(eventHandler);
  }
  unregisterEventHandler() {
    MusicContentCenterInternal._event_handlers = [];
    return super.unregisterEventHandler();
  }
  release() {
    MusicContentCenterInternal._event_handlers = [];
    this.removeAllListeners();
    super.release();
  }
  createMusicPlayer() {
    // @ts-ignore
    const mediaPlayerId = super.createMusicPlayer();
    return new MusicPlayerInternal(mediaPlayerId);
  }
}
exports.MusicContentCenterInternal = MusicContentCenterInternal;
class _MusicPlayerInternal extends _IAgoraMusicContentCenterImpl.IMusicPlayerImpl {
  constructor(mediaPlayerId) {
    super();
    this._mediaPlayerId = mediaPlayerId;
  }
  getMediaPlayerId() {
    return this._mediaPlayerId;
  }
}
class MusicPlayerInternal extends _MediaPlayerInternal.MediaPlayerInternal {
  constructor(mediaPlayerId) {
    super(mediaPlayerId);
    // @ts-ignore
    this._musicPlayer = new _MusicPlayerInternal(mediaPlayerId);
  }
  setPlayMode(mode) {
    return this._musicPlayer.setPlayMode(mode);
  }
  openWithSongCode(songCode, startPos) {
    return this._musicPlayer.openWithSongCode(songCode, startPos);
  }
}
exports.MusicPlayerInternal = MusicPlayerInternal;
class _MusicCollection extends _IAgoraMusicContentCenter.MusicCollection {
  constructor(collection) {
    super();
    this.count = collection.count;
    this.music = collection.music;
    this.page = collection.page;
    this.pageSize = collection.pageSize;
    this.total = collection.total;
  }
  getCount() {
    return this.count;
  }
  getMusic(index) {
    return this.music[index] ?? {};
  }
  getPage() {
    return this.page;
  }
  getPageSize() {
    return this.pageSize;
  }
  getTotal() {
    return this.total;
  }
}
class MusicCollectionInternal extends _IAgoraMusicContentCenterImpl.MusicCollectionImpl {
  constructor(musicCollection) {
    super();
    this._musicCollection = new _MusicCollection(musicCollection);
  }
  getCount() {
    return this._musicCollection.getCount();
  }
  getMusic(index) {
    return this._musicCollection.getMusic(index);
  }
  getPage() {
    return this._musicCollection.getPage();
  }
  getPageSize() {
    return this._musicCollection.getPageSize();
  }
  getTotal() {
    return this._musicCollection.getTotal();
  }
}
exports.MusicCollectionInternal = MusicCollectionInternal;
