// react-native-agora/lib/module/internal/event.js
import EventEmitter from 'eventemitter3';
// @ts-ignore
export const DeviceEventEmitter = new EventEmitter();
const scopedDeviceEventListeners = new Map();
function getOwnerScopedEventListeners(owner, create = false) {
  let ownerScopedEventListeners = scopedDeviceEventListeners.get(owner);
  if (ownerScopedEventListeners === undefined && create) {
    ownerScopedEventListeners = new Map();
    scopedDeviceEventListeners.set(owner, ownerScopedEventListeners);
  }
  return ownerScopedEventListeners;
}
function getScopedEventListeners(owner, eventType, create = false) {
  const ownerScopedEventListeners = getOwnerScopedEventListeners(owner, create);
  if (ownerScopedEventListeners === undefined) {
    return undefined;
  }
  let scopedEventListeners = ownerScopedEventListeners.get(eventType);
  if (scopedEventListeners === undefined && create) {
    scopedEventListeners = new Set();
    ownerScopedEventListeners.set(eventType, scopedEventListeners);
  }
  return scopedEventListeners;
}
function cleanupScopedEventListeners(owner, eventType) {
  const ownerScopedEventListeners = getOwnerScopedEventListeners(owner);
  if (ownerScopedEventListeners === undefined) {
    return;
  }
  const scopedEventListeners = ownerScopedEventListeners.get(eventType);
  if ((scopedEventListeners === null || scopedEventListeners === void 0 ? void 0 : scopedEventListeners.size) === 0) {
    ownerScopedEventListeners.delete(eventType);
  }
  if (ownerScopedEventListeners.size === 0) {
    scopedDeviceEventListeners.delete(owner);
  }
}
export function addScopedEventListener(owner, eventType, listener) {
  var _getScopedEventListen;
  (_getScopedEventListen = getScopedEventListeners(owner, eventType, true)) === null || _getScopedEventListen === void 0 || _getScopedEventListen.add(listener);
  DeviceEventEmitter.addListener(eventType, listener);
}
export function removeScopedEventListener(owner, eventType, listener) {
  const scopedEventListeners = getScopedEventListeners(owner, eventType);
  if (scopedEventListeners === undefined) {
    return;
  }
  const listenersToRemove = listener !== undefined ? [listener] : Array.from(scopedEventListeners);
  listenersToRemove.forEach(it => {
    DeviceEventEmitter.removeListener(eventType, it);
    scopedEventListeners.delete(it);
  });
  cleanupScopedEventListeners(owner, eventType);
}
export function removeAllScopedEventListeners(owner, eventType) {
  if (eventType !== undefined) {
    removeScopedEventListener(owner, eventType);
    return;
  }
  const ownerScopedEventListeners = getOwnerScopedEventListeners(owner);
  if (ownerScopedEventListeners === undefined) {
    return;
  }
  ownerScopedEventListeners.forEach((scopedEventListeners, registeredEvent) => {
    scopedEventListeners.forEach(listener => {
      DeviceEventEmitter.removeListener(registeredEvent, listener);
    });
  });
  scopedDeviceEventListeners.delete(owner);
}
export function removeAllEventListeners() {
  DeviceEventEmitter.removeAllListeners();
  scopedDeviceEventListeners.clear();
}

/**
 * @internal
 */

export let EVENT_TYPE = /*#__PURE__*/function (EVENT_TYPE) {
  EVENT_TYPE[EVENT_TYPE["IMediaEngine"] = 0] = "IMediaEngine";
  EVENT_TYPE[EVENT_TYPE["IMediaPlayer"] = 1] = "IMediaPlayer";
  EVENT_TYPE[EVENT_TYPE["IMediaRecorder"] = 2] = "IMediaRecorder";
  EVENT_TYPE[EVENT_TYPE["IRtcEngine"] = 3] = "IRtcEngine";
  EVENT_TYPE[EVENT_TYPE["IMusicContentCenter"] = 4] = "IMusicContentCenter";
  EVENT_TYPE[EVENT_TYPE["IAgoraH265Transcoder"] = 5] = "IAgoraH265Transcoder";
  EVENT_TYPE[EVENT_TYPE["IAgoraPip"] = 6] = "IAgoraPip";
  return EVENT_TYPE;
}({});
/**
 * @internal
 */
export function emitEvent(eventType, eventProcessor, data) {
  DeviceEventEmitter.emit(eventType, eventProcessor, data);
}
