// react-native-agora/lib/module/internal/IAgoraRtcRenderView.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { VideoSourceType } from '../AgoraMediaBase';
export function getFuncName(canvas, connection) {
  let funcName;
  if (canvas.sourceType === undefined) {
    if (canvas.uid) {
      funcName = 'RtcEngine_setupRemoteVideo_acc9c38';
    } else {
      funcName = 'RtcEngine_setupLocalVideo_acc9c38';
    }
  } else if (canvas.sourceType === VideoSourceType.VideoSourceRemote) {
    funcName = 'RtcEngine_setupRemoteVideo_acc9c38';
  } else {
    funcName = 'RtcEngine_setupLocalVideo_acc9c38';
  }
  if (funcName === 'RtcEngine_setupRemoteVideo_acc9c38' && connection) {
    funcName = 'RtcEngineEx_setupRemoteVideoEx_522a409';
  }
  return funcName;
}
export function getParams(props) {
  const {
    canvas,
    connection
  } = props;
  let {
    mediaPlayerId
  } = canvas;
  if (canvas.sourceType === VideoSourceType.VideoSourceMediaPlayer && canvas.mediaPlayerId === undefined) {
    mediaPlayerId = canvas.uid;
  }
  return {
    funcName: getFuncName(canvas, connection),
    params: JSON.stringify({
      canvas: {
        ...canvas,
        mediaPlayerId
      },
      connection
    })
  };
}
export default class IAgoraRtcRenderView extends Component {
  render() {
    const {
      ...others
    } = this.props;
    const AgoraRtcRenderer = this.view;
    return /*#__PURE__*/React.createElement(AgoraRtcRenderer, _extends({
      style: styles.renderer,
      callApi: getParams(this.props)
    }, others));
  }
}
const styles = StyleSheet.create({
  renderer: {
    flex: 1
  }
});
