// react-native-agora/lib/module/AgoraMediaBase.js
import './extension/AgoraMediaBaseExtension';
/**
 * Plugin context information.
 */
export class ExtensionContext {}

/**
 * Type of video source.
 */
export let VideoSourceType = /*#__PURE__*/function (VideoSourceType) {
  /**
   * 0: (Default) The video source is the first camera.
   */
  VideoSourceType[VideoSourceType["VideoSourceCameraPrimary"] = 0] = "VideoSourceCameraPrimary";
  /**
   * 0: (Default) The video source is the first camera.
   */
  VideoSourceType[VideoSourceType["VideoSourceCamera"] = 0] = "VideoSourceCamera";
  /**
   * 1: The video source is the second camera.
   */
  VideoSourceType[VideoSourceType["VideoSourceCameraSecondary"] = 1] = "VideoSourceCameraSecondary";
  /**
   * 2: The video source is the first screen.
   */
  VideoSourceType[VideoSourceType["VideoSourceScreenPrimary"] = 2] = "VideoSourceScreenPrimary";
  /**
   * 2: The video source is the first screen.
   */
  VideoSourceType[VideoSourceType["VideoSourceScreen"] = 2] = "VideoSourceScreen";
  /**
   * 3: The video source is the second screen.
   */
  VideoSourceType[VideoSourceType["VideoSourceScreenSecondary"] = 3] = "VideoSourceScreenSecondary";
  /**
   * 4: A custom video source.
   */
  VideoSourceType[VideoSourceType["VideoSourceCustom"] = 4] = "VideoSourceCustom";
  /**
   * 5: The video source is a media player.
   */
  VideoSourceType[VideoSourceType["VideoSourceMediaPlayer"] = 5] = "VideoSourceMediaPlayer";
  /**
   * 6: The video source is a PNG image.
   */
  VideoSourceType[VideoSourceType["VideoSourceRtcImagePng"] = 6] = "VideoSourceRtcImagePng";
  /**
   * 7: The video source is a JPEG image.
   */
  VideoSourceType[VideoSourceType["VideoSourceRtcImageJpeg"] = 7] = "VideoSourceRtcImageJpeg";
  /**
   * 8: The video source is a GIF image.
   */
  VideoSourceType[VideoSourceType["VideoSourceRtcImageGif"] = 8] = "VideoSourceRtcImageGif";
  /**
   * 9: The video source is a remote video fetched from the network.
   */
  VideoSourceType[VideoSourceType["VideoSourceRemote"] = 9] = "VideoSourceRemote";
  /**
   * 10: A transcoded video source.
   */
  VideoSourceType[VideoSourceType["VideoSourceTranscoded"] = 10] = "VideoSourceTranscoded";
  /**
   * 11: (Android only) The video source is the third camera.
   */
  VideoSourceType[VideoSourceType["VideoSourceCameraThird"] = 11] = "VideoSourceCameraThird";
  /**
   * 12: (Android only) The video source is the fourth camera.
   */
  VideoSourceType[VideoSourceType["VideoSourceCameraFourth"] = 12] = "VideoSourceCameraFourth";
  /**
   * @ignore
   */
  VideoSourceType[VideoSourceType["VideoSourceScreenThird"] = 13] = "VideoSourceScreenThird";
  /**
   * @ignore
   */
  VideoSourceType[VideoSourceType["VideoSourceScreenFourth"] = 14] = "VideoSourceScreenFourth";
  /**
   * 15: The video source is video processed by a speech-driven plugin.
   */
  VideoSourceType[VideoSourceType["VideoSourceSpeechDriven"] = 15] = "VideoSourceSpeechDriven";
  /**
   * 100: Unknown video source.
   */
  VideoSourceType[VideoSourceType["VideoSourceUnknown"] = 100] = "VideoSourceUnknown";
  return VideoSourceType;
}({});

/**
 * Audio source type.
 */
export let AudioSourceType = /*#__PURE__*/function (AudioSourceType) {
  /**
   * 0: (Default) Microphone.
   */
  AudioSourceType[AudioSourceType["AudioSourceMicrophone"] = 0] = "AudioSourceMicrophone";
  /**
   * 1: Custom captured audio stream.
   */
  AudioSourceType[AudioSourceType["AudioSourceCustom"] = 1] = "AudioSourceCustom";
  /**
   * 2: Media player.
   */
  AudioSourceType[AudioSourceType["AudioSourceMediaPlayer"] = 2] = "AudioSourceMediaPlayer";
  /**
   * 3: System audio stream captured during screen sharing.
   */
  AudioSourceType[AudioSourceType["AudioSourceLoopbackRecording"] = 3] = "AudioSourceLoopbackRecording";
  /**
   * @ignore
   */
  AudioSourceType[AudioSourceType["AudioSourceMixedStream"] = 4] = "AudioSourceMixedStream";
  /**
   * 5: Audio stream from a specified remote user.
   */
  AudioSourceType[AudioSourceType["AudioSourceRemoteUser"] = 5] = "AudioSourceRemoteUser";
  /**
   * 6: Mixed audio stream from all users in the current channel.
   */
  AudioSourceType[AudioSourceType["AudioSourceRemoteChannel"] = 6] = "AudioSourceRemoteChannel";
  /**
   * 100: Unknown audio source.
   */
  AudioSourceType[AudioSourceType["AudioSourceUnknown"] = 100] = "AudioSourceUnknown";
  return AudioSourceType;
}({});

/**
 * Type of audio route.
 */
export let AudioRoute = /*#__PURE__*/function (AudioRoute) {
  /**
   * -1: Use the default audio route.
   */
  AudioRoute[AudioRoute["RouteDefault"] = -1] = "RouteDefault";
  /**
   * 0: Audio route is a headset with microphone.
   */
  AudioRoute[AudioRoute["RouteHeadset"] = 0] = "RouteHeadset";
  /**
   * 1: Audio route is the earpiece.
   */
  AudioRoute[AudioRoute["RouteEarpiece"] = 1] = "RouteEarpiece";
  /**
   * 2: Audio route is a headset without microphone.
   */
  AudioRoute[AudioRoute["RouteHeadsetnomic"] = 2] = "RouteHeadsetnomic";
  /**
   * 3: Audio route is the built-in speaker of the device.
   */
  AudioRoute[AudioRoute["RouteSpeakerphone"] = 3] = "RouteSpeakerphone";
  /**
   * 4: Audio route is an external speaker. (iOS only)
   */
  AudioRoute[AudioRoute["RouteLoudspeaker"] = 4] = "RouteLoudspeaker";
  /**
   * 5: Audio route is a Bluetooth device using the HFP protocol.
   */
  AudioRoute[AudioRoute["RouteBluetoothDeviceHfp"] = 5] = "RouteBluetoothDeviceHfp";
  /**
   * @ignore
   */
  AudioRoute[AudioRoute["RouteUsb"] = 6] = "RouteUsb";
  /**
   * @ignore
   */
  AudioRoute[AudioRoute["RouteHdmi"] = 7] = "RouteHdmi";
  /**
   * @ignore
   */
  AudioRoute[AudioRoute["RouteDisplayport"] = 8] = "RouteDisplayport";
  /**
   * @ignore
   */
  AudioRoute[AudioRoute["RouteAirplay"] = 9] = "RouteAirplay";
  /**
   * 10: Audio route is a Bluetooth device using the A2DP protocol.
   */
  AudioRoute[AudioRoute["RouteBluetoothDeviceA2dp"] = 10] = "RouteBluetoothDeviceA2dp";
  return AudioRoute;
}({});

/**
 * @ignore
 */
export let BytesPerSample = /*#__PURE__*/function (BytesPerSample) {
  /**
   * @ignore
   */
  BytesPerSample[BytesPerSample["TwoBytesPerSample"] = 2] = "TwoBytesPerSample";
  return BytesPerSample;
}({});

/**
 * @ignore
 */
export class AudioParameters {}

/**
 * Usage modes for audio data.
 */
export let RawAudioFrameOpModeType = /*#__PURE__*/function (RawAudioFrameOpModeType) {
  /**
   * 0: (Default) Read-only mode. For example, if you collect data using the SDK and perform CDN streaming yourself, you can choose this mode.
   */
  RawAudioFrameOpModeType[RawAudioFrameOpModeType["RawAudioFrameOpModeReadOnly"] = 0] = "RawAudioFrameOpModeReadOnly";
  /**
   * 2: Read-write mode. For example, if you have your own audio effects processing module and want to pre-process the data as needed (such as voice changing), you can choose this mode.
   */
  RawAudioFrameOpModeType[RawAudioFrameOpModeType["RawAudioFrameOpModeReadWrite"] = 2] = "RawAudioFrameOpModeReadWrite";
  return RawAudioFrameOpModeType;
}({});

/**
 * The type of media source.
 */
export let MediaSourceType = /*#__PURE__*/function (MediaSourceType) {
  /**
   * 0: Audio playback device.
   */
  MediaSourceType[MediaSourceType["AudioPlayoutSource"] = 0] = "AudioPlayoutSource";
  /**
   * 1: Audio recording device.
   */
  MediaSourceType[MediaSourceType["AudioRecordingSource"] = 1] = "AudioRecordingSource";
  /**
   * 2: Primary camera.
   */
  MediaSourceType[MediaSourceType["PrimaryCameraSource"] = 2] = "PrimaryCameraSource";
  /**
   * 3: Secondary camera.
   */
  MediaSourceType[MediaSourceType["SecondaryCameraSource"] = 3] = "SecondaryCameraSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["PrimaryScreenSource"] = 4] = "PrimaryScreenSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["SecondaryScreenSource"] = 5] = "SecondaryScreenSource";
  /**
   * 6: Custom video capture source.
   */
  MediaSourceType[MediaSourceType["CustomVideoSource"] = 6] = "CustomVideoSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["MediaPlayerSource"] = 7] = "MediaPlayerSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["RtcImagePngSource"] = 8] = "RtcImagePngSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["RtcImageJpegSource"] = 9] = "RtcImageJpegSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["RtcImageGifSource"] = 10] = "RtcImageGifSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["RemoteVideoSource"] = 11] = "RemoteVideoSource";
  /**
   * @ignore
   */
  MediaSourceType[MediaSourceType["TranscodedVideoSource"] = 12] = "TranscodedVideoSource";
  /**
   * 13: Video source processed by speech-driven plugin.
   */
  MediaSourceType[MediaSourceType["SpeechDrivenVideoSource"] = 13] = "SpeechDrivenVideoSource";
  /**
   * 100: Unknown media source.
   */
  MediaSourceType[MediaSourceType["UnknownMediaSource"] = 100] = "UnknownMediaSource";
  return MediaSourceType;
}({});

/**
 * @ignore
 */
export class PacketOptions {}

/**
 * @ignore
 */
export class AudioEncodedFrameInfo {}

/**
 * Information of external PCM format audio frame.
 */
export class AudioPcmFrame {}

/**
 * Channel mode.
 */
export let AudioDualMonoMode = /*#__PURE__*/function (AudioDualMonoMode) {
  /**
   * 0: Original mode.
   */
  AudioDualMonoMode[AudioDualMonoMode["AudioDualMonoStereo"] = 0] = "AudioDualMonoStereo";
  /**
   * 1: Left channel mode. This mode replaces the right channel audio with the left channel audio, so the user hears only the left channel.
   */
  AudioDualMonoMode[AudioDualMonoMode["AudioDualMonoL"] = 1] = "AudioDualMonoL";
  /**
   * 2: Right channel mode. This mode replaces the left channel audio with the right channel audio, so the user hears only the right channel.
   */
  AudioDualMonoMode[AudioDualMonoMode["AudioDualMonoR"] = 2] = "AudioDualMonoR";
  /**
   * 3: Mixed mode. This mode mixes the left and right channels, so the user hears both channels simultaneously.
   */
  AudioDualMonoMode[AudioDualMonoMode["AudioDualMonoMix"] = 3] = "AudioDualMonoMix";
  return AudioDualMonoMode;
}({});

/**
 * Video pixel format.
 */
export let VideoPixelFormat = /*#__PURE__*/function (VideoPixelFormat) {
  /**
   * 0: Original video pixel format.
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelDefault"] = 0] = "VideoPixelDefault";
  /**
   * 1: I420 format.
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelI420"] = 1] = "VideoPixelI420";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelBgra"] = 2] = "VideoPixelBgra";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelNv21"] = 3] = "VideoPixelNv21";
  /**
   * 4: RGBA format.
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelRgba"] = 4] = "VideoPixelRgba";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelNv12"] = 8] = "VideoPixelNv12";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoTexture2d"] = 10] = "VideoTexture2d";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoTextureOes"] = 11] = "VideoTextureOes";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoCvpixelNv12"] = 12] = "VideoCvpixelNv12";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoCvpixelI420"] = 13] = "VideoCvpixelI420";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoCvpixelBgra"] = 14] = "VideoCvpixelBgra";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoCvpixelP010"] = 15] = "VideoCvpixelP010";
  /**
   * 16: I422 format.
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelI422"] = 16] = "VideoPixelI422";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoTextureId3d11texture2d"] = 17] = "VideoTextureId3d11texture2d";
  /**
   * @ignore
   */
  VideoPixelFormat[VideoPixelFormat["VideoPixelI010"] = 18] = "VideoPixelI010";
  return VideoPixelFormat;
}({});

/**
 * Video display mode.
 */
export let RenderModeType = /*#__PURE__*/function (RenderModeType) {
  /**
   * 1: The video is scaled proportionally. Priority is given to filling the view. Any excess part of the video that does not fit due to aspect ratio differences will be cropped.
   */
  RenderModeType[RenderModeType["RenderModeHidden"] = 1] = "RenderModeHidden";
  /**
   * 2: The video is scaled proportionally. Priority is given to displaying the entire video content. Any area not filled due to aspect ratio differences will be filled with black.
   */
  RenderModeType[RenderModeType["RenderModeFit"] = 2] = "RenderModeFit";
  /**
   * @ignore
   */
  RenderModeType[RenderModeType["RenderModeAdaptive"] = 3] = "RenderModeAdaptive";
  return RenderModeType;
}({});

/**
 * @ignore
 */
export let CameraVideoSourceType = /*#__PURE__*/function (CameraVideoSourceType) {
  /**
   * @ignore
   */
  CameraVideoSourceType[CameraVideoSourceType["CameraSourceFront"] = 0] = "CameraSourceFront";
  /**
   * @ignore
   */
  CameraVideoSourceType[CameraVideoSourceType["CameraSourceBack"] = 1] = "CameraSourceBack";
  /**
   * @ignore
   */
  CameraVideoSourceType[CameraVideoSourceType["VideoSourceUnspecified"] = 2] = "VideoSourceUnspecified";
  return CameraVideoSourceType;
}({});

/**
 * @ignore
 */
export let MetaInfoKey = /*#__PURE__*/function (MetaInfoKey) {
  /**
   * @ignore
   */
  MetaInfoKey[MetaInfoKey["KeyFaceCapture"] = 0] = "KeyFaceCapture";
  return MetaInfoKey;
}({});

/**
 * @ignore
 */
export class IVideoFrameMetaInfo {}

/**
 * @ignore
 */
export let PrimaryID = /*#__PURE__*/function (PrimaryID) {
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidBt709"] = 1] = "PrimaryidBt709";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidUnspecified"] = 2] = "PrimaryidUnspecified";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidBt470m"] = 4] = "PrimaryidBt470m";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidBt470bg"] = 5] = "PrimaryidBt470bg";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidSmpte170m"] = 6] = "PrimaryidSmpte170m";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidSmpte240m"] = 7] = "PrimaryidSmpte240m";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidFilm"] = 8] = "PrimaryidFilm";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidBt2020"] = 9] = "PrimaryidBt2020";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidSmptest428"] = 10] = "PrimaryidSmptest428";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidSmptest431"] = 11] = "PrimaryidSmptest431";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidSmptest432"] = 12] = "PrimaryidSmptest432";
  /**
   * @ignore
   */
  PrimaryID[PrimaryID["PrimaryidJedecp22"] = 22] = "PrimaryidJedecp22";
  return PrimaryID;
}({});

/**
 * @ignore
 */
export let RangeID = /*#__PURE__*/function (RangeID) {
  /**
   * @ignore
   */
  RangeID[RangeID["RangeidInvalid"] = 0] = "RangeidInvalid";
  /**
   * @ignore
   */
  RangeID[RangeID["RangeidLimited"] = 1] = "RangeidLimited";
  /**
   * @ignore
   */
  RangeID[RangeID["RangeidFull"] = 2] = "RangeidFull";
  /**
   * @ignore
   */
  RangeID[RangeID["RangeidDerived"] = 3] = "RangeidDerived";
  return RangeID;
}({});

/**
 * @ignore
 */
export let MatrixID = /*#__PURE__*/function (MatrixID) {
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidRgb"] = 0] = "MatrixidRgb";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidBt709"] = 1] = "MatrixidBt709";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidUnspecified"] = 2] = "MatrixidUnspecified";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidFcc"] = 4] = "MatrixidFcc";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidBt470bg"] = 5] = "MatrixidBt470bg";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidSmpte170m"] = 6] = "MatrixidSmpte170m";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidSmpte240m"] = 7] = "MatrixidSmpte240m";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidYcocg"] = 8] = "MatrixidYcocg";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidBt2020Ncl"] = 9] = "MatrixidBt2020Ncl";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidBt2020Cl"] = 10] = "MatrixidBt2020Cl";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidSmpte2085"] = 11] = "MatrixidSmpte2085";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidCdncls"] = 12] = "MatrixidCdncls";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidCdcls"] = 13] = "MatrixidCdcls";
  /**
   * @ignore
   */
  MatrixID[MatrixID["MatrixidBt2100Ictcp"] = 14] = "MatrixidBt2100Ictcp";
  return MatrixID;
}({});

/**
 * @ignore
 */
export let TransferID = /*#__PURE__*/function (TransferID) {
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidBt709"] = 1] = "TransferidBt709";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidUnspecified"] = 2] = "TransferidUnspecified";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidGamma22"] = 4] = "TransferidGamma22";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidGamma28"] = 5] = "TransferidGamma28";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidSmpte170m"] = 6] = "TransferidSmpte170m";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidSmpte240m"] = 7] = "TransferidSmpte240m";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidLinear"] = 8] = "TransferidLinear";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidLog"] = 9] = "TransferidLog";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidLogSqrt"] = 10] = "TransferidLogSqrt";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidIec6196624"] = 11] = "TransferidIec6196624";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidBt1361Ecg"] = 12] = "TransferidBt1361Ecg";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidIec6196621"] = 13] = "TransferidIec6196621";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidBt202010"] = 14] = "TransferidBt202010";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidBt202012"] = 15] = "TransferidBt202012";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidSmptest2084"] = 16] = "TransferidSmptest2084";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidSmptest428"] = 17] = "TransferidSmptest428";
  /**
   * @ignore
   */
  TransferID[TransferID["TransferidAribStdB67"] = 18] = "TransferidAribStdB67";
  return TransferID;
}({});

/**
 * @ignore
 */
export class ColorSpace {}

/**
 * @ignore
 */
export class Hdr10MetadataInfo {}

/**
 * The relative position of alphaBuffer and the video frame.
 */
export let AlphaStitchMode = /*#__PURE__*/function (AlphaStitchMode) {
  /**
   * 0: (Default) Video frame only, i.e., alphaBuffer is not stitched with the video frame.
   */
  AlphaStitchMode[AlphaStitchMode["NoAlphaStitch"] = 0] = "NoAlphaStitch";
  /**
   * 1: alphaBuffer is above the video frame.
   */
  AlphaStitchMode[AlphaStitchMode["AlphaStitchUp"] = 1] = "AlphaStitchUp";
  /**
   * 2: alphaBuffer is below the video frame.
   */
  AlphaStitchMode[AlphaStitchMode["AlphaStitchBelow"] = 2] = "AlphaStitchBelow";
  /**
   * 3: alphaBuffer is to the left of the video frame.
   */
  AlphaStitchMode[AlphaStitchMode["AlphaStitchLeft"] = 3] = "AlphaStitchLeft";
  /**
   * 4: alphaBuffer is to the right of the video frame.
   */
  AlphaStitchMode[AlphaStitchMode["AlphaStitchRight"] = 4] = "AlphaStitchRight";
  return AlphaStitchMode;
}({});

/**
 * @ignore
 */
export let EglContextType = /*#__PURE__*/function (EglContextType) {
  /**
   * @ignore
   */
  EglContextType[EglContextType["EglContext10"] = 0] = "EglContext10";
  /**
   * @ignore
   */
  EglContextType[EglContextType["EglContext14"] = 1] = "EglContext14";
  return EglContextType;
}({});

/**
 * Video buffer type.
 */
export let VideoBufferType = /*#__PURE__*/function (VideoBufferType) {
  /**
   * 1: Type is raw data.
   */
  VideoBufferType[VideoBufferType["VideoBufferRawData"] = 1] = "VideoBufferRawData";
  /**
   * 2: Type is raw data.
   */
  VideoBufferType[VideoBufferType["VideoBufferArray"] = 2] = "VideoBufferArray";
  /**
   * 3: Type is Texture.
   */
  VideoBufferType[VideoBufferType["VideoBufferTexture"] = 3] = "VideoBufferTexture";
  return VideoBufferType;
}({});

/**
 * External video frame.
 */
export class ExternalVideoFrame {}

/**
 * Properties of a video frame.
 *
 * The buffer is a pointer to a pointer. This interface cannot modify the pointer of the buffer, only its contents.
 */
export class VideoFrame {}

/**
 * @ignore
 */
export let MediaPlayerSourceType = /*#__PURE__*/function (MediaPlayerSourceType) {
  /**
   * @ignore
   */
  MediaPlayerSourceType[MediaPlayerSourceType["MediaPlayerSourceDefault"] = 0] = "MediaPlayerSourceDefault";
  /**
   * @ignore
   */
  MediaPlayerSourceType[MediaPlayerSourceType["MediaPlayerSourceFullFeatured"] = 1] = "MediaPlayerSourceFullFeatured";
  /**
   * @ignore
   */
  MediaPlayerSourceType[MediaPlayerSourceType["MediaPlayerSourceSimple"] = 2] = "MediaPlayerSourceSimple";
  return MediaPlayerSourceType;
}({});

/**
 * Video observation position.
 */
export let VideoModulePosition = /*#__PURE__*/function (VideoModulePosition) {
  /**
   * 1: The position after local video is captured and pre-processed, corresponding to the onCaptureVideoFrame callback. The video observed here includes pre-processing effects, which can be verified by enabling beauty effects, virtual background, or watermark.
   */
  VideoModulePosition[VideoModulePosition["PositionPostCapturer"] = 1] = "PositionPostCapturer";
  /**
   * 2: The position before rendering the received remote video, corresponding to the onRenderVideoFrame callback.
   */
  VideoModulePosition[VideoModulePosition["PositionPreRenderer"] = 2] = "PositionPreRenderer";
  /**
   * 4: The position before local video encoding, corresponding to the onPreEncodeVideoFrame callback. The video observed here includes both pre-processing and pre-encoding processing effects:
   *  For pre-processing effects, you can verify by enabling beauty effects, virtual background, or watermark.
   *  For pre-encoding effects, you can verify by setting a lower frame rate (e.g., 5 fps).
   */
  VideoModulePosition[VideoModulePosition["PositionPreEncoder"] = 4] = "PositionPreEncoder";
  /**
   * 8: The position after local video is captured but before pre-processing. The video observed here does not include pre-processing effects and can be verified by enabling beauty effects, virtual background, or setting a watermark.
   */
  VideoModulePosition[VideoModulePosition["PositionPostCapturerOrigin"] = 8] = "PositionPostCapturerOrigin";
  return VideoModulePosition;
}({});

/**
 * @ignore
 */
export let ContentInspectResult = /*#__PURE__*/function (ContentInspectResult) {
  /**
   * @ignore
   */
  ContentInspectResult[ContentInspectResult["ContentInspectNeutral"] = 1] = "ContentInspectNeutral";
  /**
   * @ignore
   */
  ContentInspectResult[ContentInspectResult["ContentInspectSexy"] = 2] = "ContentInspectSexy";
  /**
   * @ignore
   */
  ContentInspectResult[ContentInspectResult["ContentInspectPorn"] = 3] = "ContentInspectPorn";
  return ContentInspectResult;
}({});

/**
 * Type of video content inspection module.
 */
export let ContentInspectType = /*#__PURE__*/function (ContentInspectType) {
  /**
   * 0: (Default) This module has no actual functionality. Do not set type to this value.
   */
  ContentInspectType[ContentInspectType["ContentInspectInvalid"] = 0] = "ContentInspectInvalid";
  /**
   * @ignore
   */
  ContentInspectType[ContentInspectType["ContentInspectModeration"] = 1] = "ContentInspectModeration";
  /**
   * 2: Use Agora self-developed plugin for screenshot upload. The SDK takes screenshots of the video stream and uploads them.
   */
  ContentInspectType[ContentInspectType["ContentInspectSupervision"] = 2] = "ContentInspectSupervision";
  /**
   * 3: Use cloud marketplace plugin for screenshot upload. The SDK uses the cloud marketplace video moderation plugin to take screenshots of the video stream and upload them.
   */
  ContentInspectType[ContentInspectType["ContentInspectImageModeration"] = 3] = "ContentInspectImageModeration";
  return ContentInspectType;
}({});

/**
 * ContentInspectModule struct used to configure the frequency of local screenshot uploads.
 */
export class ContentInspectModule {}

/**
 * Local screenshot upload configuration.
 */
export class ContentInspectConfig {}

/**
 * Video snapshot settings.
 */
export class SnapshotConfig {}

/**
 * This class is used to obtain raw PCM audio data.
 *
 * You can inherit this class and implement the onFrame callback to get PCM audio data.
 */

/**
 * Audio frame type.
 */
export let AudioFrameType = /*#__PURE__*/function (AudioFrameType) {
  /**
   * 0: PCM 16
   */
  AudioFrameType[AudioFrameType["FrameTypePcm16"] = 0] = "FrameTypePcm16";
  return AudioFrameType;
}({});

/**
 * Raw audio data.
 */
export class AudioFrame {}

/**
 * @ignore
 */
export let AudioFramePosition = /*#__PURE__*/function (AudioFramePosition) {
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionNone"] = 0] = "AudioFramePositionNone";
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionPlayback"] = 1] = "AudioFramePositionPlayback";
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionRecord"] = 2] = "AudioFramePositionRecord";
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionMixed"] = 4] = "AudioFramePositionMixed";
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionBeforeMixing"] = 8] = "AudioFramePositionBeforeMixing";
  /**
   * @ignore
   */
  AudioFramePosition[AudioFramePosition["AudioFramePositionEarMonitoring"] = 16] = "AudioFramePositionEarMonitoring";
  return AudioFramePosition;
}({});

/**
 * Audio data format.
 *
 * The SDK sets the audio data format in the following callbacks based on AudioParams : onRecordAudioFrame onPlaybackAudioFrame onMixedAudioFrame
 *  The SDK calculates the sampling interval using the samplesPerCall, sampleRate, and channel parameters in AudioParams, and triggers the onRecordAudioFrame, onPlaybackAudioFrame, onMixedAudioFrame, and onEarMonitoringAudioFrame callbacks accordingly.
 *  Sampling interval = samplesPerCall / (sampleRate × channel).
 *  Ensure the sampling interval is not less than 0.01 (s).
 */
export class AudioParams {}

/**
 * Audio observer.
 *
 * You can call registerAudioFrameObserver to register or unregister the IAudioFrameObserverBase audio observer.
 */

/**
 * Audio observer.
 *
 * You can call registerAudioFrameObserver to register or unregister the IAudioFrameObserver audio observer.
 */

/**
 * Audio spectrum data.
 */
export class AudioSpectrumData {}

/**
 * Audio spectrum information of a remote user.
 */
export class UserAudioSpectrumInfo {}

/**
 * Audio spectrum observer.
 */

/**
 * Class for receiving encoded video frames.
 */

/**
 * Video frame processing mode.
 */
export let VideoFrameProcessMode = /*#__PURE__*/function (VideoFrameProcessMode) {
  /**
   * Read-only mode.
   * In read-only mode, you do not modify the video frame, and the video observer acts as a renderer.
   */
  VideoFrameProcessMode[VideoFrameProcessMode["ProcessModeReadOnly"] = 0] = "ProcessModeReadOnly";
  /**
   * Read-write mode.
   * In read-write mode, you modify the video frame, and the video observer acts as a video filter.
   */
  VideoFrameProcessMode[VideoFrameProcessMode["ProcessModeReadWrite"] = 1] = "ProcessModeReadWrite";
  return VideoFrameProcessMode;
}({});

/**
 * Video observer.
 *
 * You can call registerVideoFrameObserver to register or unregister the IVideoFrameObserver video observer.
 */

/**
 * Encoding type of external video frames.
 */
export let ExternalVideoSourceType = /*#__PURE__*/function (ExternalVideoSourceType) {
  /**
   * 0: Unencoded video frame.
   */
  ExternalVideoSourceType[ExternalVideoSourceType["VideoFrame"] = 0] = "VideoFrame";
  /**
   * 1: Encoded video frame.
   */
  ExternalVideoSourceType[ExternalVideoSourceType["EncodedVideoFrame"] = 1] = "EncodedVideoFrame";
  return ExternalVideoSourceType;
}({});

/**
 * @ignore
 */
export let MediaRecorderContainerFormat = /*#__PURE__*/function (MediaRecorderContainerFormat) {
  /**
   * @ignore
   */
  MediaRecorderContainerFormat[MediaRecorderContainerFormat["FormatMp4"] = 1] = "FormatMp4";
  return MediaRecorderContainerFormat;
}({});

/**
 * @ignore
 */
export let MediaRecorderStreamType = /*#__PURE__*/function (MediaRecorderStreamType) {
  /**
   * @ignore
   */
  MediaRecorderStreamType[MediaRecorderStreamType["StreamTypeAudio"] = 1] = "StreamTypeAudio";
  /**
   * @ignore
   */
  MediaRecorderStreamType[MediaRecorderStreamType["StreamTypeVideo"] = 2] = "StreamTypeVideo";
  /**
   * @ignore
   */
  MediaRecorderStreamType[MediaRecorderStreamType["StreamTypeBoth"] = 3] = "StreamTypeBoth";
  return MediaRecorderStreamType;
}({});

/**
 * Current recording state.
 */
export let RecorderState = /*#__PURE__*/function (RecorderState) {
  /**
   * -1: Audio/video stream recording error. See RecorderReasonCode.
   */
  RecorderState[RecorderState["RecorderStateError"] = -1] = "RecorderStateError";
  /**
   * 2: Audio/video stream recording started.
   */
  RecorderState[RecorderState["RecorderStateStart"] = 2] = "RecorderStateStart";
  /**
   * 3: Audio/video stream recording stopped.
   */
  RecorderState[RecorderState["RecorderStateStop"] = 3] = "RecorderStateStop";
  return RecorderState;
}({});

/**
 * Reasons for recording state errors.
 */
export let RecorderReasonCode = /*#__PURE__*/function (RecorderReasonCode) {
  /**
   * 0: Everything is normal.
   */
  RecorderReasonCode[RecorderReasonCode["RecorderReasonNone"] = 0] = "RecorderReasonNone";
  /**
   * @ignore
   */
  RecorderReasonCode[RecorderReasonCode["RecorderReasonWriteFailed"] = 1] = "RecorderReasonWriteFailed";
  /**
   * @ignore
   */
  RecorderReasonCode[RecorderReasonCode["RecorderReasonNoStream"] = 2] = "RecorderReasonNoStream";
  /**
   * @ignore
   */
  RecorderReasonCode[RecorderReasonCode["RecorderReasonOverMaxDuration"] = 3] = "RecorderReasonOverMaxDuration";
  /**
   * @ignore
   */
  RecorderReasonCode[RecorderReasonCode["RecorderReasonConfigChanged"] = 4] = "RecorderReasonConfigChanged";
  return RecorderReasonCode;
}({});

/**
 * @ignore
 */
export class MediaRecorderConfiguration {}

/**
 * Face information observer.
 *
 * You can call registerFaceInfoObserver to register the IFaceInfoObserver observer.
 */

/**
 * @ignore
 */
export class RecorderInfo {}

/**
 * @ignore
 */
