// react-native-agora/lib/module/AgoraBase.js
import './extension/AgoraBaseExtension';
/**
 * Channel profile.
 */
export let ChannelProfileType = /*#__PURE__*/function (ChannelProfileType) {
  /**
   * 0: Communication profile. We recommend using the live broadcasting profile for better audio and video experience.
   */
  ChannelProfileType[ChannelProfileType["ChannelProfileCommunication"] = 0] = "ChannelProfileCommunication";
  /**
   * 1: (Default) Live broadcasting profile.
   */
  ChannelProfileType[ChannelProfileType["ChannelProfileLiveBroadcasting"] = 1] = "ChannelProfileLiveBroadcasting";
  /**
   * 2: Gaming profile. Deprecated: Use ChannelProfileLiveBroadcasting instead.
   */
  ChannelProfileType[ChannelProfileType["ChannelProfileGame"] = 2] = "ChannelProfileGame";
  /**
   * 3: Interactive profile. This profile is optimized for low latency. If your scenario involves frequent interactions, we recommend using this profile. Deprecated: Use ChannelProfileLiveBroadcasting instead.
   */
  ChannelProfileType[ChannelProfileType["ChannelProfileCloudGaming"] = 3] = "ChannelProfileCloudGaming";
  /**
   * @ignore
   */
  ChannelProfileType[ChannelProfileType["ChannelProfileCommunication1v1"] = 4] = "ChannelProfileCommunication1v1";
  return ChannelProfileType;
}({});

/**
 * @ignore
 */
export let WarnCodeType = /*#__PURE__*/function (WarnCodeType) {
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnInvalidView"] = 8] = "WarnInvalidView";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnInitVideo"] = 16] = "WarnInitVideo";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnPending"] = 20] = "WarnPending";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnNoAvailableChannel"] = 103] = "WarnNoAvailableChannel";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnLookupChannelTimeout"] = 104] = "WarnLookupChannelTimeout";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnLookupChannelRejected"] = 105] = "WarnLookupChannelRejected";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnOpenChannelTimeout"] = 106] = "WarnOpenChannelTimeout";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnOpenChannelRejected"] = 107] = "WarnOpenChannelRejected";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnSwitchLiveVideoTimeout"] = 111] = "WarnSwitchLiveVideoTimeout";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnSetClientRoleTimeout"] = 118] = "WarnSetClientRoleTimeout";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnOpenChannelInvalidTicket"] = 121] = "WarnOpenChannelInvalidTicket";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnOpenChannelTryNextVos"] = 122] = "WarnOpenChannelTryNextVos";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnChannelConnectionUnrecoverable"] = 131] = "WarnChannelConnectionUnrecoverable";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnChannelConnectionIpChanged"] = 132] = "WarnChannelConnectionIpChanged";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnChannelConnectionPortChanged"] = 133] = "WarnChannelConnectionPortChanged";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnChannelSocketError"] = 134] = "WarnChannelSocketError";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAudioMixingOpenError"] = 701] = "WarnAudioMixingOpenError";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRuntimePlayoutWarning"] = 1014] = "WarnAdmRuntimePlayoutWarning";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRuntimeRecordingWarning"] = 1016] = "WarnAdmRuntimeRecordingWarning";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRecordAudioSilence"] = 1019] = "WarnAdmRecordAudioSilence";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmPlayoutMalfunction"] = 1020] = "WarnAdmPlayoutMalfunction";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRecordMalfunction"] = 1021] = "WarnAdmRecordMalfunction";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRecordAudioLowlevel"] = 1031] = "WarnAdmRecordAudioLowlevel";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmPlayoutAudioLowlevel"] = 1032] = "WarnAdmPlayoutAudioLowlevel";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmRecordIsOccupied"] = 1033] = "WarnAdmRecordIsOccupied";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmWindowsNoDataReadyEvent"] = 1040] = "WarnAdmWindowsNoDataReadyEvent";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnApmHowling"] = 1051] = "WarnApmHowling";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmGlitchState"] = 1052] = "WarnAdmGlitchState";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmImproperSettings"] = 1053] = "WarnAdmImproperSettings";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmPopState"] = 1055] = "WarnAdmPopState";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmWinCoreNoRecordingDevice"] = 1322] = "WarnAdmWinCoreNoRecordingDevice";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmWinCoreNoPlayoutDevice"] = 1323] = "WarnAdmWinCoreNoPlayoutDevice";
  /**
   * @ignore
   */
  WarnCodeType[WarnCodeType["WarnAdmWinCoreImproperCaptureRelease"] = 1324] = "WarnAdmWinCoreImproperCaptureRelease";
  return WarnCodeType;
}({});

/**
 * Error codes.
 *
 * Error codes indicate that the SDK encountered an unrecoverable error and requires intervention from the application. For example, an error is returned when the camera fails to open, and the app needs to inform the user that the camera cannot be used.
 */
export let ErrorCodeType = /*#__PURE__*/function (ErrorCodeType) {
  /**
   * 0: No error.
   */
  ErrorCodeType[ErrorCodeType["ErrOk"] = 0] = "ErrOk";
  /**
   * 1: A general error (no specific classification). Please retry the method call.
   */
  ErrorCodeType[ErrorCodeType["ErrFailed"] = 1] = "ErrFailed";
  /**
   * 2: Invalid parameter set in the method. For example, the specified channel name contains illegal characters. Please reset the parameter.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidArgument"] = 2] = "ErrInvalidArgument";
  /**
   * 3: The SDK is not ready. Possible reasons include: IRtcEngine initialization failed. Please reinitialize IRtcEngine.
   *  The user has not joined a channel when calling the method. Please check the method call logic.
   *  The user has not left the channel when calling rate or complain. Please check the method call logic.
   *  The audio module is not enabled.
   *  The assembly is incomplete.
   */
  ErrorCodeType[ErrorCodeType["ErrNotReady"] = 3] = "ErrNotReady";
  /**
   * 4: The current state of IRtcEngine does not support this operation. Possible reasons include:
   *  When using built-in encryption, the encryption mode is incorrect, or loading the external encryption library failed. Please check if the encryption enumeration value is correct, or reload the external encryption library.
   */
  ErrorCodeType[ErrorCodeType["ErrNotSupported"] = 4] = "ErrNotSupported";
  /**
   * 5: This method call was rejected. Possible reasons include: IRtcEngine initialization failed. Please reinitialize IRtcEngine.
   *  The channel name was set to an empty string "" when joining the channel. Please reset the channel name.
   *  In multi-channel scenarios, the specified channel name already exists when calling joinChannelEx to join the channel. Please reset the channel name.
   */
  ErrorCodeType[ErrorCodeType["ErrRefused"] = 5] = "ErrRefused";
  /**
   * 6: The buffer size is insufficient to hold the returned data.
   */
  ErrorCodeType[ErrorCodeType["ErrBufferTooSmall"] = 6] = "ErrBufferTooSmall";
  /**
   * 7: The method is called before IRtcEngine is initialized. Please ensure the IRtcEngine object is created and initialized before calling this method.
   */
  ErrorCodeType[ErrorCodeType["ErrNotInitialized"] = 7] = "ErrNotInitialized";
  /**
   * 8: The current state is invalid.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidState"] = 8] = "ErrInvalidState";
  /**
   * 9: No permission to operate. Please check whether the user has granted the app access to audio and video devices.
   */
  ErrorCodeType[ErrorCodeType["ErrNoPermission"] = 9] = "ErrNoPermission";
  /**
   * 10: The method call timed out. Some method calls require the SDK to return a result. If the SDK takes too long (over 10 seconds) to process the event, this error occurs.
   */
  ErrorCodeType[ErrorCodeType["ErrTimedout"] = 10] = "ErrTimedout";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCanceled"] = 11] = "ErrCanceled";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrTooOften"] = 12] = "ErrTooOften";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrBindSocket"] = 13] = "ErrBindSocket";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrNetDown"] = 14] = "ErrNetDown";
  /**
   * 17: Joining the channel was rejected. Possible reasons include:
   *  The user is already in the channel. It is recommended to determine whether the user is in the channel through the onConnectionStateChanged callback. Do not call this method again to join the channel unless receiving the ConnectionStateDisconnected (1) state.
   *  The user attempted to join a channel after calling startEchoTest for a call test without first calling stopEchoTest to end the current test. After starting a call test, you must call stopEchoTest to end the test before joining a channel.
   */
  ErrorCodeType[ErrorCodeType["ErrJoinChannelRejected"] = 17] = "ErrJoinChannelRejected";
  /**
   * 18: Failed to leave the channel. Possible reasons include:
   *  The user has already left the channel before calling leaveChannel. Just stop calling this method.
   *  The user called leaveChannel to exit the channel without having joined it. No further action is needed in this case.
   */
  ErrorCodeType[ErrorCodeType["ErrLeaveChannelRejected"] = 18] = "ErrLeaveChannelRejected";
  /**
   * 19: The resource is already in use and cannot be reused.
   */
  ErrorCodeType[ErrorCodeType["ErrAlreadyInUse"] = 19] = "ErrAlreadyInUse";
  /**
   * 20: The SDK aborted the request, possibly due to too many requests.
   */
  ErrorCodeType[ErrorCodeType["ErrAborted"] = 20] = "ErrAborted";
  /**
   * 21: Specific firewall settings on Windows caused IRtcEngine initialization to fail and crash.
   */
  ErrorCodeType[ErrorCodeType["ErrInitNetEngine"] = 21] = "ErrInitNetEngine";
  /**
   * 22: The SDK failed to allocate resources, possibly due to excessive resource usage by the app or system resource exhaustion.
   */
  ErrorCodeType[ErrorCodeType["ErrResourceLimited"] = 22] = "ErrResourceLimited";
  /**
   * 101: Invalid App ID. Please use a valid App ID to rejoin the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidAppId"] = 101] = "ErrInvalidAppId";
  /**
   * 102: Invalid channel name. Possible reason is incorrect data type for the parameter. Please use a valid channel name to rejoin the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidChannelName"] = 102] = "ErrInvalidChannelName";
  /**
   * 103: Unable to acquire server resources in the current region. Try specifying another region when initializing IRtcEngine.
   */
  ErrorCodeType[ErrorCodeType["ErrNoServerResources"] = 103] = "ErrNoServerResources";
  /**
   * 109: The current Token has expired and is no longer valid. Please request a new Token from your server and call renewToken to update it. Deprecated: This enumeration is deprecated. Use ConnectionChangedTokenExpired (9) in the onConnectionStateChanged callback instead.
   */
  ErrorCodeType[ErrorCodeType["ErrTokenExpired"] = 109] = "ErrTokenExpired";
  /**
   * Deprecated: This enumeration is deprecated. Use ConnectionChangedInvalidToken (8) in the onConnectionStateChanged callback instead. 110: Invalid Token. Common reasons include:
   *  App certificate is enabled in the console, but App ID + Token authentication is not used. When the App certificate is enabled, Token authentication must be used.
   *  The uid field used when generating the Token does not match the uid used when the user joins the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidToken"] = 110] = "ErrInvalidToken";
  /**
   * 111: Network connection interrupted. After the SDK establishes a connection with the server, it loses network connectivity for more than 4 seconds.
   */
  ErrorCodeType[ErrorCodeType["ErrConnectionInterrupted"] = 111] = "ErrConnectionInterrupted";
  /**
   * 112: Network connection lost. The network is disconnected and the SDK fails to reconnect to the server within 10 seconds.
   */
  ErrorCodeType[ErrorCodeType["ErrConnectionLost"] = 112] = "ErrConnectionLost";
  /**
   * 113: The user is not in a channel when calling the sendStreamMessage method.
   */
  ErrorCodeType[ErrorCodeType["ErrNotInChannel"] = 113] = "ErrNotInChannel";
  /**
   * 114: The data length exceeds 1 KB when calling sendStreamMessage.
   */
  ErrorCodeType[ErrorCodeType["ErrSizeTooLarge"] = 114] = "ErrSizeTooLarge";
  /**
   * 115: The data sending rate exceeds the limit (6 KB/s) when calling sendStreamMessage.
   */
  ErrorCodeType[ErrorCodeType["ErrBitrateLimit"] = 115] = "ErrBitrateLimit";
  /**
   * 116: The number of data streams created exceeds the limit (5) when calling createDataStream.
   */
  ErrorCodeType[ErrorCodeType["ErrTooManyDataStreams"] = 116] = "ErrTooManyDataStreams";
  /**
   * 117: Data stream sending timed out.
   */
  ErrorCodeType[ErrorCodeType["ErrStreamMessageTimeout"] = 117] = "ErrStreamMessageTimeout";
  /**
   * 119: Failed to switch user role. Please try rejoining the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrSetClientRoleNotAuthorized"] = 119] = "ErrSetClientRoleNotAuthorized";
  /**
   * 120: Media stream decryption failed. Possibly due to an incorrect key used when the user joined the channel. Please check the key entered when joining the channel, or guide the user to try rejoining.
   */
  ErrorCodeType[ErrorCodeType["ErrDecryptionFailed"] = 120] = "ErrDecryptionFailed";
  /**
   * 121: Invalid user ID.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidUserId"] = 121] = "ErrInvalidUserId";
  /**
   * 122: Data stream decryption failed. Possibly due to an incorrect key used when the user joined the channel. Please check the key entered when joining the channel, or guide the user to try rejoining.
   */
  ErrorCodeType[ErrorCodeType["ErrDatastreamDecryptionFailed"] = 122] = "ErrDatastreamDecryptionFailed";
  /**
   * 123: This user is banned by the server.
   */
  ErrorCodeType[ErrorCodeType["ErrClientIsBannedByServer"] = 123] = "ErrClientIsBannedByServer";
  /**
   * 130: The SDK does not support pushing encrypted streams to CDN.
   */
  ErrorCodeType[ErrorCodeType["ErrEncryptedStreamNotAllowedPublish"] = 130] = "ErrEncryptedStreamNotAllowedPublish";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrLicenseCredentialInvalid"] = 131] = "ErrLicenseCredentialInvalid";
  /**
   * 134: Invalid user account, possibly due to an invalid parameter.
   */
  ErrorCodeType[ErrorCodeType["ErrInvalidUserAccount"] = 134] = "ErrInvalidUserAccount";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrModuleNotFound"] = 157] = "ErrModuleNotFound";
  /**
   * 1001: Failed to load the media engine.
   */
  ErrorCodeType[ErrorCodeType["ErrCertRaw"] = 157] = "ErrCertRaw";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertJsonPart"] = 158] = "ErrCertJsonPart";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertJsonInval"] = 159] = "ErrCertJsonInval";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertJsonNomem"] = 160] = "ErrCertJsonNomem";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertCustom"] = 161] = "ErrCertCustom";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertCredential"] = 162] = "ErrCertCredential";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertSign"] = 163] = "ErrCertSign";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertFail"] = 164] = "ErrCertFail";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertBuf"] = 165] = "ErrCertBuf";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertNull"] = 166] = "ErrCertNull";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertDuedate"] = 167] = "ErrCertDuedate";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrCertRequest"] = 168] = "ErrCertRequest";
  /**
   * 200: Unsupported PCM format.
   */
  ErrorCodeType[ErrorCodeType["ErrPcmsendFormat"] = 200] = "ErrPcmsendFormat";
  /**
   * 201: Buffer overflow due to PCM sending rate being too fast.
   */
  ErrorCodeType[ErrorCodeType["ErrPcmsendBufferoverflow"] = 201] = "ErrPcmsendBufferoverflow";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrLoginAlreadyLogin"] = 428] = "ErrLoginAlreadyLogin";
  /**
   * @ignore
   */
  ErrorCodeType[ErrorCodeType["ErrLoadMediaEngine"] = 1001] = "ErrLoadMediaEngine";
  /**
   * 1005: Audio device error (unspecified). Please check whether the audio device is occupied by another application, or try rejoining the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmGeneralError"] = 1005] = "ErrAdmGeneralError";
  /**
   * 1008: Failed to initialize the playback device. Please check whether the playback device is occupied by another application, or try rejoining the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmInitPlayout"] = 1008] = "ErrAdmInitPlayout";
  /**
   * 1009: Failed to start the playback device. Please check whether the playback device is functioning properly.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmStartPlayout"] = 1009] = "ErrAdmStartPlayout";
  /**
   * 1010: Failed to stop the playback device.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmStopPlayout"] = 1010] = "ErrAdmStopPlayout";
  /**
   * 1011: Failed to initialize the recording device. Please check whether the recording device is functioning properly, or try rejoining the channel.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmInitRecording"] = 1011] = "ErrAdmInitRecording";
  /**
   * 1012: Failed to start the recording device. Please check whether the recording device is functioning properly.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmStartRecording"] = 1012] = "ErrAdmStartRecording";
  /**
   * 1013: Failed to stop the recording device.
   */
  ErrorCodeType[ErrorCodeType["ErrAdmStopRecording"] = 1013] = "ErrAdmStopRecording";
  /**
   * 1501: No permission to use the camera. Please check whether camera permission is enabled.
   */
  ErrorCodeType[ErrorCodeType["ErrVdmCameraNotAuthorized"] = 1501] = "ErrVdmCameraNotAuthorized";
  return ErrorCodeType;
}({});

/**
 * @ignore
 */
export let LicenseErrorType = /*#__PURE__*/function (LicenseErrorType) {
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrInvalid"] = 1] = "LicenseErrInvalid";
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrExpire"] = 2] = "LicenseErrExpire";
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrMinutesExceed"] = 3] = "LicenseErrMinutesExceed";
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrLimitedPeriod"] = 4] = "LicenseErrLimitedPeriod";
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrDiffDevices"] = 5] = "LicenseErrDiffDevices";
  /**
   * @ignore
   */
  LicenseErrorType[LicenseErrorType["LicenseErrInternal"] = 99] = "LicenseErrInternal";
  return LicenseErrorType;
}({});

/**
 * Permissions for SDK operations on the Audio Session.
 */
export let AudioSessionOperationRestriction = /*#__PURE__*/function (AudioSessionOperationRestriction) {
  /**
   * 0: No restriction. The SDK can modify the Audio Session.
   */
  AudioSessionOperationRestriction[AudioSessionOperationRestriction["AudioSessionOperationRestrictionNone"] = 0] = "AudioSessionOperationRestrictionNone";
  /**
   * 1: The SDK cannot modify the category of the Audio Session.
   */
  AudioSessionOperationRestriction[AudioSessionOperationRestriction["AudioSessionOperationRestrictionSetCategory"] = 1] = "AudioSessionOperationRestrictionSetCategory";
  /**
   * 2: The SDK cannot modify the category, mode, or categoryOptions of the Audio Session.
   */
  AudioSessionOperationRestriction[AudioSessionOperationRestriction["AudioSessionOperationRestrictionConfigureSession"] = 2] = "AudioSessionOperationRestrictionConfigureSession";
  /**
   * 4: When leaving the channel, the SDK keeps the Audio Session active, such as for playing audio in the background.
   */
  AudioSessionOperationRestriction[AudioSessionOperationRestriction["AudioSessionOperationRestrictionDeactivateSession"] = 4] = "AudioSessionOperationRestrictionDeactivateSession";
  /**
   * 128: Fully restricts the SDK from operating on the Audio Session. The SDK can no longer make any changes to the Audio Session.
   */
  AudioSessionOperationRestriction[AudioSessionOperationRestriction["AudioSessionOperationRestrictionAll"] = 128] = "AudioSessionOperationRestrictionAll";
  return AudioSessionOperationRestriction;
}({});

/**
 * Reason for user going offline.
 */
export let UserOfflineReasonType = /*#__PURE__*/function (UserOfflineReasonType) {
  /**
   * 0: The user left voluntarily.
   */
  UserOfflineReasonType[UserOfflineReasonType["UserOfflineQuit"] = 0] = "UserOfflineQuit";
  /**
   * 1: Timed out due to not receiving packets from the peer for a long time. Since the SDK uses an unreliable channel, it is also possible that the peer left the channel voluntarily, but the local side did not receive the leave message and mistakenly determined it as a timeout.
   */
  UserOfflineReasonType[UserOfflineReasonType["UserOfflineDropped"] = 1] = "UserOfflineDropped";
  /**
   * 2: The user's role changed from host to audience.
   */
  UserOfflineReasonType[UserOfflineReasonType["UserOfflineBecomeAudience"] = 2] = "UserOfflineBecomeAudience";
  return UserOfflineReasonType;
}({});

/**
 * Interface class.
 */
export let InterfaceIdType = /*#__PURE__*/function (InterfaceIdType) {
  /**
   * 1: IAudioDeviceManager interface class.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidAudioDeviceManager"] = 1] = "AgoraIidAudioDeviceManager";
  /**
   * 2: IVideoDeviceManager interface class.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidVideoDeviceManager"] = 2] = "AgoraIidVideoDeviceManager";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidParameterEngine"] = 3] = "AgoraIidParameterEngine";
  /**
   * 4: IMediaEngine interface class.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidMediaEngine"] = 4] = "AgoraIidMediaEngine";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidAudioEngine"] = 5] = "AgoraIidAudioEngine";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidVideoEngine"] = 6] = "AgoraIidVideoEngine";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidRtcConnection"] = 7] = "AgoraIidRtcConnection";
  /**
   * 8: This interface class is deprecated.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidSignalingEngine"] = 8] = "AgoraIidSignalingEngine";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidMediaEngineRegulator"] = 9] = "AgoraIidMediaEngineRegulator";
  /**
   * 11: ILocalSpatialAudioEngine interface class.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidLocalSpatialAudio"] = 11] = "AgoraIidLocalSpatialAudio";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidStateSync"] = 13] = "AgoraIidStateSync";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidMetaService"] = 14] = "AgoraIidMetaService";
  /**
   * 15: IMusicContentCenter interface class.
   */
  InterfaceIdType[InterfaceIdType["AgoraIidMusicContentCenter"] = 15] = "AgoraIidMusicContentCenter";
  /**
   * @ignore
   */
  InterfaceIdType[InterfaceIdType["AgoraIidH265Transcoder"] = 16] = "AgoraIidH265Transcoder";
  return InterfaceIdType;
}({});

/**
 * Network quality.
 */
export let QualityType = /*#__PURE__*/function (QualityType) {
  /**
   * 0: Unknown network quality.
   */
  QualityType[QualityType["QualityUnknown"] = 0] = "QualityUnknown";
  /**
   * 1: Excellent network quality.
   */
  QualityType[QualityType["QualityExcellent"] = 1] = "QualityExcellent";
  /**
   * 2: Subjectively similar to excellent, but bitrate may be slightly lower.
   */
  QualityType[QualityType["QualityGood"] = 2] = "QualityGood";
  /**
   * 3: Slightly impaired experience but communication is not affected.
   */
  QualityType[QualityType["QualityPoor"] = 3] = "QualityPoor";
  /**
   * 4: Barely communicable but not smooth.
   */
  QualityType[QualityType["QualityBad"] = 4] = "QualityBad";
  /**
   * 5: Very poor network quality, communication is nearly impossible.
   */
  QualityType[QualityType["QualityVbad"] = 5] = "QualityVbad";
  /**
   * 6: Communication is completely impossible.
   */
  QualityType[QualityType["QualityDown"] = 6] = "QualityDown";
  /**
   * @ignore
   */
  QualityType[QualityType["QualityUnsupported"] = 7] = "QualityUnsupported";
  /**
   * 8: Network quality detection in progress.
   */
  QualityType[QualityType["QualityDetecting"] = 8] = "QualityDetecting";
  return QualityType;
}({});

/**
 * @ignore
 */
export let FitModeType = /*#__PURE__*/function (FitModeType) {
  /**
   * @ignore
   */
  FitModeType[FitModeType["ModeCover"] = 1] = "ModeCover";
  /**
   * @ignore
   */
  FitModeType[FitModeType["ModeContain"] = 2] = "ModeContain";
  return FitModeType;
}({});

/**
 * Clockwise video rotation information.
 */
export let VideoOrientation = /*#__PURE__*/function (VideoOrientation) {
  /**
   * 0: (Default) Rotate 0 degrees clockwise.
   */
  VideoOrientation[VideoOrientation["VideoOrientation0"] = 0] = "VideoOrientation0";
  /**
   * 90: Rotate 90 degrees clockwise.
   */
  VideoOrientation[VideoOrientation["VideoOrientation90"] = 90] = "VideoOrientation90";
  /**
   * 180: Rotate 180 degrees clockwise.
   */
  VideoOrientation[VideoOrientation["VideoOrientation180"] = 180] = "VideoOrientation180";
  /**
   * 270: Rotate 270 degrees clockwise.
   */
  VideoOrientation[VideoOrientation["VideoOrientation270"] = 270] = "VideoOrientation270";
  return VideoOrientation;
}({});

/**
 * Video frame rate.
 */
export let FrameRate = /*#__PURE__*/function (FrameRate) {
  /**
   * 1: 1 fps.
   */
  FrameRate[FrameRate["FrameRateFps1"] = 1] = "FrameRateFps1";
  /**
   * 7: 7 fps.
   */
  FrameRate[FrameRate["FrameRateFps7"] = 7] = "FrameRateFps7";
  /**
   * 10: 10 fps.
   */
  FrameRate[FrameRate["FrameRateFps10"] = 10] = "FrameRateFps10";
  /**
   * 15: 15 fps.
   */
  FrameRate[FrameRate["FrameRateFps15"] = 15] = "FrameRateFps15";
  /**
   * 24: 24 fps.
   */
  FrameRate[FrameRate["FrameRateFps24"] = 24] = "FrameRateFps24";
  /**
   * 30: 30 fps.
   */
  FrameRate[FrameRate["FrameRateFps30"] = 30] = "FrameRateFps30";
  /**
   * @ignore
   */
  FrameRate[FrameRate["FrameRateFps60"] = 60] = "FrameRateFps60";
  return FrameRate;
}({});

/**
 * @ignore
 */
export let FrameWidth = /*#__PURE__*/function (FrameWidth) {
  /**
   * @ignore
   */
  FrameWidth[FrameWidth["FrameWidth960"] = 960] = "FrameWidth960";
  return FrameWidth;
}({});

/**
 * @ignore
 */
export let FrameHeight = /*#__PURE__*/function (FrameHeight) {
  /**
   * @ignore
   */
  FrameHeight[FrameHeight["FrameHeight540"] = 540] = "FrameHeight540";
  return FrameHeight;
}({});

/**
 * Video frame type.
 */
export let VideoFrameType = /*#__PURE__*/function (VideoFrameType) {
  /**
   * 0: Blank frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeBlankFrame"] = 0] = "VideoFrameTypeBlankFrame";
  /**
   * 3: Key frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeKeyFrame"] = 3] = "VideoFrameTypeKeyFrame";
  /**
   * 4: Delta frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeDeltaFrame"] = 4] = "VideoFrameTypeDeltaFrame";
  /**
   * 5: B frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeBFrame"] = 5] = "VideoFrameTypeBFrame";
  /**
   * 6: Droppable frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeDroppableFrame"] = 6] = "VideoFrameTypeDroppableFrame";
  /**
   * Unknown frame.
   */
  VideoFrameType[VideoFrameType["VideoFrameTypeUnknow"] = 7] = "VideoFrameTypeUnknow";
  return VideoFrameType;
}({});

/**
 * Orientation mode for video encoding.
 */
export let OrientationMode = /*#__PURE__*/function (OrientationMode) {
  /**
   * 0: (Default) In this mode, the SDK outputs video with the same orientation as the captured video. The receiving end rotates the video based on the received rotation information. This mode is suitable when the receiver can adjust the video orientation.
   *  If the captured video is in landscape mode, the output video is also in landscape mode.
   *  If the captured video is in portrait mode, the output video is also in portrait mode.
   */
  OrientationMode[OrientationMode["OrientationModeAdaptive"] = 0] = "OrientationModeAdaptive";
  /**
   * @ignore
   */
  OrientationMode[OrientationMode["OrientationModeFixedLandscape"] = 1] = "OrientationModeFixedLandscape";
  /**
   * @ignore
   */
  OrientationMode[OrientationMode["OrientationModeFixedPortrait"] = 2] = "OrientationModeFixedPortrait";
  return OrientationMode;
}({});

/**
 * Video encoding degradation preference when bandwidth is limited.
 */
export let DegradationPreference = /*#__PURE__*/function (DegradationPreference) {
  /**
   * -1: (Default) Auto mode. The SDK automatically selects MaintainFramerate, MaintainBalanced, or MaintainResolution based on your video scenario settings to achieve optimal overall quality of experience (QoE).
   */
  DegradationPreference[DegradationPreference["MaintainAuto"] = -1] = "MaintainAuto";
  /**
   * 0: When bandwidth is limited, prioritize reducing video frame rate while maintaining resolution. This degradation preference suits scenarios where video quality is prioritized. Deprecated: This enum is deprecated. Use other enums instead.
   */
  DegradationPreference[DegradationPreference["MaintainQuality"] = 0] = "MaintainQuality";
  /**
   * 1: When bandwidth is limited, prioritize reducing video resolution while maintaining frame rate. This degradation preference suits scenarios where smoothness is prioritized and some quality loss is acceptable.
   */
  DegradationPreference[DegradationPreference["MaintainFramerate"] = 1] = "MaintainFramerate";
  /**
   * 2: When bandwidth is limited, reduce both video frame rate and resolution. The degradation level of MaintainBalanced is lower than that of MaintainQuality and MaintainFramerate, suitable for scenarios with both smoothness and quality constraints. The resolution of the locally sent video may change. The remote user must be able to handle this. See onVideoSizeChanged.
   */
  DegradationPreference[DegradationPreference["MaintainBalanced"] = 2] = "MaintainBalanced";
  /**
   * 3: When bandwidth is limited, prioritize reducing video frame rate while maintaining resolution. This degradation preference suits scenarios where video quality is prioritized.
   */
  DegradationPreference[DegradationPreference["MaintainResolution"] = 3] = "MaintainResolution";
  /**
   * @ignore
   */
  DegradationPreference[DegradationPreference["Disabled"] = 100] = "Disabled";
  return DegradationPreference;
}({});

/**
 * Video dimensions.
 */
export class VideoDimensions {}

/**
 * Maximum frame rate supported by the screen sharing device.
 */
export let ScreenCaptureFramerateCapability = /*#__PURE__*/function (ScreenCaptureFramerateCapability) {
  /**
   * 0: Supports up to 15 fps.
   */
  ScreenCaptureFramerateCapability[ScreenCaptureFramerateCapability["ScreenCaptureFramerateCapability15Fps"] = 0] = "ScreenCaptureFramerateCapability15Fps";
  /**
   * 1: Supports up to 30 fps.
   */
  ScreenCaptureFramerateCapability[ScreenCaptureFramerateCapability["ScreenCaptureFramerateCapability30Fps"] = 1] = "ScreenCaptureFramerateCapability30Fps";
  /**
   * 2: Supports up to 60 fps.
   */
  ScreenCaptureFramerateCapability[ScreenCaptureFramerateCapability["ScreenCaptureFramerateCapability60Fps"] = 2] = "ScreenCaptureFramerateCapability60Fps";
  return ScreenCaptureFramerateCapability;
}({});

/**
 * @ignore
 */
export let VideoCodecCapabilityLevel = /*#__PURE__*/function (VideoCodecCapabilityLevel) {
  /**
   * @ignore
   */
  VideoCodecCapabilityLevel[VideoCodecCapabilityLevel["CodecCapabilityLevelUnspecified"] = -1] = "CodecCapabilityLevelUnspecified";
  /**
   * @ignore
   */
  VideoCodecCapabilityLevel[VideoCodecCapabilityLevel["CodecCapabilityLevelBasicSupport"] = 5] = "CodecCapabilityLevelBasicSupport";
  /**
   * @ignore
   */
  VideoCodecCapabilityLevel[VideoCodecCapabilityLevel["CodecCapabilityLevel1080p30fps"] = 10] = "CodecCapabilityLevel1080p30fps";
  /**
   * @ignore
   */
  VideoCodecCapabilityLevel[VideoCodecCapabilityLevel["CodecCapabilityLevel1080p60fps"] = 20] = "CodecCapabilityLevel1080p60fps";
  /**
   * @ignore
   */
  VideoCodecCapabilityLevel[VideoCodecCapabilityLevel["CodecCapabilityLevel4k60fps"] = 30] = "CodecCapabilityLevel4k60fps";
  return VideoCodecCapabilityLevel;
}({});

/**
 * Video codec format.
 */
export let VideoCodecType = /*#__PURE__*/function (VideoCodecType) {
  /**
   * 0: (Default) No specific codec format. The SDK automatically selects a suitable codec based on the video resolution and device performance.
   */
  VideoCodecType[VideoCodecType["VideoCodecNone"] = 0] = "VideoCodecNone";
  /**
   * 1: Standard VP8.
   */
  VideoCodecType[VideoCodecType["VideoCodecVp8"] = 1] = "VideoCodecVp8";
  /**
   * 2: Standard H.264.
   */
  VideoCodecType[VideoCodecType["VideoCodecH264"] = 2] = "VideoCodecH264";
  /**
   * 3: Standard H.265.
   */
  VideoCodecType[VideoCodecType["VideoCodecH265"] = 3] = "VideoCodecH265";
  /**
   * 6: Generic. This type is mainly used for transmitting raw video data (e.g., user-encrypted video frames). The video frames are returned via callback and need to be decoded and rendered by the user.
   */
  VideoCodecType[VideoCodecType["VideoCodecGeneric"] = 6] = "VideoCodecGeneric";
  /**
   * @ignore
   */
  VideoCodecType[VideoCodecType["VideoCodecGenericH264"] = 7] = "VideoCodecGenericH264";
  /**
   * @ignore
   */
  VideoCodecType[VideoCodecType["VideoCodecAv1"] = 12] = "VideoCodecAv1";
  /**
   * @ignore
   */
  VideoCodecType[VideoCodecType["VideoCodecVp9"] = 13] = "VideoCodecVp9";
  /**
   * 20: Generic JPEG. Requires less computing power and is suitable for IoT devices with limited resources.
   */
  VideoCodecType[VideoCodecType["VideoCodecGenericJpeg"] = 20] = "VideoCodecGenericJpeg";
  return VideoCodecType;
}({});

/**
 * Camera focal length type.
 */
export let CameraFocalLengthType = /*#__PURE__*/function (CameraFocalLengthType) {
  /**
   * 0: (Default) Standard lens.
   */
  CameraFocalLengthType[CameraFocalLengthType["CameraFocalLengthDefault"] = 0] = "CameraFocalLengthDefault";
  /**
   * 1: Wide-angle lens.
   */
  CameraFocalLengthType[CameraFocalLengthType["CameraFocalLengthWideAngle"] = 1] = "CameraFocalLengthWideAngle";
  /**
   * 2: Ultra wide-angle lens.
   */
  CameraFocalLengthType[CameraFocalLengthType["CameraFocalLengthUltraWide"] = 2] = "CameraFocalLengthUltraWide";
  /**
   * 3: (iOS only) Telephoto lens.
   */
  CameraFocalLengthType[CameraFocalLengthType["CameraFocalLengthTelephoto"] = 3] = "CameraFocalLengthTelephoto";
  return CameraFocalLengthType;
}({});

/**
 * @ignore
 */
export let TCcMode = /*#__PURE__*/function (TCcMode) {
  /**
   * @ignore
   */
  TCcMode[TCcMode["CcEnabled"] = 0] = "CcEnabled";
  /**
   * @ignore
   */
  TCcMode[TCcMode["CcDisabled"] = 1] = "CcDisabled";
  return TCcMode;
}({});

/**
 * @ignore
 */
export class SenderOptions {}

/**
 * Audio codec format.
 */
export let AudioCodecType = /*#__PURE__*/function (AudioCodecType) {
  /**
   * 1: OPUS.
   */
  AudioCodecType[AudioCodecType["AudioCodecOpus"] = 1] = "AudioCodecOpus";
  /**
   * 3: PCMA.
   */
  AudioCodecType[AudioCodecType["AudioCodecPcma"] = 3] = "AudioCodecPcma";
  /**
   * 4: PCMU.
   */
  AudioCodecType[AudioCodecType["AudioCodecPcmu"] = 4] = "AudioCodecPcmu";
  /**
   * 5: G722.
   */
  AudioCodecType[AudioCodecType["AudioCodecG722"] = 5] = "AudioCodecG722";
  /**
   * 8: LC-AAC.
   */
  AudioCodecType[AudioCodecType["AudioCodecAaclc"] = 8] = "AudioCodecAaclc";
  /**
   * 9: HE-AAC.
   */
  AudioCodecType[AudioCodecType["AudioCodecHeaac"] = 9] = "AudioCodecHeaac";
  /**
   * 10: JC1.
   */
  AudioCodecType[AudioCodecType["AudioCodecJc1"] = 10] = "AudioCodecJc1";
  /**
   * 11: HE-AAC v2.
   */
  AudioCodecType[AudioCodecType["AudioCodecHeaac2"] = 11] = "AudioCodecHeaac2";
  /**
   * @ignore
   */
  AudioCodecType[AudioCodecType["AudioCodecLpcnet"] = 12] = "AudioCodecLpcnet";
  /**
   * @ignore
   */
  AudioCodecType[AudioCodecType["AudioCodecOpusmc"] = 13] = "AudioCodecOpusmc";
  return AudioCodecType;
}({});

/**
 * Audio encoding type.
 */
export let AudioEncodingType = /*#__PURE__*/function (AudioEncodingType) {
  /**
   * 0x010101: AAC encoding format, 16000 Hz sampling rate, low quality. The encoded file size for 10 minutes of audio is approximately 1.2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac16000Low"] = 65793] = "AudioEncodingTypeAac16000Low";
  /**
   * 0x010102: AAC encoding format, 16000 Hz sampling rate, medium quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac16000Medium"] = 65794] = "AudioEncodingTypeAac16000Medium";
  /**
   * 0x010201: AAC encoding format, 32000 Hz sampling rate, low quality. The encoded file size for 10 minutes of audio is approximately 1.2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac32000Low"] = 66049] = "AudioEncodingTypeAac32000Low";
  /**
   * 0x010202: AAC encoding format, 32000 Hz sampling rate, medium quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac32000Medium"] = 66050] = "AudioEncodingTypeAac32000Medium";
  /**
   * 0x010203: AAC encoding format, 32000 Hz sampling rate, high quality. The encoded file size for 10 minutes of audio is approximately 3.5 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac32000High"] = 66051] = "AudioEncodingTypeAac32000High";
  /**
   * 0x010302: AAC encoding format, 48000 Hz sampling rate, medium quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac48000Medium"] = 66306] = "AudioEncodingTypeAac48000Medium";
  /**
   * 0x010303: AAC encoding format, 48000 Hz sampling rate, high quality. The encoded file size for 10 minutes of audio is approximately 3.5 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeAac48000High"] = 66307] = "AudioEncodingTypeAac48000High";
  /**
   * 0x020101: OPUS encoding format, 16000 Hz sampling rate, low quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeOpus16000Low"] = 131329] = "AudioEncodingTypeOpus16000Low";
  /**
   * 0x020102: OPUS encoding format, 16000 Hz sampling rate, medium quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeOpus16000Medium"] = 131330] = "AudioEncodingTypeOpus16000Medium";
  /**
   * 0x020302: OPUS encoding format, 48000 Hz sampling rate, medium quality. The encoded file size for 10 minutes of audio is approximately 2 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeOpus48000Medium"] = 131842] = "AudioEncodingTypeOpus48000Medium";
  /**
   * 0x020303: OPUS encoding format, 48000 Hz sampling rate, high quality. The encoded file size for 10 minutes of audio is approximately 3.5 MB.
   */
  AudioEncodingType[AudioEncodingType["AudioEncodingTypeOpus48000High"] = 131843] = "AudioEncodingTypeOpus48000High";
  return AudioEncodingType;
}({});

/**
 * Watermark adaptation mode.
 */
export let WatermarkFitMode = /*#__PURE__*/function (WatermarkFitMode) {
  /**
   * 0: Uses the values of positionInLandscapeMode and positionInPortraitMode set in WatermarkOptions. The settings in WatermarkRatio are ignored.
   */
  WatermarkFitMode[WatermarkFitMode["FitModeCoverPosition"] = 0] = "FitModeCoverPosition";
  /**
   * 1: Uses the value set in WatermarkRatio. The settings of positionInLandscapeMode and positionInPortraitMode in WatermarkOptions are ignored.
   */
  WatermarkFitMode[WatermarkFitMode["FitModeUseImageRatio"] = 1] = "FitModeUseImageRatio";
  return WatermarkFitMode;
}({});

/**
 * @ignore
 */
export class EncodedAudioFrameAdvancedSettings {}

/**
 * Information about encoded audio.
 */
export class EncodedAudioFrameInfo {}

/**
 * @ignore
 */
export class AudioPcmDataInfo {}

/**
 * @ignore
 */
export let H264PacketizeMode = /*#__PURE__*/function (H264PacketizeMode) {
  /**
   * @ignore
   */
  H264PacketizeMode[H264PacketizeMode["NonInterleaved"] = 0] = "NonInterleaved";
  /**
   * @ignore
   */
  H264PacketizeMode[H264PacketizeMode["SingleNalUnit"] = 1] = "SingleNalUnit";
  return H264PacketizeMode;
}({});

/**
 * Type of video stream.
 */
export let VideoStreamType = /*#__PURE__*/function (VideoStreamType) {
  /**
   * 0: High video stream, i.e., high resolution and high bitrate stream.
   */
  VideoStreamType[VideoStreamType["VideoStreamHigh"] = 0] = "VideoStreamHigh";
  /**
   * 1: Low video stream, i.e., low resolution and low bitrate stream.
   */
  VideoStreamType[VideoStreamType["VideoStreamLow"] = 1] = "VideoStreamLow";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer1"] = 4] = "VideoStreamLayer1";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer2"] = 5] = "VideoStreamLayer2";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer3"] = 6] = "VideoStreamLayer3";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer4"] = 7] = "VideoStreamLayer4";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer5"] = 8] = "VideoStreamLayer5";
  /**
   * @ignore
   */
  VideoStreamType[VideoStreamType["VideoStreamLayer6"] = 9] = "VideoStreamLayer6";
  return VideoStreamType;
}({});

/**
 * Video subscription settings.
 */
export class VideoSubscriptionOptions {}

/**
 * Maximum length of the user account.
 */
export let MaxUserAccountLengthType = /*#__PURE__*/function (MaxUserAccountLengthType) {
  /**
   * The maximum length of the user account is 255 characters.
   */
  MaxUserAccountLengthType[MaxUserAccountLengthType["MaxUserAccountLength"] = 256] = "MaxUserAccountLength";
  return MaxUserAccountLengthType;
}({});

/**
 * Information about externally encoded video frames.
 */
export class EncodedVideoFrameInfo {}

/**
 * Compression preference type for video encoding.
 */
export let CompressionPreference = /*#__PURE__*/function (CompressionPreference) {
  /**
   * -1: (Default) Automatic mode. The SDK automatically selects PreferLowLatency or PreferQuality based on your video scenario to provide the best user experience.
   */
  CompressionPreference[CompressionPreference["PreferCompressionAuto"] = -1] = "PreferCompressionAuto";
  /**
   * 0: Low latency preference. The SDK compresses video frames to reduce latency. Suitable for scenarios where smoothness is prioritized and some quality loss is acceptable.
   */
  CompressionPreference[CompressionPreference["PreferLowLatency"] = 0] = "PreferLowLatency";
  /**
   * 1: High quality preference. The SDK compresses video frames while maintaining video quality. Suitable for scenarios where video quality is prioritized.
   */
  CompressionPreference[CompressionPreference["PreferQuality"] = 1] = "PreferQuality";
  return CompressionPreference;
}({});

/**
 * Video encoder preference.
 */
export let EncodingPreference = /*#__PURE__*/function (EncodingPreference) {
  /**
   * -1: Adaptive preference. The SDK automatically selects the optimal encoder type based on platform, device type, and other factors.
   */
  EncodingPreference[EncodingPreference["PreferAuto"] = -1] = "PreferAuto";
  /**
   * 0: Software encoder preference. The SDK prioritizes using the software encoder for video encoding.
   */
  EncodingPreference[EncodingPreference["PreferSoftware"] = 0] = "PreferSoftware";
  /**
   * 1: Hardware encoder preference. The SDK prioritizes using the hardware encoder for video encoding. If the device does not support hardware encoding, the SDK automatically switches to software encoding and reports the current encoder type via the hwEncoderAccelerating field in the onLocalVideoStats callback.
   */
  EncodingPreference[EncodingPreference["PreferHardware"] = 1] = "PreferHardware";
  return EncodingPreference;
}({});

/**
 * Advanced options for video encoding.
 */
export class AdvanceOptions {}

/**
 * Mirror mode type.
 */
export let VideoMirrorModeType = /*#__PURE__*/function (VideoMirrorModeType) {
  /**
   * 0: The SDK determines the mirror mode.
   *  Local view mirror mode: If you use the front camera, the local view mirror mode is enabled by default; if you use the rear camera, the local view mirror mode is disabled by default.
   *  Remote user view mirror mode: The mirror mode for remote users is disabled by default.
   */
  VideoMirrorModeType[VideoMirrorModeType["VideoMirrorModeAuto"] = 0] = "VideoMirrorModeAuto";
  /**
   * 1: Enable mirror mode.
   */
  VideoMirrorModeType[VideoMirrorModeType["VideoMirrorModeEnabled"] = 1] = "VideoMirrorModeEnabled";
  /**
   * 2: Disable mirror mode.
   */
  VideoMirrorModeType[VideoMirrorModeType["VideoMirrorModeDisabled"] = 2] = "VideoMirrorModeDisabled";
  return VideoMirrorModeType;
}({});

/**
 * @ignore
 */
export let CameraFormatType = /*#__PURE__*/function (CameraFormatType) {
  /**
   * @ignore
   */
  CameraFormatType[CameraFormatType["CameraFormatNv12"] = 0] = "CameraFormatNv12";
  /**
   * @ignore
   */
  CameraFormatType[CameraFormatType["CameraFormatBgra"] = 1] = "CameraFormatBgra";
  return CameraFormatType;
}({});

/**
 * @ignore
 */
export let VideoModuleType = /*#__PURE__*/function (VideoModuleType) {
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleCapturer"] = 0] = "VideoModuleCapturer";
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleSoftwareEncoder"] = 1] = "VideoModuleSoftwareEncoder";
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleHardwareEncoder"] = 2] = "VideoModuleHardwareEncoder";
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleSoftwareDecoder"] = 3] = "VideoModuleSoftwareDecoder";
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleHardwareDecoder"] = 4] = "VideoModuleHardwareDecoder";
  /**
   * @ignore
   */
  VideoModuleType[VideoModuleType["VideoModuleRenderer"] = 5] = "VideoModuleRenderer";
  return VideoModuleType;
}({});

/**
 * @ignore
 */
export let HdrCapability = /*#__PURE__*/function (HdrCapability) {
  /**
   * @ignore
   */
  HdrCapability[HdrCapability["HdrCapabilityUnknown"] = -1] = "HdrCapabilityUnknown";
  /**
   * @ignore
   */
  HdrCapability[HdrCapability["HdrCapabilityUnsupported"] = 0] = "HdrCapabilityUnsupported";
  /**
   * @ignore
   */
  HdrCapability[HdrCapability["HdrCapabilitySupported"] = 1] = "HdrCapabilitySupported";
  return HdrCapability;
}({});

/**
 * Codec capability bit mask.
 */
export let CodecCapMask = /*#__PURE__*/function (CodecCapMask) {
  /**
   * (0): Codec not supported.
   */
  CodecCapMask[CodecCapMask["CodecCapMaskNone"] = 0] = "CodecCapMaskNone";
  /**
   * (1 << 0): Supports hardware decoding.
   */
  CodecCapMask[CodecCapMask["CodecCapMaskHwDec"] = 1] = "CodecCapMaskHwDec";
  /**
   * (1 << 1): Supports hardware encoding.
   */
  CodecCapMask[CodecCapMask["CodecCapMaskHwEnc"] = 2] = "CodecCapMaskHwEnc";
  /**
   * (1 << 2): Supports software decoding.
   */
  CodecCapMask[CodecCapMask["CodecCapMaskSwDec"] = 4] = "CodecCapMaskSwDec";
  /**
   * (1 << 3): Supports software encoding.
   */
  CodecCapMask[CodecCapMask["CodecCapMaskSwEnc"] = 8] = "CodecCapMaskSwEnc";
  return CodecCapMask;
}({});

/**
 * Codec capability levels.
 */
export class CodecCapLevels {}

/**
 * Information about codec capabilities supported by the SDK.
 */
export class CodecCapInfo {}

/**
 * Focal length information supported by the camera, including camera direction and focal length type.
 */
export class FocalLengthInfo {}

/**
 * Configuration for the video encoder.
 */
export class VideoEncoderConfiguration {}

/**
 * Data stream settings.
 *
 * The table below shows the SDK behavior under different parameter settings: syncWithAudio ordered
 * SDK Behavior false false
 * The SDK immediately triggers the onStreamMessage callback upon receiving the data packet. true false
 * If the data packet delay is within the audio delay range, the SDK triggers the onStreamMessage callback synchronized with the audio packet during playback. If the delay exceeds the audio delay, the SDK triggers the callback immediately upon receiving the packet, which may cause desynchronization between audio and data packets. false true
 * If the data packet delay is within 5 seconds, the SDK corrects the packet disorder. If the delay exceeds 5 seconds, the SDK discards the packet. true true
 * If the data packet delay is within the audio delay range, the SDK corrects the packet disorder. If the delay exceeds the audio delay, the SDK discards the packet.
 */
export class DataStreamConfig {}

/**
 * Mode for sending video streams.
 */
export let SimulcastStreamMode = /*#__PURE__*/function (SimulcastStreamMode) {
  /**
   * -1: By default, the low-quality stream is not sent until a subscription request for the low-quality stream is received from the receiver, at which point the low-quality stream is sent automatically.
   */
  SimulcastStreamMode[SimulcastStreamMode["AutoSimulcastStream"] = -1] = "AutoSimulcastStream";
  /**
   * 0: Never send the low-quality stream.
   */
  SimulcastStreamMode[SimulcastStreamMode["DisableSimulcastStream"] = 0] = "DisableSimulcastStream";
  /**
   * 1: Always send the low-quality stream.
   */
  SimulcastStreamMode[SimulcastStreamMode["EnableSimulcastStream"] = 1] = "EnableSimulcastStream";
  return SimulcastStreamMode;
}({});

/**
 * Configuration for video simulcast stream.
 */
export class SimulcastStreamConfig {}

/**
 * @ignore
 */
export let StreamLayerIndex = /*#__PURE__*/function (StreamLayerIndex) {
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer1"] = 0] = "StreamLayer1";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer2"] = 1] = "StreamLayer2";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer3"] = 2] = "StreamLayer3";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer4"] = 3] = "StreamLayer4";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer5"] = 4] = "StreamLayer5";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayer6"] = 5] = "StreamLayer6";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLow"] = 6] = "StreamLow";
  /**
   * @ignore
   */
  StreamLayerIndex[StreamLayerIndex["StreamLayerCountMax"] = 7] = "StreamLayerCountMax";
  return StreamLayerIndex;
}({});

/**
 * @ignore
 */
export class StreamLayerConfig {}

/**
 * @ignore
 */
export class SimulcastConfig {}

/**
 * The position of the target area relative to the entire screen or window. If not specified, it represents the entire screen or window.
 */
export class Rectangle {}

/**
 * Position and size of the watermark on the screen.
 *
 * The position and size of the watermark on the screen are determined by xRatio, yRatio, and widthRatio :
 *  (xRatio, yRatio) indicates the coordinates of the top-left corner of the watermark, representing the distance from the top-left corner of the screen. widthRatio determines the width of the watermark.
 */
export class WatermarkRatio {}

/**
 * Configures the watermark image.
 *
 * Used to configure the settings for the watermark image to be added.
 */
export class WatermarkOptions {}

/**
 * Mode for multipath data transmission.
 *
 * Since Available since v4.6.2.
 */
export let MultipathMode = /*#__PURE__*/function (MultipathMode) {
  /**
   * (0): Redundant transmission mode. The same data is transmitted redundantly through all available paths.
   */
  MultipathMode[MultipathMode["Duplicate"] = 0] = "Duplicate";
  /**
   * @ignore
   */
  MultipathMode[MultipathMode["Dynamic"] = 1] = "Dynamic";
  return MultipathMode;
}({});

/**
 * Network path types used for multipath transmission.
 *
 * Since Available since v4.6.2.
 */
export let MultipathType = /*#__PURE__*/function (MultipathType) {
  /**
   * (0): Local Area Network (LAN) path.
   */
  MultipathType[MultipathType["Lan"] = 0] = "Lan";
  /**
   * (1): Wi-Fi path.
   */
  MultipathType[MultipathType["Wifi"] = 1] = "Wifi";
  /**
   * (2): Mobile network path.
   */
  MultipathType[MultipathType["Mobile"] = 2] = "Mobile";
  /**
   * (99): Unknown or unspecified network path.
   */
  MultipathType[MultipathType["Unknown"] = 99] = "Unknown";
  return MultipathType;
}({});

/**
 * Used to obtain statistics for a specific network path.
 *
 * Since Available since v4.6.2.
 */
export class PathStats {}

/**
 * Used to aggregate statistics of each network path in multipath transmission.
 *
 * Since Available since v4.6.2.
 */
export class MultipathStats {}

/**
 * Call-related statistics.
 */
export class RtcStats {}

/**
 * User roles in live broadcasting.
 */
export let ClientRoleType = /*#__PURE__*/function (ClientRoleType) {
  /**
   * 1: Broadcaster. A broadcaster can both send and receive streams.
   */
  ClientRoleType[ClientRoleType["ClientRoleBroadcaster"] = 1] = "ClientRoleBroadcaster";
  /**
   * 2: (Default) Audience. An audience can only receive streams, not send.
   */
  ClientRoleType[ClientRoleType["ClientRoleAudience"] = 2] = "ClientRoleAudience";
  return ClientRoleType;
}({});

/**
 * Adaptation of local video quality since last statistics (based on target frame rate and target bitrate).
 */
export let QualityAdaptIndication = /*#__PURE__*/function (QualityAdaptIndication) {
  /**
   * 0: Local video quality remains unchanged.
   */
  QualityAdaptIndication[QualityAdaptIndication["AdaptNone"] = 0] = "AdaptNone";
  /**
   * 1: Local video quality improves due to increased network bandwidth.
   */
  QualityAdaptIndication[QualityAdaptIndication["AdaptUpBandwidth"] = 1] = "AdaptUpBandwidth";
  /**
   * 2: Local video quality degrades due to decreased network bandwidth.
   */
  QualityAdaptIndication[QualityAdaptIndication["AdaptDownBandwidth"] = 2] = "AdaptDownBandwidth";
  return QualityAdaptIndication;
}({});

/**
 * Latency level of audience in a live broadcast channel. This enum takes effect only when the user role is set to ClientRoleAudience.
 */
export let AudienceLatencyLevelType = /*#__PURE__*/function (AudienceLatencyLevelType) {
  /**
   * 1: Low latency.
   */
  AudienceLatencyLevelType[AudienceLatencyLevelType["AudienceLatencyLevelLowLatency"] = 1] = "AudienceLatencyLevelLowLatency";
  /**
   * 2: (Default) Ultra-low latency.
   */
  AudienceLatencyLevelType[AudienceLatencyLevelType["AudienceLatencyLevelUltraLowLatency"] = 2] = "AudienceLatencyLevelUltraLowLatency";
  return AudienceLatencyLevelType;
}({});

/**
 * User role property settings.
 */
export class ClientRoleOptions {}

/**
 * The subjective experience quality of the local user when receiving remote audio.
 */
export let ExperienceQualityType = /*#__PURE__*/function (ExperienceQualityType) {
  /**
   * 0: Good subjective experience quality.
   */
  ExperienceQualityType[ExperienceQualityType["ExperienceQualityGood"] = 0] = "ExperienceQualityGood";
  /**
   * 1: Poor subjective experience quality.
   */
  ExperienceQualityType[ExperienceQualityType["ExperienceQualityBad"] = 1] = "ExperienceQualityBad";
  return ExperienceQualityType;
}({});

/**
 * The reason for poor subjective experience quality of the local user when receiving remote audio.
 */
export let ExperiencePoorReason = /*#__PURE__*/function (ExperiencePoorReason) {
  /**
   * 0: No reason, indicates good subjective experience quality.
   */
  ExperiencePoorReason[ExperiencePoorReason["ExperienceReasonNone"] = 0] = "ExperienceReasonNone";
  /**
   * 1: Poor network quality of the remote user.
   */
  ExperiencePoorReason[ExperiencePoorReason["RemoteNetworkQualityPoor"] = 1] = "RemoteNetworkQualityPoor";
  /**
   * 2: Poor network quality of the local user.
   */
  ExperiencePoorReason[ExperiencePoorReason["LocalNetworkQualityPoor"] = 2] = "LocalNetworkQualityPoor";
  /**
   * 4: Weak Wi-Fi or mobile data signal of the local user.
   */
  ExperiencePoorReason[ExperiencePoorReason["WirelessSignalPoor"] = 4] = "WirelessSignalPoor";
  /**
   * 8: Wi-Fi and Bluetooth are enabled simultaneously on the local device, causing signal interference and degraded audio transmission quality.
   */
  ExperiencePoorReason[ExperiencePoorReason["WifiBluetoothCoexist"] = 8] = "WifiBluetoothCoexist";
  return ExperiencePoorReason;
}({});

/**
 * AI noise reduction mode.
 */
export let AudioAinsMode = /*#__PURE__*/function (AudioAinsMode) {
  /**
   * 0: (Default) Balanced noise reduction mode. Choose this mode if you want a balance between noise suppression and latency.
   */
  AudioAinsMode[AudioAinsMode["AinsModeBalanced"] = 0] = "AinsModeBalanced";
  /**
   * 1: Aggressive noise reduction mode. Suitable for scenarios with high noise suppression requirements, such as outdoor live streaming. This mode reduces noise more significantly but may slightly damage voice quality.
   */
  AudioAinsMode[AudioAinsMode["AinsModeAggressive"] = 1] = "AinsModeAggressive";
  /**
   * 2: Low-latency aggressive noise reduction mode. This mode has about half the latency of weak and aggressive noise reduction modes, suitable for scenarios requiring both noise reduction and low latency, such as real-time chorus.
   */
  AudioAinsMode[AudioAinsMode["AinsModeUltralowlatency"] = 2] = "AinsModeUltralowlatency";
  return AudioAinsMode;
}({});

/**
 * Audio encoding properties.
 */
export let AudioProfileType = /*#__PURE__*/function (AudioProfileType) {
  /**
   * 0: Default value.
   *  In live broadcast: 48 kHz sample rate, music encoding, mono, max bitrate 64 Kbps.
   *  In communication: 32 kHz sample rate, speech encoding, mono, max bitrate 18 Kbps.
   */
  AudioProfileType[AudioProfileType["AudioProfileDefault"] = 0] = "AudioProfileDefault";
  /**
   * 1: Sets 32 kHz sample rate, speech encoding, mono, max bitrate 18 Kbps.
   */
  AudioProfileType[AudioProfileType["AudioProfileSpeechStandard"] = 1] = "AudioProfileSpeechStandard";
  /**
   * 2: Sets 48 kHz sample rate, music encoding, mono, max bitrate 64 Kbps.
   */
  AudioProfileType[AudioProfileType["AudioProfileMusicStandard"] = 2] = "AudioProfileMusicStandard";
  /**
   * 3: Sets 48 kHz sample rate, music encoding, stereo, max bitrate 80 Kbps.
   * To enable stereo, you also need to call setAdvancedAudioOptions and set audioProcessingChannels to AudioProcessingStereo in AdvancedAudioOptions.
   */
  AudioProfileType[AudioProfileType["AudioProfileMusicStandardStereo"] = 3] = "AudioProfileMusicStandardStereo";
  /**
   * 4: Sets 48 kHz sample rate, music encoding, mono, max bitrate 96 Kbps.
   */
  AudioProfileType[AudioProfileType["AudioProfileMusicHighQuality"] = 4] = "AudioProfileMusicHighQuality";
  /**
   * 5: Sets 48 kHz sample rate, music encoding, stereo, max bitrate 128 Kbps.
   * To enable stereo, you also need to call setAdvancedAudioOptions and set audioProcessingChannels to AudioProcessingStereo in AdvancedAudioOptions.
   */
  AudioProfileType[AudioProfileType["AudioProfileMusicHighQualityStereo"] = 5] = "AudioProfileMusicHighQualityStereo";
  /**
   * 6: Sets 16 kHz sample rate, speech encoding, mono, applies echo cancellation algorithm AEC.
   */
  AudioProfileType[AudioProfileType["AudioProfileIot"] = 6] = "AudioProfileIot";
  /**
   * Boundary of enum values.
   */
  AudioProfileType[AudioProfileType["AudioProfileNum"] = 7] = "AudioProfileNum";
  return AudioProfileType;
}({});

/**
 * Audio scenario.
 */
export let AudioScenarioType = /*#__PURE__*/function (AudioScenarioType) {
  /**
   * 0: (Default) Auto scenario. Automatically matches appropriate audio quality based on user role and audio route.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioDefault"] = 0] = "AudioScenarioDefault";
  /**
   * 3: High-quality scenario, suitable for music-centric use cases. For example: instrument practice.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioGameStreaming"] = 3] = "AudioScenarioGameStreaming";
  /**
   * 5: Chatroom scenario, suitable for frequent mic on/off situations. For example: education scenarios.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioChatroom"] = 5] = "AudioScenarioChatroom";
  /**
   * 7: Chorus scenario. Suitable for real-time chorus with low latency under good network conditions.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioChorus"] = 7] = "AudioScenarioChorus";
  /**
   * 8: Meeting scenario, suitable for multi-person voice-centric meetings.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioMeeting"] = 8] = "AudioScenarioMeeting";
  /**
   * @ignore
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioAiServer"] = 9] = "AudioScenarioAiServer";
  /**
   * 10: AI dialogue scenario, only applicable to interactions with [Agora Conversational AI Engine](https://docs.agora.io/en/conversational-ai/overview/product-overview).
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioAiClient"] = 10] = "AudioScenarioAiClient";
  /**
   * Number of enum values.
   */
  AudioScenarioType[AudioScenarioType["AudioScenarioNum"] = 11] = "AudioScenarioNum";
  return AudioScenarioType;
}({});

/**
 * Video frame format.
 */
export class VideoFormat {}

/**
 * Content type of screen sharing.
 */
export let VideoContentHint = /*#__PURE__*/function (VideoContentHint) {
  /**
   * (Default) No specified content type.
   */
  VideoContentHint[VideoContentHint["ContentHintNone"] = 0] = "ContentHintNone";
  /**
   * Content type is motion. Recommended when sharing videos, movies, or video games.
   */
  VideoContentHint[VideoContentHint["ContentHintMotion"] = 1] = "ContentHintMotion";
  /**
   * Content type is details. Recommended when sharing images or text.
   */
  VideoContentHint[VideoContentHint["ContentHintDetails"] = 2] = "ContentHintDetails";
  return VideoContentHint;
}({});

/**
 * Screen sharing scenarios.
 */
export let ScreenScenarioType = /*#__PURE__*/function (ScreenScenarioType) {
  /**
   * 1: (Default) Document. In this scenario, the quality of the shared content is prioritized, and the latency for the receiver to see the shared video is reduced. You can use this scenario when sharing documents, slides, or spreadsheets.
   */
  ScreenScenarioType[ScreenScenarioType["ScreenScenarioDocument"] = 1] = "ScreenScenarioDocument";
  /**
   * 2: Gaming. In this scenario, the smoothness of the shared content is prioritized. You can use this scenario when sharing games.
   */
  ScreenScenarioType[ScreenScenarioType["ScreenScenarioGaming"] = 2] = "ScreenScenarioGaming";
  /**
   * 3: Video. In this scenario, the smoothness of the shared content is prioritized. You can use this scenario when sharing movies or live videos.
   */
  ScreenScenarioType[ScreenScenarioType["ScreenScenarioVideo"] = 3] = "ScreenScenarioVideo";
  /**
   * 4: Remote control. In this scenario, the quality of the shared content is prioritized, and the latency for the receiver to see the shared video is reduced. You can use this scenario when sharing the desktop of a remotely controlled device.
   */
  ScreenScenarioType[ScreenScenarioType["ScreenScenarioRdc"] = 4] = "ScreenScenarioRdc";
  return ScreenScenarioType;
}({});

/**
 * Video application scenario type.
 */
export let VideoApplicationScenarioType = /*#__PURE__*/function (VideoApplicationScenarioType) {
  /**
   * 0: (Default) General scenario.
   */
  VideoApplicationScenarioType[VideoApplicationScenarioType["ApplicationScenarioGeneral"] = 0] = "ApplicationScenarioGeneral";
  /**
   * 1: Meeting scenario.
   */
  VideoApplicationScenarioType[VideoApplicationScenarioType["ApplicationScenarioMeeting"] = 1] = "ApplicationScenarioMeeting";
  /**
   * 2: 1v1 video call
   */
  VideoApplicationScenarioType[VideoApplicationScenarioType["ApplicationScenario1v1"] = 2] = "ApplicationScenario1v1";
  /**
   * 3: Live show
   */
  VideoApplicationScenarioType[VideoApplicationScenarioType["ApplicationScenarioLiveshow"] = 3] = "ApplicationScenarioLiveshow";
  return VideoApplicationScenarioType;
}({});

/**
 * @ignore
 */
export let VideoQoePreferenceType = /*#__PURE__*/function (VideoQoePreferenceType) {
  /**
   * @ignore
   */
  VideoQoePreferenceType[VideoQoePreferenceType["VideoQoePreferenceBalance"] = 1] = "VideoQoePreferenceBalance";
  /**
   * @ignore
   */
  VideoQoePreferenceType[VideoQoePreferenceType["VideoQoePreferenceDelayFirst"] = 2] = "VideoQoePreferenceDelayFirst";
  /**
   * @ignore
   */
  VideoQoePreferenceType[VideoQoePreferenceType["VideoQoePreferencePictureQualityFirst"] = 3] = "VideoQoePreferencePictureQualityFirst";
  /**
   * @ignore
   */
  VideoQoePreferenceType[VideoQoePreferenceType["VideoQoePreferenceFluencyFirst"] = 4] = "VideoQoePreferenceFluencyFirst";
  return VideoQoePreferenceType;
}({});

/**
 * Brightness level of the locally captured video.
 */
export let CaptureBrightnessLevelType = /*#__PURE__*/function (CaptureBrightnessLevelType) {
  /**
   * @ignore
   */
  CaptureBrightnessLevelType[CaptureBrightnessLevelType["CaptureBrightnessLevelInvalid"] = -1] = "CaptureBrightnessLevelInvalid";
  /**
   * @ignore
   */
  CaptureBrightnessLevelType[CaptureBrightnessLevelType["CaptureBrightnessLevelNormal"] = 0] = "CaptureBrightnessLevelNormal";
  /**
   * @ignore
   */
  CaptureBrightnessLevelType[CaptureBrightnessLevelType["CaptureBrightnessLevelBright"] = 1] = "CaptureBrightnessLevelBright";
  /**
   * @ignore
   */
  CaptureBrightnessLevelType[CaptureBrightnessLevelType["CaptureBrightnessLevelDark"] = 2] = "CaptureBrightnessLevelDark";
  return CaptureBrightnessLevelType;
}({});

/**
 * Camera stabilization mode.
 *
 * Camera stabilization effect increases in the order of 1 < 2 < 3, with corresponding increase in latency.
 */
export let CameraStabilizationMode = /*#__PURE__*/function (CameraStabilizationMode) {
  /**
   * -1: (Default) Camera stabilization is off.
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeOff"] = -1] = "CameraStabilizationModeOff";
  /**
   * 0: Camera auto stabilization. The system automatically selects a stabilization mode based on the camera state. However, this mode has higher latency, and it is recommended not to use this enumeration.
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeAuto"] = 0] = "CameraStabilizationModeAuto";
  /**
   * 1: (Recommended) Camera stabilization level 1.
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeLevel1"] = 1] = "CameraStabilizationModeLevel1";
  /**
   * 2: Camera stabilization level 2.
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeLevel2"] = 2] = "CameraStabilizationModeLevel2";
  /**
   * 3: Camera stabilization level 3.
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeLevel3"] = 3] = "CameraStabilizationModeLevel3";
  /**
   * @ignore
   */
  CameraStabilizationMode[CameraStabilizationMode["CameraStabilizationModeMaxLevel"] = 3] = "CameraStabilizationModeMaxLevel";
  return CameraStabilizationMode;
}({});

/**
 * Local audio state.
 */
export let LocalAudioStreamState = /*#__PURE__*/function (LocalAudioStreamState) {
  /**
   * 0: Default initial state of local audio.
   */
  LocalAudioStreamState[LocalAudioStreamState["LocalAudioStreamStateStopped"] = 0] = "LocalAudioStreamStateStopped";
  /**
   * 1: Local audio capture device started successfully.
   */
  LocalAudioStreamState[LocalAudioStreamState["LocalAudioStreamStateRecording"] = 1] = "LocalAudioStreamStateRecording";
  /**
   * 2: First frame of local audio encoded successfully.
   */
  LocalAudioStreamState[LocalAudioStreamState["LocalAudioStreamStateEncoding"] = 2] = "LocalAudioStreamStateEncoding";
  /**
   * 3: Failed to start local audio.
   */
  LocalAudioStreamState[LocalAudioStreamState["LocalAudioStreamStateFailed"] = 3] = "LocalAudioStreamStateFailed";
  return LocalAudioStreamState;
}({});

/**
 * Reason for local audio state change.
 */
export let LocalAudioStreamReason = /*#__PURE__*/function (LocalAudioStreamReason) {
  /**
   * 0: Local audio state is normal.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonOk"] = 0] = "LocalAudioStreamReasonOk";
  /**
   * 1: The cause of the local audio error is unclear. It is recommended to prompt the user to try rejoining the channel.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonFailure"] = 1] = "LocalAudioStreamReasonFailure";
  /**
   * 2: No permission to start the local audio capture device. Prompt the user to enable the permission. Deprecated: This enumeration is deprecated. Use the onPermissionError callback with RecordAudio instead.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonDeviceNoPermission"] = 2] = "LocalAudioStreamReasonDeviceNoPermission";
  /**
   * 3: The local audio capture device is already in use. Prompt the user to check whether the microphone is occupied by another app. Local audio capture will automatically resume about 5 seconds after the microphone becomes idle. You can also try rejoining the channel after the microphone is idle.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonDeviceBusy"] = 3] = "LocalAudioStreamReasonDeviceBusy";
  /**
   * 4: Local audio capture failed.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonRecordFailure"] = 4] = "LocalAudioStreamReasonRecordFailure";
  /**
   * 5: Local audio encoding failed.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonEncodeFailure"] = 5] = "LocalAudioStreamReasonEncodeFailure";
  /**
   * @ignore
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonNoRecordingDevice"] = 6] = "LocalAudioStreamReasonNoRecordingDevice";
  /**
   * @ignore
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonNoPlayoutDevice"] = 7] = "LocalAudioStreamReasonNoPlayoutDevice";
  /**
   * 8: Local audio capture was interrupted by system phone call, voice assistant, or alarm. To resume local audio capture, ask the user to end the call, voice assistant, or alarm.
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonInterrupted"] = 8] = "LocalAudioStreamReasonInterrupted";
  /**
   * @ignore
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonRecordInvalidId"] = 9] = "LocalAudioStreamReasonRecordInvalidId";
  /**
   * @ignore
   */
  LocalAudioStreamReason[LocalAudioStreamReason["LocalAudioStreamReasonPlayoutInvalidId"] = 10] = "LocalAudioStreamReasonPlayoutInvalidId";
  return LocalAudioStreamReason;
}({});

/**
 * Local video state.
 */
export let LocalVideoStreamState = /*#__PURE__*/function (LocalVideoStreamState) {
  /**
   * 0: Default initial state of local video.
   */
  LocalVideoStreamState[LocalVideoStreamState["LocalVideoStreamStateStopped"] = 0] = "LocalVideoStreamStateStopped";
  /**
   * 1: Local video capture device started successfully.
   */
  LocalVideoStreamState[LocalVideoStreamState["LocalVideoStreamStateCapturing"] = 1] = "LocalVideoStreamStateCapturing";
  /**
   * 2: First frame of local video encoded successfully.
   */
  LocalVideoStreamState[LocalVideoStreamState["LocalVideoStreamStateEncoding"] = 2] = "LocalVideoStreamStateEncoding";
  /**
   * 3: Failed to start local video.
   */
  LocalVideoStreamState[LocalVideoStreamState["LocalVideoStreamStateFailed"] = 3] = "LocalVideoStreamStateFailed";
  return LocalVideoStreamState;
}({});

/**
 * Local video event types.
 *
 * Since Available since v4.6.1.
 */
export let LocalVideoEventType = /*#__PURE__*/function (LocalVideoEventType) {
  /**
   * (1): The screen capture window is hidden. (Android only)
   */
  LocalVideoEventType[LocalVideoEventType["LocalVideoEventTypeScreenCaptureWindowHidden"] = 1] = "LocalVideoEventTypeScreenCaptureWindowHidden";
  /**
   * (2): The screen capture window recovers from being hidden. (Android only)
   */
  LocalVideoEventType[LocalVideoEventType["LocalVideoEventTypeScreenCaptureWindowRecoverFromHidden"] = 2] = "LocalVideoEventTypeScreenCaptureWindowRecoverFromHidden";
  /**
   * (3): The screen capture is stopped by the user. (Android only)
   */
  LocalVideoEventType[LocalVideoEventType["LocalVideoEventTypeScreenCaptureStoppedByUser"] = 3] = "LocalVideoEventTypeScreenCaptureStoppedByUser";
  /**
   * (4): A system internal error occurs during screen capture. (Android only)
   */
  LocalVideoEventType[LocalVideoEventType["LocalVideoEventTypeScreenCaptureSystemInternalError"] = 4] = "LocalVideoEventTypeScreenCaptureSystemInternalError";
  return LocalVideoEventType;
}({});

/**
 * Reason for local video state change.
 */
export let LocalVideoStreamReason = /*#__PURE__*/function (LocalVideoStreamReason) {
  /**
   * 0: Local video is in a normal state.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonOk"] = 0] = "LocalVideoStreamReasonOk";
  /**
   * 1: Unknown error.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonFailure"] = 1] = "LocalVideoStreamReasonFailure";
  /**
   * 2: No permission to start the local video capture device. Prompt the user to enable device permissions before rejoining the channel. Deprecated: This enumeration is deprecated. Use onPermissionError callback with Camera instead.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceNoPermission"] = 2] = "LocalVideoStreamReasonDeviceNoPermission";
  /**
   * 3: The local video capture device is in use. Prompt the user to check if the camera is occupied by another app or try rejoining the channel.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceBusy"] = 3] = "LocalVideoStreamReasonDeviceBusy";
  /**
   * 4: Failed to capture local video. Prompt the user to check whether the video capture device is working properly, whether the camera is occupied by another app, or try rejoining the channel.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonCaptureFailure"] = 4] = "LocalVideoStreamReasonCaptureFailure";
  /**
   * 5: Failed to encode local video.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonCodecNotSupport"] = 5] = "LocalVideoStreamReasonCodecNotSupport";
  /**
   * 6: (iOS only) The app is in the background. Prompt the user that video capture is not available when the app is in the background.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonCaptureInbackground"] = 6] = "LocalVideoStreamReasonCaptureInbackground";
  /**
   * 7: (iOS only) The app is in slide-over, split-view, or picture-in-picture mode and another app is using the camera. Prompt the user that video capture is not available under these conditions.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonCaptureMultipleForegroundApps"] = 7] = "LocalVideoStreamReasonCaptureMultipleForegroundApps";
  /**
   * 8: No local video capture device found. Check if the camera is properly connected and functioning, or try rejoining the channel.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceNotFound"] = 8] = "LocalVideoStreamReasonDeviceNotFound";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceDisconnected"] = 9] = "LocalVideoStreamReasonDeviceDisconnected";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceInvalidId"] = 10] = "LocalVideoStreamReasonDeviceInvalidId";
  /**
   * 14: (Android only) Video capture is interrupted. Possible reasons:
   *  The camera is occupied by another app. Prompt the user to check if the camera is occupied.
   *  The app has been switched to the background. You can use a foreground service notification to inform the OS to continue capturing video in the background..
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceInterrupt"] = 14] = "LocalVideoStreamReasonDeviceInterrupt";
  /**
   * 15: (Android only) Video capture device error. Prompt the user to turn the camera off and on again to restore functionality. If that doesn't work, check for hardware failure.
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceFatalError"] = 15] = "LocalVideoStreamReasonDeviceFatalError";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonDeviceSystemPressure"] = 101] = "LocalVideoStreamReasonDeviceSystemPressure";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowMinimized"] = 11] = "LocalVideoStreamReasonScreenCaptureWindowMinimized";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowClosed"] = 12] = "LocalVideoStreamReasonScreenCaptureWindowClosed";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowOccluded"] = 13] = "LocalVideoStreamReasonScreenCaptureWindowOccluded";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowNotSupported"] = 20] = "LocalVideoStreamReasonScreenCaptureWindowNotSupported";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureFailure"] = 21] = "LocalVideoStreamReasonScreenCaptureFailure";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureNoPermission"] = 22] = "LocalVideoStreamReasonScreenCaptureNoPermission";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureAutoFallback"] = 24] = "LocalVideoStreamReasonScreenCaptureAutoFallback";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowHidden"] = 25] = "LocalVideoStreamReasonScreenCaptureWindowHidden";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowRecoverFromHidden"] = 26] = "LocalVideoStreamReasonScreenCaptureWindowRecoverFromHidden";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureWindowRecoverFromMinimized"] = 27] = "LocalVideoStreamReasonScreenCaptureWindowRecoverFromMinimized";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCapturePaused"] = 28] = "LocalVideoStreamReasonScreenCapturePaused";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureResumed"] = 29] = "LocalVideoStreamReasonScreenCaptureResumed";
  /**
   * @ignore
   */
  LocalVideoStreamReason[LocalVideoStreamReason["LocalVideoStreamReasonScreenCaptureDisplayDisconnected"] = 30] = "LocalVideoStreamReasonScreenCaptureDisplayDisconnected";
  return LocalVideoStreamReason;
}({});

/**
 * Remote audio stream state.
 */
export let RemoteAudioState = /*#__PURE__*/function (RemoteAudioState) {
  /**
   * 0: Default initial state of remote audio. This state is reported under RemoteAudioReasonLocalMuted, RemoteAudioReasonRemoteMuted, or RemoteAudioReasonRemoteOffline.
   */
  RemoteAudioState[RemoteAudioState["RemoteAudioStateStopped"] = 0] = "RemoteAudioStateStopped";
  /**
   * 1: Local user has received the first packet of remote audio.
   */
  RemoteAudioState[RemoteAudioState["RemoteAudioStateStarting"] = 1] = "RemoteAudioStateStarting";
  /**
   * 2: Remote audio stream is decoding and playing normally. This state is reported under RemoteAudioReasonNetworkRecovery, RemoteAudioReasonLocalUnmuted, or RemoteAudioReasonRemoteUnmuted.
   */
  RemoteAudioState[RemoteAudioState["RemoteAudioStateDecoding"] = 2] = "RemoteAudioStateDecoding";
  /**
   * 3: Remote audio stream is frozen. This state is reported under RemoteAudioReasonNetworkCongestion.
   */
  RemoteAudioState[RemoteAudioState["RemoteAudioStateFrozen"] = 3] = "RemoteAudioStateFrozen";
  /**
   * 4: Failed to play remote audio stream. This state is reported under RemoteAudioReasonInternal.
   */
  RemoteAudioState[RemoteAudioState["RemoteAudioStateFailed"] = 4] = "RemoteAudioStateFailed";
  return RemoteAudioState;
}({});

/**
 * Reasons for remote audio stream state changes.
 */
export let RemoteAudioStateReason = /*#__PURE__*/function (RemoteAudioStateReason) {
  /**
   * 0: Reports this reason when the audio state changes.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonInternal"] = 0] = "RemoteAudioReasonInternal";
  /**
   * 1: Network congestion.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonNetworkCongestion"] = 1] = "RemoteAudioReasonNetworkCongestion";
  /**
   * 2: Network recovered.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonNetworkRecovery"] = 2] = "RemoteAudioReasonNetworkRecovery";
  /**
   * 3: Local user stopped receiving remote audio stream or disabled the audio module.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonLocalMuted"] = 3] = "RemoteAudioReasonLocalMuted";
  /**
   * 4: Local user resumed receiving remote audio stream or enabled the audio module.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonLocalUnmuted"] = 4] = "RemoteAudioReasonLocalUnmuted";
  /**
   * 5: Remote user stopped sending audio stream or disabled the audio module.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonRemoteMuted"] = 5] = "RemoteAudioReasonRemoteMuted";
  /**
   * 6: Remote user resumed sending audio stream or enabled the audio module.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonRemoteUnmuted"] = 6] = "RemoteAudioReasonRemoteUnmuted";
  /**
   * 7: Remote user left the channel.
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonRemoteOffline"] = 7] = "RemoteAudioReasonRemoteOffline";
  /**
   * @ignore
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonNoPacketReceive"] = 8] = "RemoteAudioReasonNoPacketReceive";
  /**
   * @ignore
   */
  RemoteAudioStateReason[RemoteAudioStateReason["RemoteAudioReasonLocalPlayFailed"] = 9] = "RemoteAudioReasonLocalPlayFailed";
  return RemoteAudioStateReason;
}({});

/**
 * Remote video stream state.
 */
export let RemoteVideoState = /*#__PURE__*/function (RemoteVideoState) {
  /**
   * 0: The default initial state of the remote video. This state is reported under RemoteVideoStateReasonLocalMuted, RemoteVideoStateReasonRemoteMuted, or RemoteVideoStateReasonRemoteOffline.
   */
  RemoteVideoState[RemoteVideoState["RemoteVideoStateStopped"] = 0] = "RemoteVideoStateStopped";
  /**
   * 1: The local user has received the first packet of the remote video.
   */
  RemoteVideoState[RemoteVideoState["RemoteVideoStateStarting"] = 1] = "RemoteVideoStateStarting";
  /**
   * 2: The remote video stream is being decoded and playing normally. This state is reported under RemoteVideoStateReasonNetworkRecovery, RemoteVideoStateReasonLocalUnmuted, or RemoteVideoStateReasonRemoteUnmuted.
   */
  RemoteVideoState[RemoteVideoState["RemoteVideoStateDecoding"] = 2] = "RemoteVideoStateDecoding";
  /**
   * 3: The remote video stream is frozen. This state is reported under RemoteVideoStateReasonNetworkCongestion.
   */
  RemoteVideoState[RemoteVideoState["RemoteVideoStateFrozen"] = 3] = "RemoteVideoStateFrozen";
  /**
   * 4: Failed to play the remote video stream. This state is reported under RemoteVideoStateReasonInternal.
   */
  RemoteVideoState[RemoteVideoState["RemoteVideoStateFailed"] = 4] = "RemoteVideoStateFailed";
  return RemoteVideoState;
}({});

/**
 * Reasons for remote video stream state changes.
 */
export let RemoteVideoStateReason = /*#__PURE__*/function (RemoteVideoStateReason) {
  /**
   * 0: Reports this reason when the video state changes.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonInternal"] = 0] = "RemoteVideoStateReasonInternal";
  /**
   * 1: Network congestion.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonNetworkCongestion"] = 1] = "RemoteVideoStateReasonNetworkCongestion";
  /**
   * 2: Network recovered.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonNetworkRecovery"] = 2] = "RemoteVideoStateReasonNetworkRecovery";
  /**
   * 3: Local user stopped receiving remote video stream or disabled the video module.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonLocalMuted"] = 3] = "RemoteVideoStateReasonLocalMuted";
  /**
   * 4: Local user resumed receiving remote video stream or enabled the video module.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonLocalUnmuted"] = 4] = "RemoteVideoStateReasonLocalUnmuted";
  /**
   * 5: Remote user stopped sending video stream or disabled the video module.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonRemoteMuted"] = 5] = "RemoteVideoStateReasonRemoteMuted";
  /**
   * 6: Remote user resumed sending video stream or enabled the video module.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonRemoteUnmuted"] = 6] = "RemoteVideoStateReasonRemoteUnmuted";
  /**
   * 7: Remote user left the channel.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonRemoteOffline"] = 7] = "RemoteVideoStateReasonRemoteOffline";
  /**
   * 8: Under poor network conditions, remote audio/video stream falls back to audio only.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonAudioFallback"] = 8] = "RemoteVideoStateReasonAudioFallback";
  /**
   * 9: When the network improves, remote audio stream recovers to audio/video stream.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonAudioFallbackRecovery"] = 9] = "RemoteVideoStateReasonAudioFallbackRecovery";
  /**
   * @ignore
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonVideoStreamTypeChangeToLow"] = 10] = "RemoteVideoStateReasonVideoStreamTypeChangeToLow";
  /**
   * @ignore
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonVideoStreamTypeChangeToHigh"] = 11] = "RemoteVideoStateReasonVideoStreamTypeChangeToHigh";
  /**
   * 12: (iOS only) The remote user's app has switched to the background.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonSdkInBackground"] = 12] = "RemoteVideoStateReasonSdkInBackground";
  /**
   * 13: The local video decoder does not support decoding the received remote video stream.
   */
  RemoteVideoStateReason[RemoteVideoStateReason["RemoteVideoStateReasonCodecNotSupport"] = 13] = "RemoteVideoStateReasonCodecNotSupport";
  return RemoteVideoStateReason;
}({});

/**
 * @ignore
 */
export let RemoteUserState = /*#__PURE__*/function (RemoteUserState) {
  /**
   * @ignore
   */
  RemoteUserState[RemoteUserState["UserStateMuteAudio"] = 1] = "UserStateMuteAudio";
  /**
   * @ignore
   */
  RemoteUserState[RemoteUserState["UserStateMuteVideo"] = 2] = "UserStateMuteVideo";
  /**
   * @ignore
   */
  RemoteUserState[RemoteUserState["UserStateEnableVideo"] = 16] = "UserStateEnableVideo";
  /**
   * @ignore
   */
  RemoteUserState[RemoteUserState["UserStateEnableLocalVideo"] = 256] = "UserStateEnableLocalVideo";
  return RemoteUserState;
}({});

/**
 * @ignore
 */
export class VideoTrackInfo {}

/**
 * @ignore
 */
export let RemoteVideoDownscaleLevel = /*#__PURE__*/function (RemoteVideoDownscaleLevel) {
  /**
   * @ignore
   */
  RemoteVideoDownscaleLevel[RemoteVideoDownscaleLevel["RemoteVideoDownscaleLevelNone"] = 0] = "RemoteVideoDownscaleLevelNone";
  /**
   * @ignore
   */
  RemoteVideoDownscaleLevel[RemoteVideoDownscaleLevel["RemoteVideoDownscaleLevel1"] = 1] = "RemoteVideoDownscaleLevel1";
  /**
   * @ignore
   */
  RemoteVideoDownscaleLevel[RemoteVideoDownscaleLevel["RemoteVideoDownscaleLevel2"] = 2] = "RemoteVideoDownscaleLevel2";
  /**
   * @ignore
   */
  RemoteVideoDownscaleLevel[RemoteVideoDownscaleLevel["RemoteVideoDownscaleLevel3"] = 3] = "RemoteVideoDownscaleLevel3";
  /**
   * @ignore
   */
  RemoteVideoDownscaleLevel[RemoteVideoDownscaleLevel["RemoteVideoDownscaleLevel4"] = 4] = "RemoteVideoDownscaleLevel4";
  return RemoteVideoDownscaleLevel;
}({});

/**
 * User volume information.
 */
export class AudioVolumeInfo {}

/**
 * Audio device information.
 *
 * (Android only)
 */
export class DeviceInfo {}

/**
 * @ignore
 */
export class Packet {}

/**
 * Sample rate of audio output for streaming.
 */
export let AudioSampleRateType = /*#__PURE__*/function (AudioSampleRateType) {
  /**
   * 32000: 32 kHz
   */
  AudioSampleRateType[AudioSampleRateType["AudioSampleRate32000"] = 32000] = "AudioSampleRate32000";
  /**
   * 44100: 44.1 kHz
   */
  AudioSampleRateType[AudioSampleRateType["AudioSampleRate44100"] = 44100] = "AudioSampleRate44100";
  /**
   * 48000: (Default) 48 kHz
   */
  AudioSampleRateType[AudioSampleRateType["AudioSampleRate48000"] = 48000] = "AudioSampleRate48000";
  return AudioSampleRateType;
}({});

/**
 * Codec type for transcoded output video stream.
 */
export let VideoCodecTypeForStream = /*#__PURE__*/function (VideoCodecTypeForStream) {
  /**
   * 1: (Default) H.264.
   */
  VideoCodecTypeForStream[VideoCodecTypeForStream["VideoCodecH264ForStream"] = 1] = "VideoCodecH264ForStream";
  /**
   * 2: H.265.
   */
  VideoCodecTypeForStream[VideoCodecTypeForStream["VideoCodecH265ForStream"] = 2] = "VideoCodecH265ForStream";
  return VideoCodecTypeForStream;
}({});

/**
 * Codec profile for video in relay streaming output.
 */
export let VideoCodecProfileType = /*#__PURE__*/function (VideoCodecProfileType) {
  /**
   * 66: Baseline profile, typically used in low-end or error-tolerant applications such as video calls, mobile videos, etc.
   */
  VideoCodecProfileType[VideoCodecProfileType["VideoCodecProfileBaseline"] = 66] = "VideoCodecProfileBaseline";
  /**
   * 77: Main profile, typically used in mainstream consumer electronics such as MP4, portable video players, PSP, iPad, etc.
   */
  VideoCodecProfileType[VideoCodecProfileType["VideoCodecProfileMain"] = 77] = "VideoCodecProfileMain";
  /**
   * 100: (Default) High profile, typically used in broadcasting, video disc storage, and HDTV.
   */
  VideoCodecProfileType[VideoCodecProfileType["VideoCodecProfileHigh"] = 100] = "VideoCodecProfileHigh";
  return VideoCodecProfileType;
}({});

/**
 * Audio codec specification for stream publishing output. Defaults to LC-AAC.
 */
export let AudioCodecProfileType = /*#__PURE__*/function (AudioCodecProfileType) {
  /**
   * 0: (Default) LC-AAC specification.
   */
  AudioCodecProfileType[AudioCodecProfileType["AudioCodecProfileLcAac"] = 0] = "AudioCodecProfileLcAac";
  /**
   * 1: HE-AAC specification.
   */
  AudioCodecProfileType[AudioCodecProfileType["AudioCodecProfileHeAac"] = 1] = "AudioCodecProfileHeAac";
  /**
   * 2: HE-AAC v2 specification.
   */
  AudioCodecProfileType[AudioCodecProfileType["AudioCodecProfileHeAacV2"] = 2] = "AudioCodecProfileHeAacV2";
  return AudioCodecProfileType;
}({});

/**
 * Local audio statistics.
 */
export class LocalAudioStats {}

/**
 * Stream publishing state.
 */
export let RtmpStreamPublishState = /*#__PURE__*/function (RtmpStreamPublishState) {
  /**
   * 0: Streaming has not started or has ended.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateIdle"] = 0] = "RtmpStreamPublishStateIdle";
  /**
   * 1: Connecting to the streaming and CDN servers.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateConnecting"] = 1] = "RtmpStreamPublishStateConnecting";
  /**
   * 2: Streaming is ongoing. This state is returned after successful publishing.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateRunning"] = 2] = "RtmpStreamPublishStateRunning";
  /**
   * 3: Recovering the stream. When the CDN encounters an issue or the stream is briefly interrupted, the SDK automatically attempts to recover the stream and returns this state.
   *  If recovery is successful, it transitions to RtmpStreamPublishStateRunning(2).
   *  If the server fails or recovery is unsuccessful within 60 seconds, it transitions to RtmpStreamPublishStateFailure(4). If 60 seconds is too long, you can also manually attempt to reconnect.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateRecovering"] = 3] = "RtmpStreamPublishStateRecovering";
  /**
   * 4: Streaming failed. After failure, you can troubleshoot using the returned error code.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateFailure"] = 4] = "RtmpStreamPublishStateFailure";
  /**
   * 5: The SDK is disconnecting from the streaming and CDN servers. When you call stopRtmpStream to end the stream normally, the SDK reports the states RtmpStreamPublishStateDisconnecting and RtmpStreamPublishStateIdle in sequence.
   */
  RtmpStreamPublishState[RtmpStreamPublishState["RtmpStreamPublishStateDisconnecting"] = 5] = "RtmpStreamPublishStateDisconnecting";
  return RtmpStreamPublishState;
}({});

/**
 * Reason for stream publishing state change.
 */
export let RtmpStreamPublishReason = /*#__PURE__*/function (RtmpStreamPublishReason) {
  /**
   * 0: Stream published successfully.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonOk"] = 0] = "RtmpStreamPublishReasonOk";
  /**
   * 1: Invalid parameter. Please check whether the input parameters are correct.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonInvalidArgument"] = 1] = "RtmpStreamPublishReasonInvalidArgument";
  /**
   * 2: The stream is encrypted and cannot be published.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonEncryptedStreamNotAllowed"] = 2] = "RtmpStreamPublishReasonEncryptedStreamNotAllowed";
  /**
   * 3: Stream publishing timed out.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonConnectionTimeout"] = 3] = "RtmpStreamPublishReasonConnectionTimeout";
  /**
   * 4: An error occurred on the streaming server.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonInternalServerError"] = 4] = "RtmpStreamPublishReasonInternalServerError";
  /**
   * 5: An error occurred on the CDN server.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonRtmpServerError"] = 5] = "RtmpStreamPublishReasonRtmpServerError";
  /**
   * 6: Stream publishing requests are too frequent.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonTooOften"] = 6] = "RtmpStreamPublishReasonTooOften";
  /**
   * 7: The number of stream URLs for a single host has reached the limit of 10. Please delete some unused stream URLs before adding new ones.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonReachLimit"] = 7] = "RtmpStreamPublishReasonReachLimit";
  /**
   * 8: The host is operating on a stream that does not belong to them, such as updating or stopping another host's stream. Please check your app logic.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonNotAuthorized"] = 8] = "RtmpStreamPublishReasonNotAuthorized";
  /**
   * 9: The server did not find the stream.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonStreamNotFound"] = 9] = "RtmpStreamPublishReasonStreamNotFound";
  /**
   * 10: The stream URL format is incorrect. Please verify the format of the stream URL.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonFormatNotSupported"] = 10] = "RtmpStreamPublishReasonFormatNotSupported";
  /**
   * 11: The user is not a broadcaster and cannot use the streaming feature. Please check your application logic.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonNotBroadcaster"] = 11] = "RtmpStreamPublishReasonNotBroadcaster";
  /**
   * 13: Called updateRtmpTranscoding to update transcoding properties without enabling transcoding. Please check your application logic.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonTranscodingNoMixStream"] = 13] = "RtmpStreamPublishReasonTranscodingNoMixStream";
  /**
   * 14: The host's network encountered an error.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonNetDown"] = 14] = "RtmpStreamPublishReasonNetDown";
  /**
   * @ignore
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonInvalidAppid"] = 15] = "RtmpStreamPublishReasonInvalidAppid";
  /**
   * 16: Your project does not have permission to use the streaming service.
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamPublishReasonInvalidPrivilege"] = 16] = "RtmpStreamPublishReasonInvalidPrivilege";
  /**
   * @ignore
   */
  RtmpStreamPublishReason[RtmpStreamPublishReason["RtmpStreamUnpublishReasonOk"] = 100] = "RtmpStreamUnpublishReasonOk";
  return RtmpStreamPublishReason;
}({});

/**
 * Events that occur during CDN live streaming.
 */
export let RtmpStreamingEvent = /*#__PURE__*/function (RtmpStreamingEvent) {
  /**
   * 1: Failed to add background image or watermark during CDN live streaming.
   */
  RtmpStreamingEvent[RtmpStreamingEvent["RtmpStreamingEventFailedLoadImage"] = 1] = "RtmpStreamingEventFailedLoadImage";
  /**
   * 2: The stream URL is already in use. If you want to start a new stream, please use a new stream URL.
   */
  RtmpStreamingEvent[RtmpStreamingEvent["RtmpStreamingEventUrlAlreadyInUse"] = 2] = "RtmpStreamingEventUrlAlreadyInUse";
  /**
   * 3: Feature not supported.
   */
  RtmpStreamingEvent[RtmpStreamingEvent["RtmpStreamingEventAdvancedFeatureNotSupport"] = 3] = "RtmpStreamingEventAdvancedFeatureNotSupport";
  /**
   * 4: Reserved parameter.
   */
  RtmpStreamingEvent[RtmpStreamingEvent["RtmpStreamingEventRequestTooOften"] = 4] = "RtmpStreamingEventRequestTooOften";
  return RtmpStreamingEvent;
}({});

/**
 * Image properties.
 *
 * Used to set the watermark and background image properties for live video.
 */
export class RtcImage {}

/**
 * Advanced feature configuration for live transcoding.
 *
 * To use advanced features for live transcoding, please [contact sales](mailto:support@agora.io).
 */
export class LiveStreamAdvancedFeature {}

/**
 * Network connection state.
 */
export let ConnectionStateType = /*#__PURE__*/function (ConnectionStateType) {
  /**
   * 1: Network connection is disconnected. This state indicates that the SDK is in:
   *  The initialization phase before calling joinChannel to join a channel.
   *  Or the phase after calling leaveChannel to leave a channel.
   */
  ConnectionStateType[ConnectionStateType["ConnectionStateDisconnected"] = 1] = "ConnectionStateDisconnected";
  /**
   * 2: Connecting to the network. This state indicates that the SDK is establishing a connection with the specified channel after calling joinChannel.
   *  If the channel is joined successfully, the app receives the onConnectionStateChanged callback, indicating the network state has changed to ConnectionStateConnected.
   *  After the connection is established, the SDK initializes media and triggers the onJoinChannelSuccess callback once everything is ready.
   */
  ConnectionStateType[ConnectionStateType["ConnectionStateConnecting"] = 2] = "ConnectionStateConnecting";
  /**
   * 3: Network is connected. This state indicates that the user has joined the channel and can publish or subscribe to media streams. If the connection is interrupted due to network disconnection or switching, the SDK automatically reconnects. The app receives the onConnectionStateChanged callback, indicating the network state has changed to ConnectionStateReconnecting.
   */
  ConnectionStateType[ConnectionStateType["ConnectionStateConnected"] = 3] = "ConnectionStateConnected";
  /**
   * 4: Reconnecting to the network. This state indicates that the SDK had previously joined a channel but the connection was interrupted due to network issues. The SDK is now trying to reconnect to the channel.
   *  If the SDK cannot rejoin the channel within 10 seconds, onConnectionLost is triggered. The SDK remains in the ConnectionStateReconnecting state and continues trying to rejoin.
   *  If the SDK still cannot rejoin the channel within 20 minutes after disconnection, the app receives the onConnectionStateChanged callback, indicating that the network state has changed to ConnectionStateFailed, and the SDK stops trying to reconnect.
   */
  ConnectionStateType[ConnectionStateType["ConnectionStateReconnecting"] = 4] = "ConnectionStateReconnecting";
  /**
   * 5: Network connection failed. This state indicates that the SDK has stopped trying to rejoin the channel and you need to call leaveChannel to leave the channel.
   *  If the user wants to rejoin the channel, they need to call joinChannel again.
   *  If the SDK is prohibited from joining the channel by the server using RESTful API, the app will receive onConnectionStateChanged.
   */
  ConnectionStateType[ConnectionStateType["ConnectionStateFailed"] = 5] = "ConnectionStateFailed";
  return ConnectionStateType;
}({});

/**
 * Settings for each host participating in the transcoding mix.
 */
export class TranscodingUser {}

/**
 * Transcoding properties for CDN streaming.
 */
export class LiveTranscoding {}

/**
 * Video stream participating in local composition.
 */
export class TranscodingVideoStream {}

/**
 * Local video mixing configuration.
 */
export class LocalTranscoderConfiguration {}

/**
 * Local video composition error codes.
 */
export let VideoTranscoderError = /*#__PURE__*/function (VideoTranscoderError) {
  /**
   * 1: The specified video source has not started video capture. You need to create a video track and start video capture for it.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrVideoSourceNotReady"] = 1] = "VtErrVideoSourceNotReady";
  /**
   * 2: Invalid video source type. You need to re-specify a supported video source type.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrInvalidVideoSourceType"] = 2] = "VtErrInvalidVideoSourceType";
  /**
   * 3: Invalid image path. You need to re-specify a correct image path.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrInvalidImagePath"] = 3] = "VtErrInvalidImagePath";
  /**
   * 4: Invalid image format. Make sure the image format is one of PNG, JPEG, or GIF.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrUnsupportImageFormat"] = 4] = "VtErrUnsupportImageFormat";
  /**
   * 5: Invalid video encoding resolution after composition.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrInvalidLayout"] = 5] = "VtErrInvalidLayout";
  /**
   * 20: Unknown internal error.
   */
  VideoTranscoderError[VideoTranscoderError["VtErrInternal"] = 20] = "VtErrInternal";
  return VideoTranscoderError;
}({});

/**
 * Audio source for local audio mixing.
 */
export class MixedAudioStream {}

/**
 * Local audio mixing configuration.
 */
export class LocalAudioMixerConfiguration {}

/**
 * Last mile network probe configuration.
 */
export class LastmileProbeConfig {}

/**
 * Status of last mile quality probe result.
 */
export let LastmileProbeResultState = /*#__PURE__*/function (LastmileProbeResultState) {
  /**
   * 1: Indicates the last mile quality probe result is complete.
   */
  LastmileProbeResultState[LastmileProbeResultState["LastmileProbeResultComplete"] = 1] = "LastmileProbeResultComplete";
  /**
   * 2: Indicates the last mile quality probe did not perform bandwidth estimation, so the result is incomplete. One possible reason is limited testing resources.
   */
  LastmileProbeResultState[LastmileProbeResultState["LastmileProbeResultIncompleteNoBwe"] = 2] = "LastmileProbeResultIncompleteNoBwe";
  /**
   * 3: Last mile quality probe was not performed. One possible reason is network disconnection.
   */
  LastmileProbeResultState[LastmileProbeResultState["LastmileProbeResultUnavailable"] = 3] = "LastmileProbeResultUnavailable";
  return LastmileProbeResultState;
}({});

/**
 * Last mile network quality probe result for uplink or downlink.
 */
export class LastmileProbeOneWayResult {}

/**
 * Last mile network quality probe result for both uplink and downlink.
 */
export class LastmileProbeResult {}

/**
 * The reason for the change in network connection state.
 */
export let ConnectionChangedReasonType = /*#__PURE__*/function (ConnectionChangedReasonType) {
  /**
   * 0: Connecting to the network.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedConnecting"] = 0] = "ConnectionChangedConnecting";
  /**
   * 1: Successfully joined the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedJoinSuccess"] = 1] = "ConnectionChangedJoinSuccess";
  /**
   * 2: Network connection interrupted.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedInterrupted"] = 2] = "ConnectionChangedInterrupted";
  /**
   * 3: Network connection is banned by the server. For example, this status is returned when the user is kicked out of the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedBannedByServer"] = 3] = "ConnectionChangedBannedByServer";
  /**
   * 4: Failed to join the channel. If the SDK fails to join the channel after trying for 20 minutes, this status is returned and the SDK stops trying to reconnect. Prompt the user to switch networks and try joining the channel again.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedJoinFailed"] = 4] = "ConnectionChangedJoinFailed";
  /**
   * 5: Left the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedLeaveChannel"] = 5] = "ConnectionChangedLeaveChannel";
  /**
   * 6: Invalid App ID. Use a valid App ID to rejoin the channel and ensure the App ID matches the one generated in the Agora Console.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedInvalidAppId"] = 6] = "ConnectionChangedInvalidAppId";
  /**
   * 7: Invalid channel name. Use a valid channel name to rejoin the channel. A valid channel name is a string within 64 bytes. The supported character set includes 89 characters:
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedInvalidChannelName"] = 7] = "ConnectionChangedInvalidChannelName";
  /**
   * 8: Invalid Token. Possible reasons:
   *  Your project has App Certificate enabled, but you joined the channel without using a Token.
   *  The user ID specified in joinChannel does not match the one used to generate the Token.
   *  The generated Token does not match the one used to join the channel. Ensure that:
   *  When App Certificate is enabled, use a Token to join the channel.
   *  The user ID used to generate the Token matches the one used to join the channel.
   *  The generated Token matches the one used to join the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedInvalidToken"] = 8] = "ConnectionChangedInvalidToken";
  /**
   * 9: The current Token has expired. Generate a new Token on your server and use it to rejoin the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedTokenExpired"] = 9] = "ConnectionChangedTokenExpired";
  /**
   * 10: This user is banned by the server. Possible reasons:
   *  The user has already joined the channel and calls the join channel API again, such as joinChannel, which returns this status. Stop calling this method.
   *  The user tries to join a channel during a call test. Wait until the call test ends before joining the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedRejectedByServer"] = 10] = "ConnectionChangedRejectedByServer";
  /**
   * 11: SDK attempts to reconnect due to proxy server settings.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedSettingProxyServer"] = 11] = "ConnectionChangedSettingProxyServer";
  /**
   * 12: Network connection state changed due to Token renewal.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedRenewToken"] = 12] = "ConnectionChangedRenewToken";
  /**
   * 13: Client IP address changed. If this status code is received multiple times, prompt the user to switch networks and try rejoining the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedClientIpAddressChanged"] = 13] = "ConnectionChangedClientIpAddressChanged";
  /**
   * 14: Connection keep-alive timeout between SDK and server, entering auto-reconnect state.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedKeepAliveTimeout"] = 14] = "ConnectionChangedKeepAliveTimeout";
  /**
   * 15: Successfully rejoined the channel.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedRejoinSuccess"] = 15] = "ConnectionChangedRejoinSuccess";
  /**
   * 16: SDK lost connection with the server.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedLost"] = 16] = "ConnectionChangedLost";
  /**
   * 17: Connection state changed due to echo test.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedEchoTest"] = 17] = "ConnectionChangedEchoTest";
  /**
   * 18: Local IP address changed by the user.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedClientIpAddressChangedByUser"] = 18] = "ConnectionChangedClientIpAddressChangedByUser";
  /**
   * 19: The same UID joined the same channel from different devices.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedSameUidLogin"] = 19] = "ConnectionChangedSameUidLogin";
  /**
   * 20: The number of broadcasters in the channel has reached the limit.
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedTooManyBroadcasters"] = 20] = "ConnectionChangedTooManyBroadcasters";
  /**
   * @ignore
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedLicenseValidationFailure"] = 21] = "ConnectionChangedLicenseValidationFailure";
  /**
   * @ignore
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedCertificationVeryfyFailure"] = 22] = "ConnectionChangedCertificationVeryfyFailure";
  /**
   * @ignore
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedStreamChannelNotAvailable"] = 23] = "ConnectionChangedStreamChannelNotAvailable";
  /**
   * @ignore
   */
  ConnectionChangedReasonType[ConnectionChangedReasonType["ConnectionChangedInconsistentAppid"] = 24] = "ConnectionChangedInconsistentAppid";
  return ConnectionChangedReasonType;
}({});

/**
 * Reasons for client role change failure.
 */
export let ClientRoleChangeFailedReason = /*#__PURE__*/function (ClientRoleChangeFailedReason) {
  /**
   * 1: The number of broadcasters in the channel has reached the limit. This enum is reported only when the 128-user feature is enabled. The limit depends on the actual configuration when enabling the 128-user feature.
   */
  ClientRoleChangeFailedReason[ClientRoleChangeFailedReason["ClientRoleChangeFailedTooManyBroadcasters"] = 1] = "ClientRoleChangeFailedTooManyBroadcasters";
  /**
   * 2: The request is rejected by the server. Prompt the user to retry changing the role.
   */
  ClientRoleChangeFailedReason[ClientRoleChangeFailedReason["ClientRoleChangeFailedNotAuthorized"] = 2] = "ClientRoleChangeFailedNotAuthorized";
  /**
   * 3: Request timed out. Prompt the user to check the network connection and retry. Deprecated: This enum value is deprecated since v4.4.0 and not recommended for use.
   */
  ClientRoleChangeFailedReason[ClientRoleChangeFailedReason["ClientRoleChangeFailedRequestTimeOut"] = 3] = "ClientRoleChangeFailedRequestTimeOut";
  /**
   * 4: Network connection failed. You can troubleshoot the cause based on the reason reported in onConnectionStateChanged. Deprecated: This enum value is deprecated since v4.4.0 and not recommended for use.
   */
  ClientRoleChangeFailedReason[ClientRoleChangeFailedReason["ClientRoleChangeFailedConnectionFailed"] = 4] = "ClientRoleChangeFailedConnectionFailed";
  return ClientRoleChangeFailedReason;
}({});

/**
 * @ignore
 */
export let WlaccMessageReason = /*#__PURE__*/function (WlaccMessageReason) {
  /**
   * @ignore
   */
  WlaccMessageReason[WlaccMessageReason["WlaccMessageReasonWeakSignal"] = 0] = "WlaccMessageReasonWeakSignal";
  /**
   * @ignore
   */
  WlaccMessageReason[WlaccMessageReason["WlaccMessageReasonChannelCongestion"] = 1] = "WlaccMessageReasonChannelCongestion";
  return WlaccMessageReason;
}({});

/**
 * @ignore
 */
export let WlaccSuggestAction = /*#__PURE__*/function (WlaccSuggestAction) {
  /**
   * @ignore
   */
  WlaccSuggestAction[WlaccSuggestAction["WlaccSuggestActionCloseToWifi"] = 0] = "WlaccSuggestActionCloseToWifi";
  /**
   * @ignore
   */
  WlaccSuggestAction[WlaccSuggestAction["WlaccSuggestActionConnectSsid"] = 1] = "WlaccSuggestActionConnectSsid";
  /**
   * @ignore
   */
  WlaccSuggestAction[WlaccSuggestAction["WlaccSuggestActionCheck5g"] = 2] = "WlaccSuggestActionCheck5g";
  /**
   * @ignore
   */
  WlaccSuggestAction[WlaccSuggestAction["WlaccSuggestActionModifySsid"] = 3] = "WlaccSuggestActionModifySsid";
  return WlaccSuggestAction;
}({});

/**
 * @ignore
 */
export class WlAccStats {}

/**
 * Network connection type.
 */
export let NetworkType = /*#__PURE__*/function (NetworkType) {
  /**
   * -1: Unknown network connection type.
   */
  NetworkType[NetworkType["NetworkTypeUnknown"] = -1] = "NetworkTypeUnknown";
  /**
   * 0: Network connection is disconnected.
   */
  NetworkType[NetworkType["NetworkTypeDisconnected"] = 0] = "NetworkTypeDisconnected";
  /**
   * 1: LAN network type.
   */
  NetworkType[NetworkType["NetworkTypeLan"] = 1] = "NetworkTypeLan";
  /**
   * 2: Wi-Fi network type (including hotspot).
   */
  NetworkType[NetworkType["NetworkTypeWifi"] = 2] = "NetworkTypeWifi";
  /**
   * 3: 2G mobile network type.
   */
  NetworkType[NetworkType["NetworkTypeMobile2g"] = 3] = "NetworkTypeMobile2g";
  /**
   * 4: 3G mobile network type.
   */
  NetworkType[NetworkType["NetworkTypeMobile3g"] = 4] = "NetworkTypeMobile3g";
  /**
   * 5: 4G mobile network type.
   */
  NetworkType[NetworkType["NetworkTypeMobile4g"] = 5] = "NetworkTypeMobile4g";
  /**
   * 6: 5G mobile network type.
   */
  NetworkType[NetworkType["NetworkTypeMobile5g"] = 6] = "NetworkTypeMobile5g";
  return NetworkType;
}({});

/**
 * View setup mode.
 */
export let VideoViewSetupMode = /*#__PURE__*/function (VideoViewSetupMode) {
  /**
   * 0: (Default) Clears all added views and replaces them with a new view.
   */
  VideoViewSetupMode[VideoViewSetupMode["VideoViewSetupReplace"] = 0] = "VideoViewSetupReplace";
  /**
   * 1: Adds a view.
   */
  VideoViewSetupMode[VideoViewSetupMode["VideoViewSetupAdd"] = 1] = "VideoViewSetupAdd";
  /**
   * 2: Removes a view. When you no longer need a view, it is recommended to set setupMode to VideoViewSetupRemove to remove the view in time, otherwise it may cause rendering resource leaks.
   */
  VideoViewSetupMode[VideoViewSetupMode["VideoViewSetupRemove"] = 2] = "VideoViewSetupRemove";
  return VideoViewSetupMode;
}({});

/**
 * Properties of the video canvas object.
 */
export class VideoCanvas {}

/**
 * Brightness contrast level.
 */
export let LighteningContrastLevel = /*#__PURE__*/function (LighteningContrastLevel) {
  /**
   * 0: Low contrast.
   */
  LighteningContrastLevel[LighteningContrastLevel["LighteningContrastLow"] = 0] = "LighteningContrastLow";
  /**
   * 1: Normal contrast.
   */
  LighteningContrastLevel[LighteningContrastLevel["LighteningContrastNormal"] = 1] = "LighteningContrastNormal";
  /**
   * 2: High contrast.
   */
  LighteningContrastLevel[LighteningContrastLevel["LighteningContrastHigh"] = 2] = "LighteningContrastHigh";
  return LighteningContrastLevel;
}({});

/**
 * Beauty options.
 */
export class BeautyOptions {}

/**
 * Selects the specific facial area to be adjusted.
 *
 * Since Available since v4.4.0.
 */
export let FaceShapeArea = /*#__PURE__*/function (FaceShapeArea) {
  /**
   * (-1): Default value, indicates an invalid area, the facial shaping effect is not applied.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNone"] = -1] = "FaceShapeAreaNone";
  /**
   * (100): Head area, used to achieve a smaller head effect. Value range: [0, 100], default is 50. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaHeadscale"] = 100] = "FaceShapeAreaHeadscale";
  /**
   * (101): Forehead area, used to adjust the hairline height. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaForehead"] = 101] = "FaceShapeAreaForehead";
  /**
   * (102): Face contour area, used to achieve a slimming face effect. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaFacecontour"] = 102] = "FaceShapeAreaFacecontour";
  /**
   * (103): Face length area, used to elongate the face. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaFacelength"] = 103] = "FaceShapeAreaFacelength";
  /**
   * (104): Face width area, used to achieve a narrower face effect. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaFacewidth"] = 104] = "FaceShapeAreaFacewidth";
  /**
   * (105): Cheekbone area, used to adjust the width of the cheekbones. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaCheekbone"] = 105] = "FaceShapeAreaCheekbone";
  /**
   * (106): Cheek area, used to adjust the width of the cheeks. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaCheek"] = 106] = "FaceShapeAreaCheek";
  /**
   * (107): Mandible area, used to adjust the width of the jawbone. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaMandible"] = 107] = "FaceShapeAreaMandible";
  /**
   * (108): Chin area, used to adjust the length of the chin. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaChin"] = 108] = "FaceShapeAreaChin";
  /**
   * (200): Eye area, used to achieve a bigger eye effect. Value range: [0, 100], default is 50. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyescale"] = 200] = "FaceShapeAreaEyescale";
  /**
   * (201): Eye distance area, used to adjust the distance between the eyes. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyedistance"] = 201] = "FaceShapeAreaEyedistance";
  /**
   * (202): Eye position area, used to adjust the overall position of the eyes. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyeposition"] = 202] = "FaceShapeAreaEyeposition";
  /**
   * (203): Lower eyelid area, used to adjust the shape of the lower eyelid. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaLowereyelid"] = 203] = "FaceShapeAreaLowereyelid";
  /**
   * (204): Pupil area, used to adjust the size of the pupils. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyepupils"] = 204] = "FaceShapeAreaEyepupils";
  /**
   * (205): Inner eye corner area, used to adjust the shape of the inner eye corners. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyeinnercorner"] = 205] = "FaceShapeAreaEyeinnercorner";
  /**
   * (206): Outer eye corner area, used to adjust the shape of the outer eye corners. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyeoutercorner"] = 206] = "FaceShapeAreaEyeoutercorner";
  /**
   * (300): Nose length area, used to elongate the nose. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNoselength"] = 300] = "FaceShapeAreaNoselength";
  /**
   * (301): Nose width area, used to achieve a slimmer nose effect. Value range: [0, 100], default is 0. The higher the value, the more noticeable the slimming effect.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNosewidth"] = 301] = "FaceShapeAreaNosewidth";
  /**
   * (302): Nose wing area, used to adjust the width of the nose wings. Value range: [0, 100], default is 10. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNosewing"] = 302] = "FaceShapeAreaNosewing";
  /**
   * (303): Nose root area, used to adjust the height of the nose root. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNoseroot"] = 303] = "FaceShapeAreaNoseroot";
  /**
   * (304): Nose bridge area, used to adjust the height of the nose bridge. Value range: [0, 100], default is 50. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNosebridge"] = 304] = "FaceShapeAreaNosebridge";
  /**
   * (305): Nose tip area, used to adjust the shape of the nose tip. Value range: [0, 100], default is 50. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNosetip"] = 305] = "FaceShapeAreaNosetip";
  /**
   * (306): Overall nose area, used to uniformly adjust the shape of the nose. Value range: [-100, 100], default is 50. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaNosegeneral"] = 306] = "FaceShapeAreaNosegeneral";
  /**
   * (400): Mouth area, used to achieve a larger mouth effect. Value range: [-100, 100], default is 20. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaMouthscale"] = 400] = "FaceShapeAreaMouthscale";
  /**
   * (401): Mouth position area, used to adjust the overall position of the mouth. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaMouthposition"] = 401] = "FaceShapeAreaMouthposition";
  /**
   * (402): Mouth smile area, used to adjust the degree of mouth corner lift. Value range: [0, 1], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaMouthsmile"] = 402] = "FaceShapeAreaMouthsmile";
  /**
   * (403): Lip shape area, used to adjust the shape of the lips. Value range: [0, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaMouthlip"] = 403] = "FaceShapeAreaMouthlip";
  /**
   * (500): Eyebrow position area, used to adjust the overall position of the eyebrows. Value range: [-100, 100], default is 0. The greater the absolute value, the more noticeable the adjustment. Negative values indicate the opposite direction.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyebrowposition"] = 500] = "FaceShapeAreaEyebrowposition";
  /**
   * (501): Eyebrow thickness area, used to adjust the thickness of the eyebrows. Value range: [-100, 100], default is 0. The higher the value, the more noticeable the adjustment.
   */
  FaceShapeArea[FaceShapeArea["FaceShapeAreaEyebrowthickness"] = 501] = "FaceShapeAreaEyebrowthickness";
  return FaceShapeArea;
}({});

/**
 * Filter effect options.
 */
export class FaceShapeAreaOptions {}

/**
 * Face shaping makeup effect style options.
 *
 * Since Available since v4.4.0.
 */
export let FaceShapeBeautyStyle = /*#__PURE__*/function (FaceShapeBeautyStyle) {
  /**
   * (0): (Default) Female style makeup effect.
   */
  FaceShapeBeautyStyle[FaceShapeBeautyStyle["FaceShapeBeautyStyleFemale"] = 0] = "FaceShapeBeautyStyleFemale";
  /**
   * (1): Male style makeup effect.
   */
  FaceShapeBeautyStyle[FaceShapeBeautyStyle["FaceShapeBeautyStyleMale"] = 1] = "FaceShapeBeautyStyleMale";
  /**
   * (2): Natural style makeup effect, makes minimal adjustments to facial features only.
   */
  FaceShapeBeautyStyle[FaceShapeBeautyStyle["FaceShapeBeautyStyleNatural"] = 2] = "FaceShapeBeautyStyleNatural";
  return FaceShapeBeautyStyle;
}({});

/**
 * Face shaping style options.
 */
export class FaceShapeBeautyOptions {}

/**
 * Filter effect options.
 */
export class FilterEffectOptions {}

/**
 * Low-light enhancement mode.
 */
export let LowLightEnhanceMode = /*#__PURE__*/function (LowLightEnhanceMode) {
  /**
   * 0: (Default) Auto mode. The SDK automatically enables or disables low-light enhancement based on ambient brightness to provide proper lighting and avoid overexposure.
   */
  LowLightEnhanceMode[LowLightEnhanceMode["LowLightEnhanceAuto"] = 0] = "LowLightEnhanceAuto";
  /**
   * 1: Manual mode. You need to manually enable or disable low-light enhancement.
   */
  LowLightEnhanceMode[LowLightEnhanceMode["LowLightEnhanceManual"] = 1] = "LowLightEnhanceManual";
  return LowLightEnhanceMode;
}({});

/**
 * Low-light enhancement level.
 */
export let LowLightEnhanceLevel = /*#__PURE__*/function (LowLightEnhanceLevel) {
  /**
   * 0: (Default) High-quality low-light enhancement. Optimizes brightness, detail, and noise in the video image with moderate performance consumption and processing speed, offering the best overall image quality.
   */
  LowLightEnhanceLevel[LowLightEnhanceLevel["LowLightEnhanceLevelHighQuality"] = 0] = "LowLightEnhanceLevelHighQuality";
  /**
   * 1: Performance-first low-light enhancement. Optimizes brightness and detail with lower performance consumption and faster processing speed.
   */
  LowLightEnhanceLevel[LowLightEnhanceLevel["LowLightEnhanceLevelFast"] = 1] = "LowLightEnhanceLevelFast";
  return LowLightEnhanceLevel;
}({});

/**
 * Low-light enhancement options.
 */
export class LowlightEnhanceOptions {}

/**
 * Video denoising mode.
 */
export let VideoDenoiserMode = /*#__PURE__*/function (VideoDenoiserMode) {
  /**
   * 0: (Default) Auto mode. The SDK automatically enables or disables video denoising based on ambient brightness.
   */
  VideoDenoiserMode[VideoDenoiserMode["VideoDenoiserAuto"] = 0] = "VideoDenoiserAuto";
  /**
   * 1: Manual mode. You need to manually enable or disable video denoising.
   */
  VideoDenoiserMode[VideoDenoiserMode["VideoDenoiserManual"] = 1] = "VideoDenoiserManual";
  return VideoDenoiserMode;
}({});

/**
 * Video denoising level.
 */
export let VideoDenoiserLevel = /*#__PURE__*/function (VideoDenoiserLevel) {
  /**
   * 0: (Default) Denoising with priority on video quality. Balances performance consumption and denoising effect. Moderate performance usage, moderate speed, optimal overall quality.
   */
  VideoDenoiserLevel[VideoDenoiserLevel["VideoDenoiserLevelHighQuality"] = 0] = "VideoDenoiserLevelHighQuality";
  /**
   * 1: Denoising with priority on performance. Focuses on saving performance over denoising effect. Low performance consumption, fast speed. To avoid noticeable ghosting, it is recommended to use this setting when the camera is stationary.
   */
  VideoDenoiserLevel[VideoDenoiserLevel["VideoDenoiserLevelFast"] = 1] = "VideoDenoiserLevelFast";
  return VideoDenoiserLevel;
}({});

/**
 * Video denoising options.
 */
export class VideoDenoiserOptions {}

/**
 * Color enhancement options.
 */
export class ColorEnhanceOptions {}

/**
 * Custom background.
 */
export let BackgroundSourceType = /*#__PURE__*/function (BackgroundSourceType) {
  /**
   * @ignore
   */
  BackgroundSourceType[BackgroundSourceType["BackgroundNone"] = 0] = "BackgroundNone";
  /**
   * 1: (Default) Solid color background.
   */
  BackgroundSourceType[BackgroundSourceType["BackgroundColor"] = 1] = "BackgroundColor";
  /**
   * 2: Background image in PNG or JPG format.
   */
  BackgroundSourceType[BackgroundSourceType["BackgroundImg"] = 2] = "BackgroundImg";
  /**
   * 3: Blurred background.
   */
  BackgroundSourceType[BackgroundSourceType["BackgroundBlur"] = 3] = "BackgroundBlur";
  /**
   * 4: Local video background in formats such as MP4, AVI, MKV, FLV.
   */
  BackgroundSourceType[BackgroundSourceType["BackgroundVideo"] = 4] = "BackgroundVideo";
  return BackgroundSourceType;
}({});

/**
 * Degree of background blur for custom background image.
 */
export let BackgroundBlurDegree = /*#__PURE__*/function (BackgroundBlurDegree) {
  /**
   * 1: Low blur level for custom background image. Users can almost clearly see the background.
   */
  BackgroundBlurDegree[BackgroundBlurDegree["BlurDegreeLow"] = 1] = "BlurDegreeLow";
  /**
   * 2: Medium blur level for custom background image. Users have difficulty seeing the background clearly.
   */
  BackgroundBlurDegree[BackgroundBlurDegree["BlurDegreeMedium"] = 2] = "BlurDegreeMedium";
  /**
   * 3: (Default) High blur level for custom background image. Users can barely see the background.
   */
  BackgroundBlurDegree[BackgroundBlurDegree["BlurDegreeHigh"] = 3] = "BlurDegreeHigh";
  return BackgroundBlurDegree;
}({});

/**
 * Custom background.
 */
export class VirtualBackgroundSource {}

/**
 * Algorithm for background processing.
 */
export let SegModelType = /*#__PURE__*/function (SegModelType) {
  /**
   * 1: (Default) Background processing algorithm suitable for all scenarios.
   */
  SegModelType[SegModelType["SegModelAi"] = 1] = "SegModelAi";
  /**
   * 2: Background processing algorithm (green screen only).
   */
  SegModelType[SegModelType["SegModelGreen"] = 2] = "SegModelGreen";
  return SegModelType;
}({});

/**
 * Background image processing properties.
 */
export class SegmentationProperty {}

/**
 * Type of custom audio capture track.
 */
export let AudioTrackType = /*#__PURE__*/function (AudioTrackType) {
  /**
   * @ignore
   */
  AudioTrackType[AudioTrackType["AudioTrackInvalid"] = -1] = "AudioTrackInvalid";
  /**
   * 0: Mixable audio track. Supports mixing with other audio streams (e.g., microphone audio) before local playback or publishing to the channel. Has higher latency compared to non-mixable audio tracks.
   */
  AudioTrackType[AudioTrackType["AudioTrackMixable"] = 0] = "AudioTrackMixable";
  /**
   * 1: Non-mixable audio track. Replaces microphone capture and does not support mixing with other audio streams. Has lower latency compared to mixable audio tracks. If you specify AudioTrackDirect, you must set publishMicrophoneTrack in ChannelMediaOptions to false when calling joinChannel, otherwise joining the channel will fail and return error code -2.
   */
  AudioTrackType[AudioTrackType["AudioTrackDirect"] = 1] = "AudioTrackDirect";
  return AudioTrackType;
}({});

/**
 * Configuration options for custom audio tracks.
 */
export class AudioTrackConfig {}

/**
 * Preset voice beautifier options.
 */
export let VoiceBeautifierPreset = /*#__PURE__*/function (VoiceBeautifierPreset) {
  /**
   * Original voice, i.e., disables voice beautifier effects.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["VoiceBeautifierOff"] = 0] = "VoiceBeautifierOff";
  /**
   * Magnetic (male). This setting is only effective for male voices. Do not use it for female voices, or audio distortion may occur.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["ChatBeautifierMagnetic"] = 16843008] = "ChatBeautifierMagnetic";
  /**
   * Fresh (female). This setting is only effective for female voices. Do not use it for male voices, or audio distortion may occur.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["ChatBeautifierFresh"] = 16843264] = "ChatBeautifierFresh";
  /**
   * Energetic (female). This setting is only effective for female voices. Do not use it for male voices, or audio distortion may occur.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["ChatBeautifierVitality"] = 16843520] = "ChatBeautifierVitality";
  /**
   * Singing beautifier.
   *  If you call setVoiceBeautifierPreset (SingingBeautifier), you can beautify male voices and add a small room reverb effect. Do not use for female voices, or audio distortion may occur.
   *  If you call setVoiceBeautifierParameters (SingingBeautifier, param1, param2), you can beautify both male and female voices and add reverb effects.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["SingingBeautifier"] = 16908544] = "SingingBeautifier";
  /**
   * Vigorous.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationVigorous"] = 16974080] = "TimbreTransformationVigorous";
  /**
   * Deep.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationDeep"] = 16974336] = "TimbreTransformationDeep";
  /**
   * Mellow.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationMellow"] = 16974592] = "TimbreTransformationMellow";
  /**
   * Falsetto.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationFalsetto"] = 16974848] = "TimbreTransformationFalsetto";
  /**
   * Full.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationFull"] = 16975104] = "TimbreTransformationFull";
  /**
   * Clear.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationClear"] = 16975360] = "TimbreTransformationClear";
  /**
   * Resounding.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationResounding"] = 16975616] = "TimbreTransformationResounding";
  /**
   * Ringing.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["TimbreTransformationRinging"] = 16975872] = "TimbreTransformationRinging";
  /**
   * Ultra-high quality voice, which makes the audio clearer and more detailed.
   *  For better results, it is recommended to set the profile parameter of setAudioProfile2 to AudioProfileMusicHighQuality (4) or AudioProfileMusicHighQualityStereo (5), and the scenario parameter to AudioScenarioGameStreaming (3), before calling setVoiceBeautifierPreset.
   *  If the user's audio capture device can highly restore audio details, it is recommended not to enable ultra-high quality voice, otherwise the SDK may over-restore audio details and fail to achieve the expected effect.
   */
  VoiceBeautifierPreset[VoiceBeautifierPreset["UltraHighQualityVoice"] = 17039616] = "UltraHighQualityVoice";
  return VoiceBeautifierPreset;
}({});

/**
 * Preset audio effects.
 *
 * setAudioProfile profile
 * Preset audio effects profile
 *  RoomAcousticsVirtualStereo
 *  RoomAcoustics3dVoice
 *  RoomAcousticsVirtualSurroundSound AudioProfileMusicHighQualityStereo or AudioProfileMusicStandardStereo Other preset effects (except AudioEffectOff) AudioProfileMusicHighQuality or AudioProfileMusicHighQualityStereo
 */
export let AudioEffectPreset = /*#__PURE__*/function (AudioEffectPreset) {
  /**
   * Original sound, i.e., disables voice effects.
   */
  AudioEffectPreset[AudioEffectPreset["AudioEffectOff"] = 0] = "AudioEffectOff";
  /**
   * KTV.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsKtv"] = 33620224] = "RoomAcousticsKtv";
  /**
   * Concert.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsVocalConcert"] = 33620480] = "RoomAcousticsVocalConcert";
  /**
   * Studio.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsStudio"] = 33620736] = "RoomAcousticsStudio";
  /**
   * Phonograph.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsPhonograph"] = 33620992] = "RoomAcousticsPhonograph";
  /**
   * Virtual stereo, where the SDK renders mono audio into stereo effect.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsVirtualStereo"] = 33621248] = "RoomAcousticsVirtualStereo";
  /**
   * Spacious.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsSpacial"] = 33621504] = "RoomAcousticsSpacial";
  /**
   * Ethereal.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsEthereal"] = 33621760] = "RoomAcousticsEthereal";
  /**
   * 3D voice, where the SDK renders audio to surround the user. The default surround cycle is 10 seconds. After setting this effect, you can also call setAudioEffectParameters to modify the surround cycle. To hear the expected effect after enabling 3D voice, users need to use audio playback devices that support stereo.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcoustics3dVoice"] = 33622016] = "RoomAcoustics3dVoice";
  /**
   * Virtual surround sound, where the SDK simulates a surround sound field based on stereo to create a surround effect. To hear the expected effect after enabling virtual surround sound, users need to use audio playback devices that support stereo.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsVirtualSurroundSound"] = 33622272] = "RoomAcousticsVirtualSurroundSound";
  /**
   * Chorus. Agora recommends using this in chorus scenarios to enhance the spatial stereo effect of vocals.
   */
  AudioEffectPreset[AudioEffectPreset["RoomAcousticsChorus"] = 33623296] = "RoomAcousticsChorus";
  /**
   * Uncle. Recommended for processing male voices; otherwise, the effect may not meet expectations.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectUncle"] = 33685760] = "VoiceChangerEffectUncle";
  /**
   * Elderly male. Recommended for processing male voices; otherwise, the effect may not meet expectations.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectOldman"] = 33686016] = "VoiceChangerEffectOldman";
  /**
   * Boy. Recommended for processing male voices; otherwise, the effect may not meet expectations.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectBoy"] = 33686272] = "VoiceChangerEffectBoy";
  /**
   * Young woman. Recommended for processing female voices; otherwise, the effect may not meet expectations.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectSister"] = 33686528] = "VoiceChangerEffectSister";
  /**
   * Girl. Recommended for processing female voices; otherwise, the effect may not meet expectations.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectGirl"] = 33686784] = "VoiceChangerEffectGirl";
  /**
   * Pigsy.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectPigking"] = 33687040] = "VoiceChangerEffectPigking";
  /**
   * Hulk.
   */
  AudioEffectPreset[AudioEffectPreset["VoiceChangerEffectHulk"] = 33687296] = "VoiceChangerEffectHulk";
  /**
   * R&B.
   */
  AudioEffectPreset[AudioEffectPreset["StyleTransformationRnb"] = 33751296] = "StyleTransformationRnb";
  /**
   * Pop.
   */
  AudioEffectPreset[AudioEffectPreset["StyleTransformationPopular"] = 33751552] = "StyleTransformationPopular";
  /**
   * Electronic music, where the SDK corrects the actual pitch of the audio based on the natural major scale with C as the tonic. After setting this effect, you can also call setAudioEffectParameters to adjust the base scale and tonic pitch.
   */
  AudioEffectPreset[AudioEffectPreset["PitchCorrection"] = 33816832] = "PitchCorrection";
  return AudioEffectPreset;
}({});

/**
 * Preset voice conversion options.
 */
export let VoiceConversionPreset = /*#__PURE__*/function (VoiceConversionPreset) {
  /**
   * Original voice, i.e., disables voice conversion effects.
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceConversionOff"] = 0] = "VoiceConversionOff";
  /**
   * Neutral. To avoid audio distortion, make sure to apply this effect only to female voices.
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerNeutral"] = 50397440] = "VoiceChangerNeutral";
  /**
   * Sweet. To avoid audio distortion, make sure to apply this effect only to female voices.
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerSweet"] = 50397696] = "VoiceChangerSweet";
  /**
   * Steady. To avoid audio distortion, make sure to apply this effect only to male voices.
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerSolid"] = 50397952] = "VoiceChangerSolid";
  /**
   * Deep. To avoid audio distortion, make sure to apply this effect only to male voices.
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerBass"] = 50398208] = "VoiceChangerBass";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerCartoon"] = 50398464] = "VoiceChangerCartoon";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerChildlike"] = 50398720] = "VoiceChangerChildlike";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerPhoneOperator"] = 50398976] = "VoiceChangerPhoneOperator";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerMonster"] = 50399232] = "VoiceChangerMonster";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerTransformers"] = 50399488] = "VoiceChangerTransformers";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerGroot"] = 50399744] = "VoiceChangerGroot";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerDarthVader"] = 50400000] = "VoiceChangerDarthVader";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerIronLady"] = 50400256] = "VoiceChangerIronLady";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerShinChan"] = 50400512] = "VoiceChangerShinChan";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerGirlishMan"] = 50400768] = "VoiceChangerGirlishMan";
  /**
   * @ignore
   */
  VoiceConversionPreset[VoiceConversionPreset["VoiceChangerChipmunk"] = 50401024] = "VoiceChangerChipmunk";
  return VoiceConversionPreset;
}({});

/**
 * Preset headphone equalizer types.
 */
export let HeadphoneEqualizerPreset = /*#__PURE__*/function (HeadphoneEqualizerPreset) {
  /**
   * Disable headphone equalizer and listen to the original audio.
   */
  HeadphoneEqualizerPreset[HeadphoneEqualizerPreset["HeadphoneEqualizerOff"] = 0] = "HeadphoneEqualizerOff";
  /**
   * Use the equalizer for over-ear headphones.
   */
  HeadphoneEqualizerPreset[HeadphoneEqualizerPreset["HeadphoneEqualizerOverear"] = 67108865] = "HeadphoneEqualizerOverear";
  /**
   * Use the equalizer for in-ear headphones.
   */
  HeadphoneEqualizerPreset[HeadphoneEqualizerPreset["HeadphoneEqualizerInear"] = 67108866] = "HeadphoneEqualizerInear";
  return HeadphoneEqualizerPreset;
}({});

/**
 * AI tuner voice effect types.
 */
export let VoiceAiTunerType = /*#__PURE__*/function (VoiceAiTunerType) {
  /**
   * 0: Mature male voice. A deep and magnetic male voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerMatureMale"] = 0] = "VoiceAiTunerMatureMale";
  /**
   * 1: Fresh male voice. A fresh and slightly sweet male voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerFreshMale"] = 1] = "VoiceAiTunerFreshMale";
  /**
   * 2: Mature female voice. A deep and charming female voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerElegantFemale"] = 2] = "VoiceAiTunerElegantFemale";
  /**
   * 3: Cute female voice. A high-pitched and adorable female voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerSweetFemale"] = 3] = "VoiceAiTunerSweetFemale";
  /**
   * 4: Warm male singing voice. A warm and melodious male voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerWarmMaleSinging"] = 4] = "VoiceAiTunerWarmMaleSinging";
  /**
   * 5: Gentle female singing voice. A soft and delicate female voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerGentleFemaleSinging"] = 5] = "VoiceAiTunerGentleFemaleSinging";
  /**
   * 6: Husky mature male singing voice. A unique hoarse male voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerHuskyMaleSinging"] = 6] = "VoiceAiTunerHuskyMaleSinging";
  /**
   * 7: Warm mature female singing voice. A warm and mature female voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerWarmElegantFemaleSinging"] = 7] = "VoiceAiTunerWarmElegantFemaleSinging";
  /**
   * 8: Powerful male singing voice. A strong and forceful male voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerPowerfulMaleSinging"] = 8] = "VoiceAiTunerPowerfulMaleSinging";
  /**
   * 9: Dreamy female singing voice. A dreamy and soft female voice.
   */
  VoiceAiTunerType[VoiceAiTunerType["VoiceAiTunerDreamyFemaleSinging"] = 9] = "VoiceAiTunerDreamyFemaleSinging";
  return VoiceAiTunerType;
}({});

/**
 * Audio configuration for the shared screen stream.
 *
 * (Android only) Applies only when captureAudio is set to true.
 */
export class ScreenAudioParameters {}

/**
 * @ignore
 */
export class ScreenCaptureParameters {}

/**
 * Recording quality.
 */
export let AudioRecordingQualityType = /*#__PURE__*/function (AudioRecordingQualityType) {
  /**
   * 0: Low quality. Sample rate is 32 kHz, file size for 10 minutes of recording is approximately 1.2 MB.
   */
  AudioRecordingQualityType[AudioRecordingQualityType["AudioRecordingQualityLow"] = 0] = "AudioRecordingQualityLow";
  /**
   * 1: Medium quality. Sample rate is 32 kHz, file size for 10 minutes of recording is approximately 2 MB.
   */
  AudioRecordingQualityType[AudioRecordingQualityType["AudioRecordingQualityMedium"] = 1] = "AudioRecordingQualityMedium";
  /**
   * 2: High quality. Sample rate is 32 kHz, file size for 10 minutes of recording is approximately 3.75 MB.
   */
  AudioRecordingQualityType[AudioRecordingQualityType["AudioRecordingQualityHigh"] = 2] = "AudioRecordingQualityHigh";
  /**
   * 3: Ultra-high quality. Sample rate is 32 kHz, file size for 10 minutes of recording is approximately 7.5 MB.
   */
  AudioRecordingQualityType[AudioRecordingQualityType["AudioRecordingQualityUltraHigh"] = 3] = "AudioRecordingQualityUltraHigh";
  return AudioRecordingQualityType;
}({});

/**
 * Recording content. Set in startAudioRecording.
 */
export let AudioFileRecordingType = /*#__PURE__*/function (AudioFileRecordingType) {
  /**
   * 1: Record only the local user's audio.
   */
  AudioFileRecordingType[AudioFileRecordingType["AudioFileRecordingMic"] = 1] = "AudioFileRecordingMic";
  /**
   * 2: Record only the audio of all remote users.
   */
  AudioFileRecordingType[AudioFileRecordingType["AudioFileRecordingPlayback"] = 2] = "AudioFileRecordingPlayback";
  /**
   * 3: Record the mixed audio of the local and all remote users.
   */
  AudioFileRecordingType[AudioFileRecordingType["AudioFileRecordingMixed"] = 3] = "AudioFileRecordingMixed";
  return AudioFileRecordingType;
}({});

/**
 * Audio encoding content.
 */
export let AudioEncodedFrameObserverPosition = /*#__PURE__*/function (AudioEncodedFrameObserverPosition) {
  /**
   * 1: Encode only the local user's audio.
   */
  AudioEncodedFrameObserverPosition[AudioEncodedFrameObserverPosition["AudioEncodedFrameObserverPositionRecord"] = 1] = "AudioEncodedFrameObserverPositionRecord";
  /**
   * 2: Encode only the audio of all remote users.
   */
  AudioEncodedFrameObserverPosition[AudioEncodedFrameObserverPosition["AudioEncodedFrameObserverPositionPlayback"] = 2] = "AudioEncodedFrameObserverPositionPlayback";
  /**
   * 3: Encode the mixed audio of the local and all remote users.
   */
  AudioEncodedFrameObserverPosition[AudioEncodedFrameObserverPosition["AudioEncodedFrameObserverPositionMixed"] = 3] = "AudioEncodedFrameObserverPositionMixed";
  return AudioEncodedFrameObserverPosition;
}({});

/**
 * Recording configuration.
 */
export class AudioRecordingConfiguration {}

/**
 * Observer settings for encoded audio.
 */
export class AudioEncodedFrameObserverConfig {}

/**
 * Observer for encoded audio.
 */

/**
 * Access region, i.e., the region where the SDK connects to the server.
 */
export let AreaCode = /*#__PURE__*/function (AreaCode) {
  /**
   * Mainland China.
   */
  AreaCode[AreaCode["AreaCodeCn"] = 1] = "AreaCodeCn";
  /**
   * North America region.
   */
  AreaCode[AreaCode["AreaCodeNa"] = 2] = "AreaCodeNa";
  /**
   * Europe region.
   */
  AreaCode[AreaCode["AreaCodeEu"] = 4] = "AreaCodeEu";
  /**
   * Asia region excluding China.
   */
  AreaCode[AreaCode["AreaCodeAs"] = 8] = "AreaCodeAs";
  /**
   * Japan.
   */
  AreaCode[AreaCode["AreaCodeJp"] = 16] = "AreaCodeJp";
  /**
   * India.
   */
  AreaCode[AreaCode["AreaCodeIn"] = 32] = "AreaCodeIn";
  /**
   * Global.
   */
  AreaCode[AreaCode["AreaCodeGlob"] = 4294967295] = "AreaCodeGlob";
  return AreaCode;
}({});

/**
 * @ignore
 */
export let AreaCodeEx = /*#__PURE__*/function (AreaCodeEx) {
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeOc"] = 64] = "AreaCodeOc";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeSa"] = 128] = "AreaCodeSa";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeAf"] = 256] = "AreaCodeAf";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeKr"] = 512] = "AreaCodeKr";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeHkmc"] = 1024] = "AreaCodeHkmc";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeUs"] = 2048] = "AreaCodeUs";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeRu"] = 4096] = "AreaCodeRu";
  /**
   * @ignore
   */
  AreaCodeEx[AreaCodeEx["AreaCodeOvs"] = 4294967294] = "AreaCodeOvs";
  return AreaCodeEx;
}({});

/**
 * Error codes for channel media relay failures.
 */
export let ChannelMediaRelayError = /*#__PURE__*/function (ChannelMediaRelayError) {
  /**
   * 0: Everything works fine.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayOk"] = 0] = "RelayOk";
  /**
   * 1: Server response error.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorServerErrorResponse"] = 1] = "RelayErrorServerErrorResponse";
  /**
   * 2: No response from the server.
   * This error may be caused by poor network conditions. If this error is reported when initiating channel media relay, you can retry later; if it occurs during the relay process, you can call the leaveChannel method to leave the channel.
   * This error may also occur if the current App ID has not enabled the channel media relay feature. You can [contact technical support](https://www.agora.io/cn/contact/) to request enabling channel media relay.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorServerNoResponse"] = 2] = "RelayErrorServerNoResponse";
  /**
   * 3: SDK fails to get the service, possibly due to limited server resources.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorNoResourceAvailable"] = 3] = "RelayErrorNoResourceAvailable";
  /**
   * 4: Failed to initiate channel media relay request.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorFailedJoinSrc"] = 4] = "RelayErrorFailedJoinSrc";
  /**
   * 5: Failed to accept channel media relay request.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorFailedJoinDest"] = 5] = "RelayErrorFailedJoinDest";
  /**
   * 6: Server fails to receive media stream from the source channel.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorFailedPacketReceivedFromSrc"] = 6] = "RelayErrorFailedPacketReceivedFromSrc";
  /**
   * 7: Server fails to send media stream to the destination channel.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorFailedPacketSentToDest"] = 7] = "RelayErrorFailedPacketSentToDest";
  /**
   * 8: SDK loses connection with the server due to poor network quality. You can call the leaveChannel method to leave the current channel.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorServerConnectionLost"] = 8] = "RelayErrorServerConnectionLost";
  /**
   * 9: Internal server error.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorInternalError"] = 9] = "RelayErrorInternalError";
  /**
   * 10: The token of the source channel has expired.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorSrcTokenExpired"] = 10] = "RelayErrorSrcTokenExpired";
  /**
   * 11: The token of the destination channel has expired.
   */
  ChannelMediaRelayError[ChannelMediaRelayError["RelayErrorDestTokenExpired"] = 11] = "RelayErrorDestTokenExpired";
  return ChannelMediaRelayError;
}({});

/**
 * State codes for channel media relay.
 */
export let ChannelMediaRelayState = /*#__PURE__*/function (ChannelMediaRelayState) {
  /**
   * 0: Idle state. After successfully calling stopChannelMediaRelay to stop the relay, onChannelMediaRelayStateChanged will report this state.
   */
  ChannelMediaRelayState[ChannelMediaRelayState["RelayStateIdle"] = 0] = "RelayStateIdle";
  /**
   * 1: SDK is attempting to relay across channels.
   */
  ChannelMediaRelayState[ChannelMediaRelayState["RelayStateConnecting"] = 1] = "RelayStateConnecting";
  /**
   * 2: The host in the source channel has successfully joined the destination channel.
   */
  ChannelMediaRelayState[ChannelMediaRelayState["RelayStateRunning"] = 2] = "RelayStateRunning";
  /**
   * 3: An error occurred. See the code parameter in onChannelMediaRelayStateChanged for details.
   */
  ChannelMediaRelayState[ChannelMediaRelayState["RelayStateFailure"] = 3] = "RelayStateFailure";
  return ChannelMediaRelayState;
}({});

/**
 * Channel media information.
 */
export class ChannelMediaInfo {}

/**
 * Cross-channel media relay configuration information.
 */
export class ChannelMediaRelayConfiguration {}

/**
 * Uplink network information.
 */
export class UplinkNetworkInfo {}

/**
 * @ignore
 */
export class PeerDownlinkInfo {}

/**
 * @ignore
 */
export class DownlinkNetworkInfo {}

/**
 * Built-in encryption modes.
 *
 * It is recommended to use the Aes128Gcm2 or Aes256Gcm2 encryption modes. These modes support salt, offering higher security.
 */
export let EncryptionMode = /*#__PURE__*/function (EncryptionMode) {
  /**
   * 1: 128-bit AES encryption, XTS mode.
   */
  EncryptionMode[EncryptionMode["Aes128Xts"] = 1] = "Aes128Xts";
  /**
   * 2: 128-bit AES encryption, ECB mode.
   */
  EncryptionMode[EncryptionMode["Aes128Ecb"] = 2] = "Aes128Ecb";
  /**
   * 3: 256-bit AES encryption, XTS mode.
   */
  EncryptionMode[EncryptionMode["Aes256Xts"] = 3] = "Aes256Xts";
  /**
   * 4: 128-bit SM4 encryption, ECB mode.
   */
  EncryptionMode[EncryptionMode["Sm4128Ecb"] = 4] = "Sm4128Ecb";
  /**
   * 5: 128-bit AES encryption, GCM mode.
   */
  EncryptionMode[EncryptionMode["Aes128Gcm"] = 5] = "Aes128Gcm";
  /**
   * 6: 256-bit AES encryption, GCM mode.
   */
  EncryptionMode[EncryptionMode["Aes256Gcm"] = 6] = "Aes256Gcm";
  /**
   * 7: (Default) 128-bit AES encryption, GCM mode. This encryption mode requires setting a salt (encryptionKdfSalt).
   */
  EncryptionMode[EncryptionMode["Aes128Gcm2"] = 7] = "Aes128Gcm2";
  /**
   * 8: 256-bit AES encryption, GCM mode. This encryption mode requires setting a salt (encryptionKdfSalt).
   */
  EncryptionMode[EncryptionMode["Aes256Gcm2"] = 8] = "Aes256Gcm2";
  /**
   * Enumeration boundary value.
   */
  EncryptionMode[EncryptionMode["ModeEnd"] = 9] = "ModeEnd";
  return EncryptionMode;
}({});

/**
 * Configures the built-in encryption mode and key.
 */
export class EncryptionConfig {}

/**
 * Built-in encryption error types.
 */
export let EncryptionErrorType = /*#__PURE__*/function (EncryptionErrorType) {
  /**
   * 0: Internal error.
   */
  EncryptionErrorType[EncryptionErrorType["EncryptionErrorInternalFailure"] = 0] = "EncryptionErrorInternalFailure";
  /**
   * 1: Media stream decryption error. Make sure the encryption mode or key used by the sender and receiver is the same.
   */
  EncryptionErrorType[EncryptionErrorType["EncryptionErrorDecryptionFailure"] = 1] = "EncryptionErrorDecryptionFailure";
  /**
   * 2: Media stream encryption error.
   */
  EncryptionErrorType[EncryptionErrorType["EncryptionErrorEncryptionFailure"] = 2] = "EncryptionErrorEncryptionFailure";
  /**
   * 3: Data stream decryption error. Make sure the encryption mode or key used by the sender and receiver is the same.
   */
  EncryptionErrorType[EncryptionErrorType["EncryptionErrorDatastreamDecryptionFailure"] = 3] = "EncryptionErrorDatastreamDecryptionFailure";
  /**
   * 4: Data stream encryption error.
   */
  EncryptionErrorType[EncryptionErrorType["EncryptionErrorDatastreamEncryptionFailure"] = 4] = "EncryptionErrorDatastreamEncryptionFailure";
  return EncryptionErrorType;
}({});

/**
 * @ignore
 */
export let UploadErrorReason = /*#__PURE__*/function (UploadErrorReason) {
  /**
   * @ignore
   */
  UploadErrorReason[UploadErrorReason["UploadSuccess"] = 0] = "UploadSuccess";
  /**
   * @ignore
   */
  UploadErrorReason[UploadErrorReason["UploadNetError"] = 1] = "UploadNetError";
  /**
   * @ignore
   */
  UploadErrorReason[UploadErrorReason["UploadServerError"] = 2] = "UploadServerError";
  return UploadErrorReason;
}({});

/**
 * Device permission types.
 */
export let PermissionType = /*#__PURE__*/function (PermissionType) {
  /**
   * 0: Permission for audio capture device.
   */
  PermissionType[PermissionType["RecordAudio"] = 0] = "RecordAudio";
  /**
   * 1: Camera permission.
   */
  PermissionType[PermissionType["Camera"] = 1] = "Camera";
  /**
   * (Android only) 2: Screen sharing permission.
   */
  PermissionType[PermissionType["ScreenCapture"] = 2] = "ScreenCapture";
  return PermissionType;
}({});

/**
 * Subscription state.
 */
export let StreamSubscribeState = /*#__PURE__*/function (StreamSubscribeState) {
  /**
   * 0: Initial subscription state after joining the channel.
   */
  StreamSubscribeState[StreamSubscribeState["SubStateIdle"] = 0] = "SubStateIdle";
  /**
   * 1: Subscription failed. Possible reasons:
   *  Remote user:
   *  Called muteLocalAudioStream(true) or muteLocalVideoStream(true) to stop sending local media streams.
   *  Called disableAudio or disableVideo to disable the local audio or video module.
   *  Called enableLocalAudio(false) or enableLocalVideo(false) to disable local audio or video capture.
   *  User role is audience.
   *  Local user called the following methods to stop receiving remote media streams:
   *  Called muteRemoteAudioStream(true) or muteAllRemoteAudioStreams(true) to stop receiving remote audio streams.
   *  Called muteRemoteVideoStream(true) or muteAllRemoteVideoStreams(true) to stop receiving remote video streams.
   */
  StreamSubscribeState[StreamSubscribeState["SubStateNoSubscribed"] = 1] = "SubStateNoSubscribed";
  /**
   * 2: Subscribing in progress.
   */
  StreamSubscribeState[StreamSubscribeState["SubStateSubscribing"] = 2] = "SubStateSubscribing";
  /**
   * 3: Remote stream received, subscription successful.
   */
  StreamSubscribeState[StreamSubscribeState["SubStateSubscribed"] = 3] = "SubStateSubscribed";
  return StreamSubscribeState;
}({});

/**
 * Publishing state.
 */
export let StreamPublishState = /*#__PURE__*/function (StreamPublishState) {
  /**
   * 0: Initial publishing state after joining the channel.
   */
  StreamPublishState[StreamPublishState["PubStateIdle"] = 0] = "PubStateIdle";
  /**
   * 1: Publishing failed. Possible reasons:
   *  The local user called muteLocalAudioStream(true) or muteLocalVideoStream(true) to stop sending local media streams.
   *  The local user called disableAudio or disableVideo to disable the local audio or video module.
   *  The local user called enableLocalAudio(false) or enableLocalVideo(false) to disable local audio or video capture.
   *  The local user's role is audience.
   */
  StreamPublishState[StreamPublishState["PubStateNoPublished"] = 1] = "PubStateNoPublished";
  /**
   * 2: Publishing in progress.
   */
  StreamPublishState[StreamPublishState["PubStatePublishing"] = 2] = "PubStatePublishing";
  /**
   * 3: Publishing succeeded.
   */
  StreamPublishState[StreamPublishState["PubStatePublished"] = 3] = "PubStatePublished";
  return StreamPublishState;
}({});

/**
 * Configuration for audio and video loopback testing.
 */
export class EchoTestConfiguration {}

/**
 * User information.
 */
export class UserInfo {}

/**
 * Ear monitoring audio filter type.
 */
export let EarMonitoringFilterType = /*#__PURE__*/function (EarMonitoringFilterType) {
  /**
   * 1<<0: Do not add audio filters in ear monitoring.
   */
  EarMonitoringFilterType[EarMonitoringFilterType["EarMonitoringFilterNone"] = 1] = "EarMonitoringFilterNone";
  /**
   * 1<<1: Add vocal effect audio filters in ear monitoring. If you implement features such as voice beautification or sound effects, users can hear the processed sound in ear monitoring.
   */
  EarMonitoringFilterType[EarMonitoringFilterType["EarMonitoringFilterBuiltInAudioFilters"] = 2] = "EarMonitoringFilterBuiltInAudioFilters";
  /**
   * 1<<2: Add noise suppression audio filters in ear monitoring.
   */
  EarMonitoringFilterType[EarMonitoringFilterType["EarMonitoringFilterNoiseSuppression"] = 4] = "EarMonitoringFilterNoiseSuppression";
  /**
   * 1<<15: Reuse audio filters that have already been applied on the sending side. Reusing audio filters reduces CPU usage for ear monitoring, but increases latency. Suitable for scenarios where reducing CPU consumption is more important than minimizing ear monitoring latency.
   */
  EarMonitoringFilterType[EarMonitoringFilterType["EarMonitoringFilterReusePostProcessingFilter"] = 32768] = "EarMonitoringFilterReusePostProcessingFilter";
  return EarMonitoringFilterType;
}({});

/**
 * @ignore
 */
export let ThreadPriorityType = /*#__PURE__*/function (ThreadPriorityType) {
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["Lowest"] = 0] = "Lowest";
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["Low"] = 1] = "Low";
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["Normal"] = 2] = "Normal";
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["High"] = 3] = "High";
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["Highest"] = 4] = "Highest";
  /**
   * @ignore
   */
  ThreadPriorityType[ThreadPriorityType["Critical"] = 5] = "Critical";
  return ThreadPriorityType;
}({});

/**
 * Video encoding configuration for the shared screen stream.
 */
export class ScreenVideoParameters {}

/**
 * Parameter configuration for screen sharing.
 */
export class ScreenCaptureParameters2 {}

/**
 * The rendering state of media frames.
 */
export let MediaTraceEvent = /*#__PURE__*/function (MediaTraceEvent) {
  /**
   * 0: Video frame rendered.
   */
  MediaTraceEvent[MediaTraceEvent["MediaTraceEventVideoRendered"] = 0] = "MediaTraceEventVideoRendered";
  /**
   * 1: Video frame decoded.
   */
  MediaTraceEvent[MediaTraceEvent["MediaTraceEventVideoDecoded"] = 1] = "MediaTraceEventVideoDecoded";
  return MediaTraceEvent;
}({});

/**
 * Metric information during the video frame rendering process.
 */
export class VideoRenderingTracingInfo {}

/**
 * @ignore
 */
export let ConfigFetchType = /*#__PURE__*/function (ConfigFetchType) {
  /**
   * @ignore
   */
  ConfigFetchType[ConfigFetchType["ConfigFetchTypeInitialize"] = 1] = "ConfigFetchTypeInitialize";
  /**
   * @ignore
   */
  ConfigFetchType[ConfigFetchType["ConfigFetchTypeJoinChannel"] = 2] = "ConfigFetchTypeJoinChannel";
  return ConfigFetchType;
}({});

/**
 * @ignore
 */
export let LocalProxyMode = /*#__PURE__*/function (LocalProxyMode) {
  /**
   * @ignore
   */
  LocalProxyMode[LocalProxyMode["ConnectivityFirst"] = 0] = "ConnectivityFirst";
  /**
   * @ignore
   */
  LocalProxyMode[LocalProxyMode["LocalOnly"] = 1] = "LocalOnly";
  return LocalProxyMode;
}({});

/**
 * Configuration information of the log server.
 */
export class LogUploadServerInfo {}

/**
 * Advanced options for Local Access Point.
 */
export class AdvancedConfigInfo {}

/**
 * Local Access Point configuration.
 */
export class LocalAccessPointConfiguration {}

/**
 * @ignore
 */
export let RecorderStreamType = /*#__PURE__*/function (RecorderStreamType) {
  /**
   * @ignore
   */
  RecorderStreamType[RecorderStreamType["Rtc"] = 0] = "Rtc";
  /**
   * @ignore
   */
  RecorderStreamType[RecorderStreamType["Preview"] = 1] = "Preview";
  return RecorderStreamType;
}({});

/**
 * @ignore
 */
export class RecorderStreamInfo {}

/**
 * Spatial audio parameters.
 */
export class SpatialAudioParams {}

/**
 * Layout information of a sub-video stream in a composite video.
 */
export class VideoLayout {}
