// react-native-agora/lib/module/IAgoraMusicContentCenter.js
import './extension/IAgoraMusicContentCenterExtension';
import { IMediaPlayer } from './IAgoraMediaPlayer';

/**
 * @ignore
 */
export let MusicPlayMode = /*#__PURE__*/function (MusicPlayMode) {
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeOriginal"] = 0] = "KMusicPlayModeOriginal";
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeAccompany"] = 1] = "KMusicPlayModeAccompany";
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeLeadSing"] = 2] = "KMusicPlayModeLeadSing";
  return MusicPlayMode;
}({});

/**
 * @ignore
 */
export let PreloadState = /*#__PURE__*/function (PreloadState) {
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateCompleted"] = 0] = "KPreloadStateCompleted";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateFailed"] = 1] = "KPreloadStateFailed";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStatePreloading"] = 2] = "KPreloadStatePreloading";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateRemoved"] = 3] = "KPreloadStateRemoved";
  return PreloadState;
}({});

/**
 * @ignore
 */
export let MusicContentCenterStateReason = /*#__PURE__*/function (MusicContentCenterStateReason) {
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonOk"] = 0] = "KMusicContentCenterReasonOk";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonError"] = 1] = "KMusicContentCenterReasonError";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonGateway"] = 2] = "KMusicContentCenterReasonGateway";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonPermissionAndResource"] = 3] = "KMusicContentCenterReasonPermissionAndResource";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonInternalDataParse"] = 4] = "KMusicContentCenterReasonInternalDataParse";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonMusicLoading"] = 5] = "KMusicContentCenterReasonMusicLoading";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonMusicDecryption"] = 6] = "KMusicContentCenterReasonMusicDecryption";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonHttpInternalError"] = 7] = "KMusicContentCenterReasonHttpInternalError";
  return MusicContentCenterStateReason;
}({});

/**
 * @ignore
 */
export class MusicChartInfo {}

/**
 * @ignore
 */
export let MusicCacheStatusType = /*#__PURE__*/function (MusicCacheStatusType) {
  /**
   * @ignore
   */
  MusicCacheStatusType[MusicCacheStatusType["MusicCacheStatusTypeCached"] = 0] = "MusicCacheStatusTypeCached";
  /**
   * @ignore
   */
  MusicCacheStatusType[MusicCacheStatusType["MusicCacheStatusTypeCaching"] = 1] = "MusicCacheStatusTypeCaching";
  return MusicCacheStatusType;
}({});

/**
 * @ignore
 */
export class MusicCacheInfo {}

/**
 * @ignore
 */
export class MusicChartCollection {}

/**
 * @ignore
 */
export class MvProperty {}

/**
 * @ignore
 */
export class ClimaxSegment {}

/**
 * @ignore
 */
export class Music {}

/**
 * @ignore
 */
export class MusicCollection {}

/**
 * @ignore
 */

/**
 * @ignore
 */
export class MusicContentCenterConfiguration {}

/**
 * @ignore
 */
export class IMusicPlayer extends IMediaPlayer {}

/**
 * @ignore
 */
export class IMusicContentCenter {}
