// react-native-agora/lib/module/IAgoraRtcEngine.js
import './extension/IAgoraRtcEngineExtension';
/**
 * Device type.
 */
export let MediaDeviceType = /*#__PURE__*/function (MediaDeviceType) {
  /**
   * -1: Unknown device type.
   */
  MediaDeviceType[MediaDeviceType["UnknownAudioDevice"] = -1] = "UnknownAudioDevice";
  /**
   * 0: Audio playback device.
   */
  MediaDeviceType[MediaDeviceType["AudioPlayoutDevice"] = 0] = "AudioPlayoutDevice";
  /**
   * 1: Audio recording device.
   */
  MediaDeviceType[MediaDeviceType["AudioRecordingDevice"] = 1] = "AudioRecordingDevice";
  /**
   * 2: Video rendering device (graphics card).
   */
  MediaDeviceType[MediaDeviceType["VideoRenderDevice"] = 2] = "VideoRenderDevice";
  /**
   * 3: Video capture device.
   */
  MediaDeviceType[MediaDeviceType["VideoCaptureDevice"] = 3] = "VideoCaptureDevice";
  /**
   * 4: Audio application playback device.
   */
  MediaDeviceType[MediaDeviceType["AudioApplicationPlayoutDevice"] = 4] = "AudioApplicationPlayoutDevice";
  /**
   * 5: Virtual audio playback device (virtual sound card).
   */
  MediaDeviceType[MediaDeviceType["AudioVirtualPlayoutDevice"] = 5] = "AudioVirtualPlayoutDevice";
  /**
   * 6: Virtual audio recording device (virtual sound card).
   */
  MediaDeviceType[MediaDeviceType["AudioVirtualRecordingDevice"] = 6] = "AudioVirtualRecordingDevice";
  return MediaDeviceType;
}({});

/**
 * Music file playback state.
 */
export let AudioMixingStateType = /*#__PURE__*/function (AudioMixingStateType) {
  /**
   * 710: Music file is playing normally.
   */
  AudioMixingStateType[AudioMixingStateType["AudioMixingStatePlaying"] = 710] = "AudioMixingStatePlaying";
  /**
   * 711: Music file playback paused.
   */
  AudioMixingStateType[AudioMixingStateType["AudioMixingStatePaused"] = 711] = "AudioMixingStatePaused";
  /**
   * 713: Music file playback stopped.
   * This state may be caused by the following reasons:
   *  AudioMixingReasonAllLoopsCompleted(723)
   *  AudioMixingReasonStoppedByUser(724)
   */
  AudioMixingStateType[AudioMixingStateType["AudioMixingStateStopped"] = 713] = "AudioMixingStateStopped";
  /**
   * 714: Music file playback error.
   * This state may be caused by the following reasons:
   *  AudioMixingReasonCanNotOpen(701)
   *  AudioMixingReasonTooFrequentCall(702)
   *  AudioMixingReasonInterruptedEof(703)
   */
  AudioMixingStateType[AudioMixingStateType["AudioMixingStateFailed"] = 714] = "AudioMixingStateFailed";
  return AudioMixingStateType;
}({});

/**
 * Reason for music file playback state change. Reported in the onAudioMixingStateChanged callback.
 */
export let AudioMixingReasonType = /*#__PURE__*/function (AudioMixingReasonType) {
  /**
   * 701: Failed to open music file. For example, the local music file does not exist, the file format is not supported, or the online music file URL is inaccessible.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonCanNotOpen"] = 701] = "AudioMixingReasonCanNotOpen";
  /**
   * 702: Music file opened too frequently. If you need to call startAudioMixing multiple times, ensure the interval between calls is more than 500 ms.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonTooFrequentCall"] = 702] = "AudioMixingReasonTooFrequentCall";
  /**
   * 703: Music file playback interrupted.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonInterruptedEof"] = 703] = "AudioMixingReasonInterruptedEof";
  /**
   * 721: One loop of the music file playback completed.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonOneLoopCompleted"] = 721] = "AudioMixingReasonOneLoopCompleted";
  /**
   * 723: All loops of the music file playback completed.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonAllLoopsCompleted"] = 723] = "AudioMixingReasonAllLoopsCompleted";
  /**
   * 724: Successfully called stopAudioMixing to stop music file playback.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonStoppedByUser"] = 724] = "AudioMixingReasonStoppedByUser";
  /**
   * @ignore
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonResumedByUser"] = 726] = "AudioMixingReasonResumedByUser";
  /**
   * 0: Music file opened successfully.
   */
  AudioMixingReasonType[AudioMixingReasonType["AudioMixingReasonOk"] = 0] = "AudioMixingReasonOk";
  return AudioMixingReasonType;
}({});

/**
 * @ignore
 */
export let InjectStreamStatus = /*#__PURE__*/function (InjectStreamStatus) {
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStartSuccess"] = 0] = "InjectStreamStatusStartSuccess";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStartAlreadyExists"] = 1] = "InjectStreamStatusStartAlreadyExists";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStartUnauthorized"] = 2] = "InjectStreamStatusStartUnauthorized";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStartTimedout"] = 3] = "InjectStreamStatusStartTimedout";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStartFailed"] = 4] = "InjectStreamStatusStartFailed";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStopSuccess"] = 5] = "InjectStreamStatusStopSuccess";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStopNotFound"] = 6] = "InjectStreamStatusStopNotFound";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStopUnauthorized"] = 7] = "InjectStreamStatusStopUnauthorized";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStopTimedout"] = 8] = "InjectStreamStatusStopTimedout";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusStopFailed"] = 9] = "InjectStreamStatusStopFailed";
  /**
   * @ignore
   */
  InjectStreamStatus[InjectStreamStatus["InjectStreamStatusBroken"] = 10] = "InjectStreamStatusBroken";
  return InjectStreamStatus;
}({});

/**
 * Center frequency of the voice equalization band.
 */
export let AudioEqualizationBandFrequency = /*#__PURE__*/function (AudioEqualizationBandFrequency) {
  /**
   * 0: 31 Hz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand31"] = 0] = "AudioEqualizationBand31";
  /**
   * 1: 62 Hz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand62"] = 1] = "AudioEqualizationBand62";
  /**
   * 2: 125 Hz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand125"] = 2] = "AudioEqualizationBand125";
  /**
   * 3: 250 Hz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand250"] = 3] = "AudioEqualizationBand250";
  /**
   * 4: 500 Hz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand500"] = 4] = "AudioEqualizationBand500";
  /**
   * 5: 1 kHz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand1k"] = 5] = "AudioEqualizationBand1k";
  /**
   * 6: 2 kHz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand2k"] = 6] = "AudioEqualizationBand2k";
  /**
   * 7: 4 kHz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand4k"] = 7] = "AudioEqualizationBand4k";
  /**
   * 8: 8 kHz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand8k"] = 8] = "AudioEqualizationBand8k";
  /**
   * 9: 16 kHz
   */
  AudioEqualizationBandFrequency[AudioEqualizationBandFrequency["AudioEqualizationBand16k"] = 9] = "AudioEqualizationBand16k";
  return AudioEqualizationBandFrequency;
}({});

/**
 * Audio reverb type.
 */
export let AudioReverbType = /*#__PURE__*/function (AudioReverbType) {
  /**
   * 0: Dry signal level, i.e., the original sound intensity. Range [-20,10], unit: dB.
   */
  AudioReverbType[AudioReverbType["AudioReverbDryLevel"] = 0] = "AudioReverbDryLevel";
  /**
   * 1: Wet signal level, i.e., early reflection signal strength. Range [-20,10], unit: dB.
   */
  AudioReverbType[AudioReverbType["AudioReverbWetLevel"] = 1] = "AudioReverbWetLevel";
  /**
   * 2: Room size for the desired reverb effect. Generally, the larger the room, the stronger the reverb. Range [0,100], unit: dB.
   */
  AudioReverbType[AudioReverbType["AudioReverbRoomSize"] = 2] = "AudioReverbRoomSize";
  /**
   * 3: Initial delay length of the wet signal. Range [0,200], unit: ms.
   */
  AudioReverbType[AudioReverbType["AudioReverbWetDelay"] = 3] = "AudioReverbWetDelay";
  /**
   * 4: Intensity of the sustained reverb. Range [0,100].
   */
  AudioReverbType[AudioReverbType["AudioReverbStrength"] = 4] = "AudioReverbStrength";
  return AudioReverbType;
}({});

/**
 * @ignore
 */
export let StreamFallbackOptions = /*#__PURE__*/function (StreamFallbackOptions) {
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionDisabled"] = 0] = "StreamFallbackOptionDisabled";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLow"] = 1] = "StreamFallbackOptionVideoStreamLow";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionAudioOnly"] = 2] = "StreamFallbackOptionAudioOnly";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer1"] = 3] = "StreamFallbackOptionVideoStreamLayer1";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer2"] = 4] = "StreamFallbackOptionVideoStreamLayer2";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer3"] = 5] = "StreamFallbackOptionVideoStreamLayer3";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer4"] = 6] = "StreamFallbackOptionVideoStreamLayer4";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer5"] = 7] = "StreamFallbackOptionVideoStreamLayer5";
  /**
   * @ignore
   */
  StreamFallbackOptions[StreamFallbackOptions["StreamFallbackOptionVideoStreamLayer6"] = 8] = "StreamFallbackOptionVideoStreamLayer6";
  return StreamFallbackOptions;
}({});

/**
 * @ignore
 */
export let PriorityType = /*#__PURE__*/function (PriorityType) {
  /**
   * @ignore
   */
  PriorityType[PriorityType["PriorityHigh"] = 50] = "PriorityHigh";
  /**
   * @ignore
   */
  PriorityType[PriorityType["PriorityNormal"] = 100] = "PriorityNormal";
  return PriorityType;
}({});

/**
 * Statistics of the local video stream.
 */
export class LocalVideoStats {}

/**
 * Audio statistics of the remote user.
 */
export class RemoteAudioStats {}

/**
 * Statistics of the remote video stream.
 */
export class RemoteVideoStats {}

/**
 * @ignore
 */
export class Region {}

/**
 * @ignore
 */
export class VideoCompositingLayout {}

/**
 * @ignore
 */
export class InjectStreamConfig {}

/**
 * Lifecycle of server-side transcoding streaming.
 *
 * Deprecated Deprecated
 */
export let RtmpStreamLifeCycleType = /*#__PURE__*/function (RtmpStreamLifeCycleType) {
  /**
   * Bound to the channel lifecycle. When all hosts leave the channel, server-side transcoding streaming stops after 30 seconds.
   */
  RtmpStreamLifeCycleType[RtmpStreamLifeCycleType["RtmpStreamLifeCycleBind2channel"] = 1] = "RtmpStreamLifeCycleBind2channel";
  /**
   * Bound to the lifecycle of the host who started the server-side transcoding streaming. When this host leaves, the streaming stops immediately.
   */
  RtmpStreamLifeCycleType[RtmpStreamLifeCycleType["RtmpStreamLifeCycleBind2owner"] = 2] = "RtmpStreamLifeCycleBind2owner";
  return RtmpStreamLifeCycleType;
}({});

/**
 * @ignore
 */
export class PublisherConfiguration {}

/**
 * Camera direction.
 */
export let CameraDirection = /*#__PURE__*/function (CameraDirection) {
  /**
   * 0: Rear camera.
   */
  CameraDirection[CameraDirection["CameraRear"] = 0] = "CameraRear";
  /**
   * 1: (Default) Front camera.
   */
  CameraDirection[CameraDirection["CameraFront"] = 1] = "CameraFront";
  return CameraDirection;
}({});

/**
 * Cloud proxy type.
 */
export let CloudProxyType = /*#__PURE__*/function (CloudProxyType) {
  /**
   * 0: Automatic mode. This is the default mode. In this mode, the SDK first tries to connect via SD-RTN™. If it fails, it automatically switches to TLS 443.
   */
  CloudProxyType[CloudProxyType["NoneProxy"] = 0] = "NoneProxy";
  /**
   * 1: UDP cloud proxy, i.e., Force UDP mode. In this mode, the SDK always transmits data via UDP.
   */
  CloudProxyType[CloudProxyType["UdpProxy"] = 1] = "UdpProxy";
  /**
   * 2: TCP (encrypted) cloud proxy, i.e., Force TCP mode. In this mode, the SDK always transmits data via TLS 443.
   */
  CloudProxyType[CloudProxyType["TcpProxy"] = 2] = "TcpProxy";
  return CloudProxyType;
}({});

/**
 * Camera capture configuration.
 */
export class CameraCapturerConfiguration {}

/**
 * @ignore
 */
export class ScreenCaptureConfiguration {}

/**
 * @ignore
 */
export class Size {}

/**
 * @ignore
 */
export class ThumbImageBuffer {}

/**
 * @ignore
 */
export let ScreenCaptureSourceType = /*#__PURE__*/function (ScreenCaptureSourceType) {
  /**
   * @ignore
   */
  ScreenCaptureSourceType[ScreenCaptureSourceType["ScreencapturesourcetypeUnknown"] = -1] = "ScreencapturesourcetypeUnknown";
  /**
   * @ignore
   */
  ScreenCaptureSourceType[ScreenCaptureSourceType["ScreencapturesourcetypeWindow"] = 0] = "ScreencapturesourcetypeWindow";
  /**
   * @ignore
   */
  ScreenCaptureSourceType[ScreenCaptureSourceType["ScreencapturesourcetypeScreen"] = 1] = "ScreencapturesourcetypeScreen";
  /**
   * @ignore
   */
  ScreenCaptureSourceType[ScreenCaptureSourceType["ScreencapturesourcetypeCustom"] = 2] = "ScreencapturesourcetypeCustom";
  return ScreenCaptureSourceType;
}({});

/**
 * @ignore
 */
export class ScreenCaptureSourceInfo {}

/**
 * Advanced options for audio.
 */
export class AdvancedAudioOptions {}

/**
 * Settings options for placeholder images.
 */
export class ImageTrackOptions {}

/**
 * Channel media configuration options.
 *
 * RtcConnection publishMicrophoneTrack publishCustomAudioTrack publishMediaPlayerAudioTrack true publishCameraTrack publishScreenCaptureVideo, publishCustomVideoTrack publishEncodedVideoTrack true It is recommended that you configure the member parameters based on your business scenario. Otherwise, the SDK will automatically assign values to the member parameters.
 */
export class ChannelMediaOptions {}

/**
 * Proxy type.
 */
export let ProxyType = /*#__PURE__*/function (ProxyType) {
  /**
   * 0: Reserved parameter, not supported yet.
   */
  ProxyType[ProxyType["NoneProxyType"] = 0] = "NoneProxyType";
  /**
   * 1: Cloud proxy using UDP protocol, i.e., Force UDP cloud proxy mode. In this mode, the SDK always transmits data via UDP protocol.
   */
  ProxyType[ProxyType["UdpProxyType"] = 1] = "UdpProxyType";
  /**
   * 2: Cloud proxy using TCP (encrypted) protocol, i.e., Force TCP cloud proxy mode. In this mode, the SDK always transmits data via TLS 443.
   */
  ProxyType[ProxyType["TcpProxyType"] = 2] = "TcpProxyType";
  /**
   * 3: Reserved parameter, not supported yet.
   */
  ProxyType[ProxyType["LocalProxyType"] = 3] = "LocalProxyType";
  /**
   * 4: Auto mode. In this mode, the SDK first tries to connect to SD-RTN™. If the connection fails, it automatically switches to TLS 443.
   */
  ProxyType[ProxyType["TcpProxyAutoFallbackType"] = 4] = "TcpProxyAutoFallbackType";
  /**
   * @ignore
   */
  ProxyType[ProxyType["HttpProxyType"] = 5] = "HttpProxyType";
  /**
   * @ignore
   */
  ProxyType[ProxyType["HttpsProxyType"] = 6] = "HttpsProxyType";
  return ProxyType;
}({});

/**
 * @ignore
 */
export let FeatureType = /*#__PURE__*/function (FeatureType) {
  /**
   * @ignore
   */
  FeatureType[FeatureType["VideoVirtualBackground"] = 1] = "VideoVirtualBackground";
  /**
   * @ignore
   */
  FeatureType[FeatureType["VideoBeautyEffect"] = 2] = "VideoBeautyEffect";
  return FeatureType;
}({});

/**
 * Options for leaving the channel.
 */
export class LeaveChannelOptions {}

/**
 * The IRtcEngineEventHandler interface class is used by the SDK to send event notifications to the App. The App obtains SDK event notifications by inheriting methods of this interface class.
 *
 * All methods of this interface class have default (empty) implementations. The App can choose to inherit only the events it cares about.
 *  In the callback methods, the App should not perform time-consuming operations or call APIs that may cause blocking (such as sendMessage), otherwise it may affect the operation of the SDK.
 *  The SDK no longer catches exceptions in the code logic implemented by the developer in the callbacks of the IRtcEngineEventHandler class. You need to handle such exceptions yourself, otherwise the App may crash when exceptions occur.
 */

/**
 * @ignore
 */
export class IVideoDeviceManager {}

/**
 * Video effect node types.
 *
 * Since Available since v4.6.2.
 */
export let VideoEffectNodeId = /*#__PURE__*/function (VideoEffectNodeId) {
  /**
   * (1): Beauty effect node.
   */
  VideoEffectNodeId[VideoEffectNodeId["Beauty"] = 1] = "Beauty";
  /**
   * (2): Style makeup effect node.
   */
  VideoEffectNodeId[VideoEffectNodeId["StyleMakeup"] = 2] = "StyleMakeup";
  /**
   * (4): Filter effect node.
   */
  VideoEffectNodeId[VideoEffectNodeId["Filter"] = 4] = "Filter";
  /**
   * @ignore
   */
  VideoEffectNodeId[VideoEffectNodeId["Sticker"] = 8] = "Sticker";
  return VideoEffectNodeId;
}({});

/**
 * Action types performed on video effect nodes.
 *
 * Since Available since v4.6.2.
 */
export let VideoEffectAction = /*#__PURE__*/function (VideoEffectAction) {
  /**
   * (1): Save the current parameters of the video effect.
   */
  VideoEffectAction[VideoEffectAction["Save"] = 1] = "Save";
  /**
   * (2): Reset the video effect to default parameters.
   */
  VideoEffectAction[VideoEffectAction["Reset"] = 2] = "Reset";
  return VideoEffectAction;
}({});

/**
 * Used to manage and configure video effects such as beautification, styled makeup, and filters.
 *
 * Since Available since v4.6.2.
 */
export class IVideoEffectObject {}

/**
 * Defines RtcEngineContext.
 */
export class RtcEngineContext {}

/**
 * The Metadata type of the observer. Currently only video-type Metadata is supported.
 */
export let MetadataType = /*#__PURE__*/function (MetadataType) {
  /**
   * -1: Unknown Metadata type.
   */
  MetadataType[MetadataType["UnknownMetadata"] = -1] = "UnknownMetadata";
  /**
   * 0: Metadata type is video.
   */
  MetadataType[MetadataType["VideoMetadata"] = 0] = "VideoMetadata";
  return MetadataType;
}({});

/**
 * @ignore
 */
export let MaxMetadataSizeType = /*#__PURE__*/function (MaxMetadataSizeType) {
  /**
   * @ignore
   */
  MaxMetadataSizeType[MaxMetadataSizeType["InvalidMetadataSizeInByte"] = -1] = "InvalidMetadataSizeInByte";
  /**
   * @ignore
   */
  MaxMetadataSizeType[MaxMetadataSizeType["DefaultMetadataSizeInByte"] = 512] = "DefaultMetadataSizeInByte";
  /**
   * @ignore
   */
  MaxMetadataSizeType[MaxMetadataSizeType["MaxMetadataSizeInByte"] = 1024] = "MaxMetadataSizeInByte";
  return MaxMetadataSizeType;
}({});

/**
 * Media metadata.
 */
export class Metadata {}

/**
 * Metadata observer.
 */

/**
 * @ignore
 */
export let DirectCdnStreamingReason = /*#__PURE__*/function (DirectCdnStreamingReason) {
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonOk"] = 0] = "DirectCdnStreamingReasonOk";
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonFailed"] = 1] = "DirectCdnStreamingReasonFailed";
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonAudioPublication"] = 2] = "DirectCdnStreamingReasonAudioPublication";
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonVideoPublication"] = 3] = "DirectCdnStreamingReasonVideoPublication";
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonNetConnect"] = 4] = "DirectCdnStreamingReasonNetConnect";
  /**
   * @ignore
   */
  DirectCdnStreamingReason[DirectCdnStreamingReason["DirectCdnStreamingReasonBadName"] = 5] = "DirectCdnStreamingReasonBadName";
  return DirectCdnStreamingReason;
}({});

/**
 * @ignore
 */
export let DirectCdnStreamingState = /*#__PURE__*/function (DirectCdnStreamingState) {
  /**
   * @ignore
   */
  DirectCdnStreamingState[DirectCdnStreamingState["DirectCdnStreamingStateIdle"] = 0] = "DirectCdnStreamingStateIdle";
  /**
   * @ignore
   */
  DirectCdnStreamingState[DirectCdnStreamingState["DirectCdnStreamingStateRunning"] = 1] = "DirectCdnStreamingStateRunning";
  /**
   * @ignore
   */
  DirectCdnStreamingState[DirectCdnStreamingState["DirectCdnStreamingStateStopped"] = 2] = "DirectCdnStreamingStateStopped";
  /**
   * @ignore
   */
  DirectCdnStreamingState[DirectCdnStreamingState["DirectCdnStreamingStateFailed"] = 3] = "DirectCdnStreamingStateFailed";
  /**
   * @ignore
   */
  DirectCdnStreamingState[DirectCdnStreamingState["DirectCdnStreamingStateRecovering"] = 4] = "DirectCdnStreamingStateRecovering";
  return DirectCdnStreamingState;
}({});

/**
 * @ignore
 */
export class DirectCdnStreamingStats {}

/**
 * @ignore
 */

/**
 * @ignore
 */
export class DirectCdnStreamingMediaOptions {}

/**
 * @ignore
 */
export class ExtensionInfo {}

/**
 * The base interface class of the RTC SDK that implements the main functions of real-time audio and video.
 *
 * IRtcEngine provides the main methods for the App to call.
 * Before calling other APIs, you must first call createAgoraRtcEngine to create an IRtcEngine object.
 */
export class IRtcEngine {}

/**
 * @ignore
 */
export let QualityReportFormatType = /*#__PURE__*/function (QualityReportFormatType) {
  /**
   * @ignore
   */
  QualityReportFormatType[QualityReportFormatType["QualityReportJson"] = 0] = "QualityReportJson";
  /**
   * @ignore
   */
  QualityReportFormatType[QualityReportFormatType["QualityReportHtml"] = 1] = "QualityReportHtml";
  return QualityReportFormatType;
}({});

/**
 * Device state.
 */
export let MediaDeviceStateType = /*#__PURE__*/function (MediaDeviceStateType) {
  /**
   * 0: Device is ready.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStateIdle"] = 0] = "MediaDeviceStateIdle";
  /**
   * 1: Device is in use.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStateActive"] = 1] = "MediaDeviceStateActive";
  /**
   * 2: Device is disabled.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStateDisabled"] = 2] = "MediaDeviceStateDisabled";
  /**
   * 3: Device is plugged in.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStatePluggedIn"] = 3] = "MediaDeviceStatePluggedIn";
  /**
   * 4: Device is not present.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStateNotPresent"] = 4] = "MediaDeviceStateNotPresent";
  /**
   * 8: Device is unplugged.
   */
  MediaDeviceStateType[MediaDeviceStateType["MediaDeviceStateUnplugged"] = 8] = "MediaDeviceStateUnplugged";
  return MediaDeviceStateType;
}({});

/**
 * @ignore
 */
export let VideoProfileType = /*#__PURE__*/function (VideoProfileType) {
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape120p"] = 0] = "VideoProfileLandscape120p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape120p3"] = 2] = "VideoProfileLandscape120p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape180p"] = 10] = "VideoProfileLandscape180p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape180p3"] = 12] = "VideoProfileLandscape180p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape180p4"] = 13] = "VideoProfileLandscape180p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape240p"] = 20] = "VideoProfileLandscape240p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape240p3"] = 22] = "VideoProfileLandscape240p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape240p4"] = 23] = "VideoProfileLandscape240p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p"] = 30] = "VideoProfileLandscape360p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p3"] = 32] = "VideoProfileLandscape360p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p4"] = 33] = "VideoProfileLandscape360p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p6"] = 35] = "VideoProfileLandscape360p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p7"] = 36] = "VideoProfileLandscape360p7";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p8"] = 37] = "VideoProfileLandscape360p8";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p9"] = 38] = "VideoProfileLandscape360p9";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p10"] = 39] = "VideoProfileLandscape360p10";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape360p11"] = 100] = "VideoProfileLandscape360p11";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p"] = 40] = "VideoProfileLandscape480p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p3"] = 42] = "VideoProfileLandscape480p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p4"] = 43] = "VideoProfileLandscape480p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p6"] = 45] = "VideoProfileLandscape480p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p8"] = 47] = "VideoProfileLandscape480p8";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p9"] = 48] = "VideoProfileLandscape480p9";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape480p10"] = 49] = "VideoProfileLandscape480p10";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape720p"] = 50] = "VideoProfileLandscape720p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape720p3"] = 52] = "VideoProfileLandscape720p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape720p5"] = 54] = "VideoProfileLandscape720p5";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape720p6"] = 55] = "VideoProfileLandscape720p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape1080p"] = 60] = "VideoProfileLandscape1080p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape1080p3"] = 62] = "VideoProfileLandscape1080p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape1080p5"] = 64] = "VideoProfileLandscape1080p5";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape1440p"] = 66] = "VideoProfileLandscape1440p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape1440p2"] = 67] = "VideoProfileLandscape1440p2";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape4k"] = 70] = "VideoProfileLandscape4k";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileLandscape4k3"] = 72] = "VideoProfileLandscape4k3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait120p"] = 1000] = "VideoProfilePortrait120p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait120p3"] = 1002] = "VideoProfilePortrait120p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait180p"] = 1010] = "VideoProfilePortrait180p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait180p3"] = 1012] = "VideoProfilePortrait180p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait180p4"] = 1013] = "VideoProfilePortrait180p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait240p"] = 1020] = "VideoProfilePortrait240p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait240p3"] = 1022] = "VideoProfilePortrait240p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait240p4"] = 1023] = "VideoProfilePortrait240p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p"] = 1030] = "VideoProfilePortrait360p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p3"] = 1032] = "VideoProfilePortrait360p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p4"] = 1033] = "VideoProfilePortrait360p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p6"] = 1035] = "VideoProfilePortrait360p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p7"] = 1036] = "VideoProfilePortrait360p7";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p8"] = 1037] = "VideoProfilePortrait360p8";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p9"] = 1038] = "VideoProfilePortrait360p9";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p10"] = 1039] = "VideoProfilePortrait360p10";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait360p11"] = 1100] = "VideoProfilePortrait360p11";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p"] = 1040] = "VideoProfilePortrait480p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p3"] = 1042] = "VideoProfilePortrait480p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p4"] = 1043] = "VideoProfilePortrait480p4";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p6"] = 1045] = "VideoProfilePortrait480p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p8"] = 1047] = "VideoProfilePortrait480p8";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p9"] = 1048] = "VideoProfilePortrait480p9";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait480p10"] = 1049] = "VideoProfilePortrait480p10";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait720p"] = 1050] = "VideoProfilePortrait720p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait720p3"] = 1052] = "VideoProfilePortrait720p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait720p5"] = 1054] = "VideoProfilePortrait720p5";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait720p6"] = 1055] = "VideoProfilePortrait720p6";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait1080p"] = 1060] = "VideoProfilePortrait1080p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait1080p3"] = 1062] = "VideoProfilePortrait1080p3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait1080p5"] = 1064] = "VideoProfilePortrait1080p5";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait1440p"] = 1066] = "VideoProfilePortrait1440p";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait1440p2"] = 1067] = "VideoProfilePortrait1440p2";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait4k"] = 1070] = "VideoProfilePortrait4k";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfilePortrait4k3"] = 1072] = "VideoProfilePortrait4k3";
  /**
   * @ignore
   */
  VideoProfileType[VideoProfileType["VideoProfileDefault"] = 30] = "VideoProfileDefault";
  return VideoProfileType;
}({});

/**
 * SDK version information.
 */
export class SDKBuildInfo {}

/**
 * The VideoDeviceInfo class contains the video device ID and device name.
 */
export class VideoDeviceInfo {}

/**
 * The AudioDeviceInfo class contains the audio device ID and device name.
 */
export class AudioDeviceInfo {}
