// react-native-agora/lib/module/IAgoraMediaStreamingSource.js
import './extension/IAgoraMediaStreamingSourceExtension';

/* enum_StreamingSrcErr */
export let StreamingSrcErr = /*#__PURE__*/function (StreamingSrcErr) {
  /* enum_StreamingSrcErr_StreamingSrcErrNone */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrNone"] = 0] = "StreamingSrcErrNone";
  /* enum_StreamingSrcErr_StreamingSrcErrUnknown */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrUnknown"] = 1] = "StreamingSrcErrUnknown";
  /* enum_StreamingSrcErr_StreamingSrcErrInvalidParam */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrInvalidParam"] = 2] = "StreamingSrcErrInvalidParam";
  /* enum_StreamingSrcErr_StreamingSrcErrBadState */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrBadState"] = 3] = "StreamingSrcErrBadState";
  /* enum_StreamingSrcErr_StreamingSrcErrNoMem */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrNoMem"] = 4] = "StreamingSrcErrNoMem";
  /* enum_StreamingSrcErr_StreamingSrcErrBufferOverflow */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrBufferOverflow"] = 5] = "StreamingSrcErrBufferOverflow";
  /* enum_StreamingSrcErr_StreamingSrcErrBufferUnderflow */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrBufferUnderflow"] = 6] = "StreamingSrcErrBufferUnderflow";
  /* enum_StreamingSrcErr_StreamingSrcErrNotFound */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrNotFound"] = 7] = "StreamingSrcErrNotFound";
  /* enum_StreamingSrcErr_StreamingSrcErrTimeout */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrTimeout"] = 8] = "StreamingSrcErrTimeout";
  /* enum_StreamingSrcErr_StreamingSrcErrExpired */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrExpired"] = 9] = "StreamingSrcErrExpired";
  /* enum_StreamingSrcErr_StreamingSrcErrUnsupported */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrUnsupported"] = 10] = "StreamingSrcErrUnsupported";
  /* enum_StreamingSrcErr_StreamingSrcErrNotExist */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrNotExist"] = 11] = "StreamingSrcErrNotExist";
  /* enum_StreamingSrcErr_StreamingSrcErrExist */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrExist"] = 12] = "StreamingSrcErrExist";
  /* enum_StreamingSrcErr_StreamingSrcErrOpen */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrOpen"] = 13] = "StreamingSrcErrOpen";
  /* enum_StreamingSrcErr_StreamingSrcErrClose */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrClose"] = 14] = "StreamingSrcErrClose";
  /* enum_StreamingSrcErr_StreamingSrcErrRead */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrRead"] = 15] = "StreamingSrcErrRead";
  /* enum_StreamingSrcErr_StreamingSrcErrWrite */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrWrite"] = 16] = "StreamingSrcErrWrite";
  /* enum_StreamingSrcErr_StreamingSrcErrSeek */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrSeek"] = 17] = "StreamingSrcErrSeek";
  /* enum_StreamingSrcErr_StreamingSrcErrEof */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrEof"] = 18] = "StreamingSrcErrEof";
  /* enum_StreamingSrcErr_StreamingSrcErrCodecopen */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrCodecopen"] = 19] = "StreamingSrcErrCodecopen";
  /* enum_StreamingSrcErr_StreamingSrcErrCodecclose */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrCodecclose"] = 20] = "StreamingSrcErrCodecclose";
  /* enum_StreamingSrcErr_StreamingSrcErrCodecproc */
  StreamingSrcErr[StreamingSrcErr["StreamingSrcErrCodecproc"] = 21] = "StreamingSrcErrCodecproc";
  return StreamingSrcErr;
}({});

/* enum_StreamingSrcState */
export let StreamingSrcState = /*#__PURE__*/function (StreamingSrcState) {
  /* enum_StreamingSrcState_StreamingSrcStateClosed */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateClosed"] = 0] = "StreamingSrcStateClosed";
  /* enum_StreamingSrcState_StreamingSrcStateOpening */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateOpening"] = 1] = "StreamingSrcStateOpening";
  /* enum_StreamingSrcState_StreamingSrcStateIdle */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateIdle"] = 2] = "StreamingSrcStateIdle";
  /* enum_StreamingSrcState_StreamingSrcStatePlaying */
  StreamingSrcState[StreamingSrcState["StreamingSrcStatePlaying"] = 3] = "StreamingSrcStatePlaying";
  /* enum_StreamingSrcState_StreamingSrcStateSeeking */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateSeeking"] = 4] = "StreamingSrcStateSeeking";
  /* enum_StreamingSrcState_StreamingSrcStateEof */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateEof"] = 5] = "StreamingSrcStateEof";
  /* enum_StreamingSrcState_StreamingSrcStateError */
  StreamingSrcState[StreamingSrcState["StreamingSrcStateError"] = 6] = "StreamingSrcStateError";
  return StreamingSrcState;
}({});

/* class_InputSeiData */
export class InputSeiData {}
