// react-native-agora/lib/module/IAgoraPip.js
/**
 * Layout configuration for picture-in-picture video streams.
 *
 * Since Available since v4.6.2. This class defines how multiple video streams are arranged in a flowing layout, from left to right and top to bottom.
 */
export class AgoraPipContentViewLayout {}

/**
 * Configuration options for Agora picture-in-picture mode.
 *
 * Since Available since v4.6.2. This class provides platform-specific options to configure picture-in-picture behavior on Android and iOS.
 */
export class AgoraPipOptions {}

/**
 * Represents the current state of Picture-in-Picture mode.
 *
 * Since Available since v4.6.2.
 */
export let AgoraPipState = /*#__PURE__*/function (AgoraPipState) {
  /**
   * 0: Picture-in-Picture mode has started successfully.
   */
  AgoraPipState[AgoraPipState["pipStateStarted"] = 0] = "pipStateStarted";
  /**
   * 1: Picture-in-Picture mode has stopped.
   */
  AgoraPipState[AgoraPipState["pipStateStopped"] = 1] = "pipStateStopped";
  /**
   * 2: Failed to start Picture-in-Picture mode or encountered an error.
   */
  AgoraPipState[AgoraPipState["pipStateFailed"] = 2] = "pipStateFailed";
  return AgoraPipState;
}({});

/**
 * Observer for picture-in-picture state changes.
 *
 * Since Available since v4.6.2. Implement this class to receive notifications of picture-in-picture state transitions and potential errors.
 */

/**
 * Controller interface for managing picture-in-picture functionality.
 *
 * Since Available since v4.6.2. This abstract class defines methods required to control picture-in-picture mode, including setup, state management, and lifecycle operations.
 */
export class AgoraPip {}
