// react-native-agora/lib/commonjs/AgoraMediaPlayerTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SrcInfo = exports.PlayerUpdatedInfo = exports.PlayerStreamInfo = exports.PlayerPreloadEvent = exports.PlayerPlaybackStats = exports.MediaStreamType = exports.MediaSource = exports.MediaPlayerState = exports.MediaPlayerReason = exports.MediaPlayerMetadataType = exports.MediaPlayerEvent = exports.CacheStatistics = void 0;
require("./extension/AgoraMediaPlayerTypesExtension");
/**
 * The state of the media player.
 */
let MediaPlayerState = exports.MediaPlayerState = /*#__PURE__*/function (MediaPlayerState) {
  /**
   * 0: Default state. The player returns this state code before you open a media file and after playback ends.
   */
  MediaPlayerState[MediaPlayerState["PlayerStateIdle"] = 0] = "PlayerStateIdle";
  /**
   * 1: Opening the media file.
   */
  MediaPlayerState[MediaPlayerState["PlayerStateOpening"] = 1] = "PlayerStateOpening";
  /**
   * 2: Media file opened successfully.
   */
  MediaPlayerState[MediaPlayerState["PlayerStateOpenCompleted"] = 2] = "PlayerStateOpenCompleted";
  /**
   * 3: Playing.
   */
  MediaPlayerState[MediaPlayerState["PlayerStatePlaying"] = 3] = "PlayerStatePlaying";
  /**
   * 4: Playback paused.
   */
  MediaPlayerState[MediaPlayerState["PlayerStatePaused"] = 4] = "PlayerStatePaused";
  /**
   * 5: Playback completed.
   */
  MediaPlayerState[MediaPlayerState["PlayerStatePlaybackCompleted"] = 5] = "PlayerStatePlaybackCompleted";
  /**
   * 6: All loops of playback completed.
   */
  MediaPlayerState[MediaPlayerState["PlayerStatePlaybackAllLoopsCompleted"] = 6] = "PlayerStatePlaybackAllLoopsCompleted";
  /**
   * 7: Playback stopped.
   */
  MediaPlayerState[MediaPlayerState["PlayerStateStopped"] = 7] = "PlayerStateStopped";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStatePausingInternal"] = 50] = "PlayerStatePausingInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateStoppingInternal"] = 51] = "PlayerStateStoppingInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateSeekingInternal"] = 52] = "PlayerStateSeekingInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateGettingInternal"] = 53] = "PlayerStateGettingInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateNoneInternal"] = 54] = "PlayerStateNoneInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateDoNothingInternal"] = 55] = "PlayerStateDoNothingInternal";
  /**
   * @ignore
   */
  MediaPlayerState[MediaPlayerState["PlayerStateSetTrackInternal"] = 56] = "PlayerStateSetTrackInternal";
  /**
   * 100: Playback failed.
   */
  MediaPlayerState[MediaPlayerState["PlayerStateFailed"] = 100] = "PlayerStateFailed";
  return MediaPlayerState;
}({});
/**
 * Reason for media player state change.
 */
let MediaPlayerReason = exports.MediaPlayerReason = /*#__PURE__*/function (MediaPlayerReason) {
  /**
   * 0: No error.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonNone"] = 0] = "PlayerReasonNone";
  /**
   * -1: Invalid arguments.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInvalidArguments"] = -1] = "PlayerReasonInvalidArguments";
  /**
   * -2: Internal error.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInternal"] = -2] = "PlayerReasonInternal";
  /**
   * -3: No resource.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonNoResource"] = -3] = "PlayerReasonNoResource";
  /**
   * -4: Invalid resource.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInvalidMediaSource"] = -4] = "PlayerReasonInvalidMediaSource";
  /**
   * -5: Unknown media stream type.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonUnknownStreamType"] = -5] = "PlayerReasonUnknownStreamType";
  /**
   * -6: Object not initialized.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonObjNotInitialized"] = -6] = "PlayerReasonObjNotInitialized";
  /**
   * -7: Codec not supported.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonCodecNotSupported"] = -7] = "PlayerReasonCodecNotSupported";
  /**
   * -8: Invalid renderer.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonVideoRenderFailed"] = -8] = "PlayerReasonVideoRenderFailed";
  /**
   * -9: Invalid internal player state.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInvalidState"] = -9] = "PlayerReasonInvalidState";
  /**
   * -10: URL not found.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonUrlNotFound"] = -10] = "PlayerReasonUrlNotFound";
  /**
   * -11: Invalid connection between the player and Agora server.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInvalidConnectionState"] = -11] = "PlayerReasonInvalidConnectionState";
  /**
   * -12: Insufficient data in the playback buffer.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonSrcBufferUnderflow"] = -12] = "PlayerReasonSrcBufferUnderflow";
  /**
   * -13: Playback was interrupted abnormally and ended.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonInterrupted"] = -13] = "PlayerReasonInterrupted";
  /**
   * -14: Unsupported API call by the SDK.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonNotSupported"] = -14] = "PlayerReasonNotSupported";
  /**
   * -15: Authentication information for the media resource network path has expired.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonTokenExpired"] = -15] = "PlayerReasonTokenExpired";
  /**
   * @ignore
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonIpExpired"] = -16] = "PlayerReasonIpExpired";
  /**
   * -17: Unknown error.
   */
  MediaPlayerReason[MediaPlayerReason["PlayerReasonUnknown"] = -17] = "PlayerReasonUnknown";
  return MediaPlayerReason;
}({});
/**
 * The type of media stream.
 */
let MediaStreamType = exports.MediaStreamType = /*#__PURE__*/function (MediaStreamType) {
  /**
   * 0: Unknown type.
   */
  MediaStreamType[MediaStreamType["StreamTypeUnknown"] = 0] = "StreamTypeUnknown";
  /**
   * 1: Video stream.
   */
  MediaStreamType[MediaStreamType["StreamTypeVideo"] = 1] = "StreamTypeVideo";
  /**
   * 2: Audio stream.
   */
  MediaStreamType[MediaStreamType["StreamTypeAudio"] = 2] = "StreamTypeAudio";
  /**
   * 3: Subtitle stream.
   */
  MediaStreamType[MediaStreamType["StreamTypeSubtitle"] = 3] = "StreamTypeSubtitle";
  return MediaStreamType;
}({});
/**
 * Media player events.
 */
let MediaPlayerEvent = exports.MediaPlayerEvent = /*#__PURE__*/function (MediaPlayerEvent) {
  /**
   * 0: Seek started.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekBegin"] = 0] = "PlayerEventSeekBegin";
  /**
   * 1: Seek completed.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekComplete"] = 1] = "PlayerEventSeekComplete";
  /**
   * 2: Seek error.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSeekError"] = 2] = "PlayerEventSeekError";
  /**
   * 5: Current audio track changed.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventAudioTrackChanged"] = 5] = "PlayerEventAudioTrackChanged";
  /**
   * 6: Current buffer is insufficient for playback.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventBufferLow"] = 6] = "PlayerEventBufferLow";
  /**
   * 7: Current buffer is just enough for playback.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventBufferRecover"] = 7] = "PlayerEventBufferRecover";
  /**
   * 8: Audio or video stutter occurred.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFreezeStart"] = 8] = "PlayerEventFreezeStart";
  /**
   * 9: Audio and video stutter stopped.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFreezeStop"] = 9] = "PlayerEventFreezeStop";
  /**
   * 10: Media resource switching started.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchBegin"] = 10] = "PlayerEventSwitchBegin";
  /**
   * 11: Media resource switching completed.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchComplete"] = 11] = "PlayerEventSwitchComplete";
  /**
   * 12: Media resource switching error.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventSwitchError"] = 12] = "PlayerEventSwitchError";
  /**
   * 13: First video frame displayed.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventFirstDisplayed"] = 13] = "PlayerEventFirstDisplayed";
  /**
   * 14: Reached the maximum number of cacheable files.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventReachCacheFileMaxCount"] = 14] = "PlayerEventReachCacheFileMaxCount";
  /**
   * 15: Reached the maximum size of cacheable files.
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventReachCacheFileMaxSize"] = 15] = "PlayerEventReachCacheFileMaxSize";
  /**
   * @ignore
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenStart"] = 16] = "PlayerEventTryOpenStart";
  /**
   * @ignore
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenSucceed"] = 17] = "PlayerEventTryOpenSucceed";
  /**
   * @ignore
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventTryOpenFailed"] = 18] = "PlayerEventTryOpenFailed";
  /**
   * @ignore
   */
  MediaPlayerEvent[MediaPlayerEvent["PlayerEventHttpRedirect"] = 19] = "PlayerEventHttpRedirect";
  return MediaPlayerEvent;
}({});
/**
 * Events that occur when preloading media resources.
 */
let PlayerPreloadEvent = exports.PlayerPreloadEvent = /*#__PURE__*/function (PlayerPreloadEvent) {
  /**
   * 0: Start preloading media resources.
   */
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventBegin"] = 0] = "PlayerPreloadEventBegin";
  /**
   * 1: Preloading of media resources completed.
   */
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventComplete"] = 1] = "PlayerPreloadEventComplete";
  /**
   * 2: Error occurred while preloading media resources.
   */
  PlayerPreloadEvent[PlayerPreloadEvent["PlayerPreloadEventError"] = 2] = "PlayerPreloadEventError";
  return PlayerPreloadEvent;
}({});
/**
 * All information about the player media stream.
 */
class PlayerStreamInfo {}

/**
 * Video bitrate information during media playback.
 */
exports.PlayerStreamInfo = PlayerStreamInfo;
class SrcInfo {}

/**
 * Media metadata type.
 */
exports.SrcInfo = SrcInfo;
let MediaPlayerMetadataType = exports.MediaPlayerMetadataType = /*#__PURE__*/function (MediaPlayerMetadataType) {
  /**
   * 0: Unknown type.
   */
  MediaPlayerMetadataType[MediaPlayerMetadataType["PlayerMetadataTypeUnknown"] = 0] = "PlayerMetadataTypeUnknown";
  /**
   * 1: SEI (Supplemental Enhancement Information) type.
   */
  MediaPlayerMetadataType[MediaPlayerMetadataType["PlayerMetadataTypeSei"] = 1] = "PlayerMetadataTypeSei";
  return MediaPlayerMetadataType;
}({});
/**
 * Statistics of cached files.
 */
class CacheStatistics {}

/**
 * Information about the currently playing media resource.
 */
exports.CacheStatistics = CacheStatistics;
class PlayerPlaybackStats {}

/**
 * Information related to the media player.
 */
exports.PlayerPlaybackStats = PlayerPlaybackStats;
class PlayerUpdatedInfo {}

/**
 * Information and playback settings for the media file to be played.
 */
exports.PlayerUpdatedInfo = PlayerUpdatedInfo;
class MediaSource {}
exports.MediaSource = MediaSource;
