// react-native-agora/lib/commonjs/IAgoraLog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogLevel = exports.LogFilterType = exports.LogConfig = void 0;
require("./extension/IAgoraLogExtension");
/**
 * Log output level.
 */
let LogLevel = exports.LogLevel = /*#__PURE__*/function (LogLevel) {
  /**
   * 0: No log output.
   */
  LogLevel[LogLevel["LogLevelNone"] = 0] = "LogLevelNone";
  /**
   * 0x0001: (Default) Outputs logs at FATAL, ERROR, WARN, and INFO levels. It is recommended to set the log level to this.
   */
  LogLevel[LogLevel["LogLevelInfo"] = 1] = "LogLevelInfo";
  /**
   * 0x0002: Outputs logs at FATAL, ERROR, and WARN levels only.
   */
  LogLevel[LogLevel["LogLevelWarn"] = 2] = "LogLevelWarn";
  /**
   * 0x0004: Outputs logs at FATAL and ERROR levels only.
   */
  LogLevel[LogLevel["LogLevelError"] = 4] = "LogLevelError";
  /**
   * 0x0008: Outputs logs at FATAL level only.
   */
  LogLevel[LogLevel["LogLevelFatal"] = 8] = "LogLevelFatal";
  /**
   * @ignore
   */
  LogLevel[LogLevel["LogLevelApiCall"] = 16] = "LogLevelApiCall";
  /**
   * @ignore
   */
  LogLevel[LogLevel["LogLevelDebug"] = 32] = "LogLevelDebug";
  return LogLevel;
}({});
/**
 * Log filter level.
 */
let LogFilterType = exports.LogFilterType = /*#__PURE__*/function (LogFilterType) {
  /**
   * 0: No log output.
   */
  LogFilterType[LogFilterType["LogFilterOff"] = 0] = "LogFilterOff";
  /**
   * 0x080f: Outputs all API log information. Set the log level to this if you want the most complete logs.
   */
  LogFilterType[LogFilterType["LogFilterDebug"] = 2063] = "LogFilterDebug";
  /**
   * 0x000f: Outputs logs at LogFilterCritical, LogFilterError, LogFilterWarn, and LogFilterInfo levels. It is recommended to set the log level to this.
   */
  LogFilterType[LogFilterType["LogFilterInfo"] = 15] = "LogFilterInfo";
  /**
   * 0x000e: Outputs logs at LogFilterCritical, LogFilterError, and LogFilterWarn levels.
   */
  LogFilterType[LogFilterType["LogFilterWarn"] = 14] = "LogFilterWarn";
  /**
   * 0x000c: Outputs logs at LogFilterCritical and LogFilterError levels.
   */
  LogFilterType[LogFilterType["LogFilterError"] = 12] = "LogFilterError";
  /**
   * 0x0008: Outputs logs at LogFilterCritical level only.
   */
  LogFilterType[LogFilterType["LogFilterCritical"] = 8] = "LogFilterCritical";
  /**
   * @ignore
   */
  LogFilterType[LogFilterType["LogFilterMask"] = 2063] = "LogFilterMask";
  return LogFilterType;
}({});
/**
 * Configuration of the SDK log file.
 */
class LogConfig {}
exports.LogConfig = LogConfig;
