// react-native-agora/lib/commonjs/IAgoraMusicContentCenter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PreloadState = exports.MvProperty = exports.MusicPlayMode = exports.MusicContentCenterStateReason = exports.MusicContentCenterConfiguration = exports.MusicCollection = exports.MusicChartInfo = exports.MusicChartCollection = exports.MusicCacheStatusType = exports.MusicCacheInfo = exports.Music = exports.IMusicPlayer = exports.IMusicContentCenter = exports.ClimaxSegment = void 0;
require("./extension/IAgoraMusicContentCenterExtension");
var _IAgoraMediaPlayer = require("./IAgoraMediaPlayer");
/**
 * @ignore
 */
let MusicPlayMode = exports.MusicPlayMode = /*#__PURE__*/function (MusicPlayMode) {
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeOriginal"] = 0] = "KMusicPlayModeOriginal";
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeAccompany"] = 1] = "KMusicPlayModeAccompany";
  /**
   * @ignore
   */
  MusicPlayMode[MusicPlayMode["KMusicPlayModeLeadSing"] = 2] = "KMusicPlayModeLeadSing";
  return MusicPlayMode;
}({});
/**
 * @ignore
 */
let PreloadState = exports.PreloadState = /*#__PURE__*/function (PreloadState) {
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateCompleted"] = 0] = "KPreloadStateCompleted";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateFailed"] = 1] = "KPreloadStateFailed";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStatePreloading"] = 2] = "KPreloadStatePreloading";
  /**
   * @ignore
   */
  PreloadState[PreloadState["KPreloadStateRemoved"] = 3] = "KPreloadStateRemoved";
  return PreloadState;
}({});
/**
 * @ignore
 */
let MusicContentCenterStateReason = exports.MusicContentCenterStateReason = /*#__PURE__*/function (MusicContentCenterStateReason) {
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonOk"] = 0] = "KMusicContentCenterReasonOk";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonError"] = 1] = "KMusicContentCenterReasonError";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonGateway"] = 2] = "KMusicContentCenterReasonGateway";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonPermissionAndResource"] = 3] = "KMusicContentCenterReasonPermissionAndResource";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonInternalDataParse"] = 4] = "KMusicContentCenterReasonInternalDataParse";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonMusicLoading"] = 5] = "KMusicContentCenterReasonMusicLoading";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonMusicDecryption"] = 6] = "KMusicContentCenterReasonMusicDecryption";
  /**
   * @ignore
   */
  MusicContentCenterStateReason[MusicContentCenterStateReason["KMusicContentCenterReasonHttpInternalError"] = 7] = "KMusicContentCenterReasonHttpInternalError";
  return MusicContentCenterStateReason;
}({});
/**
 * @ignore
 */
class MusicChartInfo {}

/**
 * @ignore
 */
exports.MusicChartInfo = MusicChartInfo;
let MusicCacheStatusType = exports.MusicCacheStatusType = /*#__PURE__*/function (MusicCacheStatusType) {
  /**
   * @ignore
   */
  MusicCacheStatusType[MusicCacheStatusType["MusicCacheStatusTypeCached"] = 0] = "MusicCacheStatusTypeCached";
  /**
   * @ignore
   */
  MusicCacheStatusType[MusicCacheStatusType["MusicCacheStatusTypeCaching"] = 1] = "MusicCacheStatusTypeCaching";
  return MusicCacheStatusType;
}({});
/**
 * @ignore
 */
class MusicCacheInfo {}

/**
 * @ignore
 */
exports.MusicCacheInfo = MusicCacheInfo;
class MusicChartCollection {}

/**
 * @ignore
 */
exports.MusicChartCollection = MusicChartCollection;
class MvProperty {}

/**
 * @ignore
 */
exports.MvProperty = MvProperty;
class ClimaxSegment {}

/**
 * @ignore
 */
exports.ClimaxSegment = ClimaxSegment;
class Music {}

/**
 * @ignore
 */
exports.Music = Music;
class MusicCollection {}

/**
 * @ignore
 */
exports.MusicCollection = MusicCollection;
/**
 * @ignore
 */
class MusicContentCenterConfiguration {}

/**
 * @ignore
 */
exports.MusicContentCenterConfiguration = MusicContentCenterConfiguration;
class IMusicPlayer extends _IAgoraMediaPlayer.IMediaPlayer {}

/**
 * @ignore
 */
exports.IMusicPlayer = IMusicPlayer;
class IMusicContentCenter {}
exports.IMusicContentCenter = IMusicContentCenter;
