// react-native-agora/lib/commonjs/IAgoraRhythmPlayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RhythmPlayerStateType = exports.RhythmPlayerReason = exports.AgoraRhythmPlayerConfig = void 0;
require("./extension/IAgoraRhythmPlayerExtension");
/**
 * Virtual metronome state.
 */
let RhythmPlayerStateType = exports.RhythmPlayerStateType = /*#__PURE__*/function (RhythmPlayerStateType) {
  /**
   * 810: The virtual metronome is not started or has been stopped.
   */
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateIdle"] = 810] = "RhythmPlayerStateIdle";
  /**
   * 811: Opening the rhythm audio file.
   */
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateOpening"] = 811] = "RhythmPlayerStateOpening";
  /**
   * 812: Decoding the rhythm audio file.
   */
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateDecoding"] = 812] = "RhythmPlayerStateDecoding";
  /**
   * 813: Playing the rhythm audio file.
   */
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStatePlaying"] = 813] = "RhythmPlayerStatePlaying";
  /**
   * 814: Failed to start the virtual metronome. You can troubleshoot using the reported error code errorCode or try starting the virtual metronome again.
   */
  RhythmPlayerStateType[RhythmPlayerStateType["RhythmPlayerStateFailed"] = 814] = "RhythmPlayerStateFailed";
  return RhythmPlayerStateType;
}({});
/**
 * Error information for the virtual metronome.
 */
let RhythmPlayerReason = exports.RhythmPlayerReason = /*#__PURE__*/function (RhythmPlayerReason) {
  /**
   * 0: The metronome audio file is playing normally, no error.
   */
  RhythmPlayerReason[RhythmPlayerReason["RhythmPlayerReasonOk"] = 0] = "RhythmPlayerReasonOk";
  /**
   * 1: General error with no specific reason.
   */
  RhythmPlayerReason[RhythmPlayerReason["RhythmPlayerReasonFailed"] = 1] = "RhythmPlayerReasonFailed";
  /**
   * 801: Failed to open the metronome audio file.
   */
  RhythmPlayerReason[RhythmPlayerReason["RhythmPlayerReasonCanNotOpen"] = 801] = "RhythmPlayerReasonCanNotOpen";
  /**
   * 802: Failed to play the metronome audio file.
   */
  RhythmPlayerReason[RhythmPlayerReason["RhythmPlayerReasonCanNotPlay"] = 802] = "RhythmPlayerReasonCanNotPlay";
  /**
   * 803: The duration of the metronome audio file exceeds the limit. The maximum duration is 1.2 seconds.
   */
  RhythmPlayerReason[RhythmPlayerReason["RhythmPlayerReasonFileOverDurationLimit"] = 803] = "RhythmPlayerReasonFileOverDurationLimit";
  return RhythmPlayerReason;
}({});
/**
 * Virtual metronome configuration.
 */
class AgoraRhythmPlayerConfig {}
exports.AgoraRhythmPlayerConfig = AgoraRhythmPlayerConfig;
