// react-native-agora/lib/commonjs/IAgoraH265Transcoder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IH265Transcoder = exports.H265TranscodeResult = void 0;
require("./extension/IAgoraH265TranscoderExtension");
/* enum_H265TranscodeResult */
let H265TranscodeResult = exports.H265TranscodeResult = /*#__PURE__*/function (H265TranscodeResult) {
  /* enum_H265TranscodeResult_H265TranscodeResultUnknown */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultUnknown"] = -1] = "H265TranscodeResultUnknown";
  /* enum_H265TranscodeResult_H265TranscodeResultSuccess */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultSuccess"] = 0] = "H265TranscodeResultSuccess";
  /* enum_H265TranscodeResult_H265TranscodeResultRequestInvalid */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultRequestInvalid"] = 1] = "H265TranscodeResultRequestInvalid";
  /* enum_H265TranscodeResult_H265TranscodeResultUnauthorized */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultUnauthorized"] = 2] = "H265TranscodeResultUnauthorized";
  /* enum_H265TranscodeResult_H265TranscodeResultTokenExpired */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultTokenExpired"] = 3] = "H265TranscodeResultTokenExpired";
  /* enum_H265TranscodeResult_H265TranscodeResultForbidden */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultForbidden"] = 4] = "H265TranscodeResultForbidden";
  /* enum_H265TranscodeResult_H265TranscodeResultNotFound */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultNotFound"] = 5] = "H265TranscodeResultNotFound";
  /* enum_H265TranscodeResult_H265TranscodeResultConflicted */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultConflicted"] = 6] = "H265TranscodeResultConflicted";
  /* enum_H265TranscodeResult_H265TranscodeResultNotSupported */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultNotSupported"] = 7] = "H265TranscodeResultNotSupported";
  /* enum_H265TranscodeResult_H265TranscodeResultTooOften */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultTooOften"] = 8] = "H265TranscodeResultTooOften";
  /* enum_H265TranscodeResult_H265TranscodeResultServerInternalError */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultServerInternalError"] = 9] = "H265TranscodeResultServerInternalError";
  /* enum_H265TranscodeResult_H265TranscodeResultServiceUnavailable */
  H265TranscodeResult[H265TranscodeResult["H265TranscodeResultServiceUnavailable"] = 10] = "H265TranscodeResultServiceUnavailable";
  return H265TranscodeResult;
}({});
/* class_IH265TranscoderObserver */
/* class_IH265Transcoder */
class IH265Transcoder {}
exports.IH265Transcoder = IH265Transcoder;
