// react-native-agora/lib/commonjs/IAgoraMediaEngine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IMediaEngine = exports.AudioMixingDualMonoMode = void 0;
require("./extension/IAgoraMediaEngineExtension");
/**
 * Channel mode.
 */
let AudioMixingDualMonoMode = exports.AudioMixingDualMonoMode = /*#__PURE__*/function (AudioMixingDualMonoMode) {
  /**
   * @ignore
   */
  AudioMixingDualMonoMode[AudioMixingDualMonoMode["AudioMixingDualMonoAuto"] = 0] = "AudioMixingDualMonoAuto";
  /**
   * 1: Left channel mode. This mode replaces the right channel audio with the left channel audio, so the user only hears the left channel audio.
   */
  AudioMixingDualMonoMode[AudioMixingDualMonoMode["AudioMixingDualMonoL"] = 1] = "AudioMixingDualMonoL";
  /**
   * 2: Right channel mode. This mode replaces the left channel audio with the right channel audio, so the user only hears the right channel audio.
   */
  AudioMixingDualMonoMode[AudioMixingDualMonoMode["AudioMixingDualMonoR"] = 2] = "AudioMixingDualMonoR";
  /**
   * 3: Mixed mode. This mode overlays the left and right channel data, so the user hears both left and right channel audio simultaneously.
   */
  AudioMixingDualMonoMode[AudioMixingDualMonoMode["AudioMixingDualMonoMix"] = 3] = "AudioMixingDualMonoMix";
  return AudioMixingDualMonoMode;
}({});
/**
 * IMediaEngine class.
 */
class IMediaEngine {}
exports.IMediaEngine = IMediaEngine;
