// react-native-agora/lib/commonjs/ti/AgoraMediaBase-ti.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.IVideoFrameObserver = exports.IVideoEncodedFrameObserver = exports.IMediaRecorderObserver = exports.IFaceInfoObserver = exports.IAudioSpectrumObserver = exports.IAudioPcmFrameSink = exports.IAudioFrameObserverBase = exports.IAudioFrameObserver = void 0;
var t = _interopRequireWildcard(require("ts-interface-checker"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/**
 * This module was automatically generated by `ts-interface-builder`
 */

// tslint:disable:object-literal-key-quotes

const IAudioPcmFrameSink = exports.IAudioPcmFrameSink = t.iface([], {
  "onFrame": t.opt(t.func("void", t.param("frame", "AudioPcmFrame")))
});
const IAudioFrameObserverBase = exports.IAudioFrameObserverBase = t.iface([], {
  "onRecordAudioFrame": t.opt(t.func("void", t.param("channelId", "string"), t.param("audioFrame", "AudioFrame"))),
  "onPlaybackAudioFrame": t.opt(t.func("void", t.param("channelId", "string"), t.param("audioFrame", "AudioFrame"))),
  "onMixedAudioFrame": t.opt(t.func("void", t.param("channelId", "string"), t.param("audioFrame", "AudioFrame"))),
  "onEarMonitoringAudioFrame": t.opt(t.func("void", t.param("audioFrame", "AudioFrame")))
});
const IAudioFrameObserver = exports.IAudioFrameObserver = t.iface(["IAudioFrameObserverBase"], {
  "onPlaybackAudioFrameBeforeMixing": t.opt(t.func("void", t.param("channelId", "string"), t.param("uid", "number"), t.param("audioFrame", "AudioFrame")))
});
const IAudioSpectrumObserver = exports.IAudioSpectrumObserver = t.iface([], {
  "onLocalAudioSpectrum": t.opt(t.func("void", t.param("data", "AudioSpectrumData"))),
  "onRemoteAudioSpectrum": t.opt(t.func("void", t.param("spectrums", t.array("UserAudioSpectrumInfo")), t.param("spectrumNumber", "number")))
});
const IVideoEncodedFrameObserver = exports.IVideoEncodedFrameObserver = t.iface([], {
  "onEncodedVideoFrameReceived": t.opt(t.func("void", t.param("uid", "number"), t.param("imageBuffer", "Uint8Array"), t.param("length", "number"), t.param("videoEncodedFrameInfo", "EncodedVideoFrameInfo")))
});
const IVideoFrameObserver = exports.IVideoFrameObserver = t.iface([], {
  "onCaptureVideoFrame": t.opt(t.func("void", t.param("sourceType", "VideoSourceType"), t.param("videoFrame", "VideoFrame"))),
  "onPreEncodeVideoFrame": t.opt(t.func("void", t.param("sourceType", "VideoSourceType"), t.param("videoFrame", "VideoFrame"))),
  "onMediaPlayerVideoFrame": t.opt(t.func("void", t.param("videoFrame", "VideoFrame"), t.param("mediaPlayerId", "number"))),
  "onRenderVideoFrame": t.opt(t.func("void", t.param("channelId", "string"), t.param("remoteUid", "number"), t.param("videoFrame", "VideoFrame"))),
  "onTranscodedVideoFrame": t.opt(t.func("void", t.param("videoFrame", "VideoFrame")))
});
const IFaceInfoObserver = exports.IFaceInfoObserver = t.iface([], {
  "onFaceInfo": t.opt(t.func("void", t.param("outFaceInfo", "string")))
});
const IMediaRecorderObserver = exports.IMediaRecorderObserver = t.iface([], {
  "onRecorderStateChanged": t.opt(t.func("void", t.param("channelId", "string"), t.param("uid", "number"), t.param("state", "RecorderState"), t.param("reason", "RecorderReasonCode"))),
  "onRecorderInfoUpdated": t.opt(t.func("void", t.param("channelId", "string"), t.param("uid", "number"), t.param("info", "RecorderInfo")))
});
const exportedTypeSuite = {
  IAudioPcmFrameSink,
  IAudioFrameObserverBase,
  IAudioFrameObserver,
  IAudioSpectrumObserver,
  IVideoEncodedFrameObserver,
  IVideoFrameObserver,
  IFaceInfoObserver,
  IMediaRecorderObserver
};
var _default = exports.default = exportedTypeSuite;
